<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.592">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Statistics in R - 8&nbsp; Binomial generalized linear models</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./2.X_mle.html" rel="next">
<link href="./2.X_introduction_to_GLMs.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>
</head>
<body class="nav-sidebar floating slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }"><div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">
<span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Binomial generalized linear models</span>
</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistics in R</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/biostats-r/biostats" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
<li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=%7Curl%7C">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=%7Curl%7C">
            <i class="bi bi-bi-facebook pe-1"></i>
          Facebook
          </a>
        </li>
    </ul>
<a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistics in R</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Introduction</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./0.1_introduction_to_stats.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction to statistics</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./0.2_types_of_data.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Types of data</span></a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Descriptive statistics</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1.0_descriptive_statistics.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Descriptive Statistics</span></a>
  </div>
</li>
    </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">Regression models</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.1_simple_linear_regression.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Simple linear Regression</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.2_anova.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Anova</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.X_introduction_to_GLMs.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Introduction to generalised linear models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.X_binomial_GLM.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Binomial generalized linear models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.X_mle.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Maximum likelihood estimation”</span></a>
  </div>
</li>
    </ul>
</li>
    </ul>
</div>
</nav><!-- margin-sidebar --><div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"> <span class="header-section-number">8.1</span> Introduction</a></li>
  <li><a href="#which-questions" id="toc-which-questions" class="nav-link" data-scroll-target="#which-questions"> <span class="header-section-number">8.2</span> <i class="far fa-question-circle"></i> Which questions?</a></li>
  <li>
<a href="#type-of-data" id="toc-type-of-data" class="nav-link" data-scroll-target="#type-of-data"> <span class="header-section-number">8.3</span> <i class="fas fa-table"></i> Type of data</a>
  <ul class="collapse">
<li><a href="#theory" id="toc-theory" class="nav-link" data-scroll-target="#theory"> <span class="header-section-number">8.3.1</span> Theory</a></li>
  <li><a href="#worked-example" id="toc-worked-example" class="nav-link" data-scroll-target="#worked-example"> <span class="header-section-number">8.3.2</span> Worked example</a></li>
  </ul>
</li>
  <li>
<a href="#model-details" id="toc-model-details" class="nav-link" data-scroll-target="#model-details"> <span class="header-section-number">8.4</span> <i class="fas fa-project-diagram"></i> Model details</a>
  <ul class="collapse">
<li><a href="#theory-1" id="toc-theory-1" class="nav-link" data-scroll-target="#theory-1"> <span class="header-section-number">8.4.1</span> Theory</a></li>
  <li><a href="#worked-example-1" id="toc-worked-example-1" class="nav-link" data-scroll-target="#worked-example-1"> <span class="header-section-number">8.4.2</span> Worked example</a></li>
  </ul>
</li>
  <li>
<a href="#parameters" id="toc-parameters" class="nav-link" data-scroll-target="#parameters"> <span class="header-section-number">8.5</span> <i class="fas fa-laptop"></i> Parameters</a>
  <ul class="collapse">
<li><a href="#theory-2" id="toc-theory-2" class="nav-link" data-scroll-target="#theory-2"> <span class="header-section-number">8.5.1</span> Theory</a></li>
  <li><a href="#worked-example-2" id="toc-worked-example-2" class="nav-link" data-scroll-target="#worked-example-2"> <span class="header-section-number">8.5.2</span> Worked example</a></li>
  </ul>
</li>
  <li>
<a href="#quantify-uncertainty" id="toc-quantify-uncertainty" class="nav-link" data-scroll-target="#quantify-uncertainty"> <span class="header-section-number">8.6</span> <i class="fas fa-arrows-alt-h"></i> Quantify uncertainty</a>
  <ul class="collapse">
<li><a href="#theory-3" id="toc-theory-3" class="nav-link" data-scroll-target="#theory-3"> <span class="header-section-number">8.6.1</span> Theory</a></li>
  <li><a href="#worked-example-3" id="toc-worked-example-3" class="nav-link" data-scroll-target="#worked-example-3"> <span class="header-section-number">8.6.2</span> Worked example</a></li>
  </ul>
</li>
  <li>
<a href="#model-checking" id="toc-model-checking" class="nav-link" data-scroll-target="#model-checking"> <span class="header-section-number">8.7</span> <i class="fas fa-tasks"></i> Model checking</a>
  <ul class="collapse">
<li><a href="#theory-4" id="toc-theory-4" class="nav-link" data-scroll-target="#theory-4"> <span class="header-section-number">8.7.1</span> Theory</a></li>
  <li><a href="#worked-example-4" id="toc-worked-example-4" class="nav-link" data-scroll-target="#worked-example-4"> <span class="header-section-number">8.7.2</span> Worked example</a></li>
  </ul>
</li>
  <li>
<a href="#draw-conclusions" id="toc-draw-conclusions" class="nav-link" data-scroll-target="#draw-conclusions"> <span class="header-section-number">8.8</span> <i class="far fa-lightbulb"></i> Draw conclusions</a>
  <ul class="collapse">
<li><a href="#theory-5" id="toc-theory-5" class="nav-link" data-scroll-target="#theory-5"> <span class="header-section-number">8.8.1</span> Theory</a></li>
  <li><a href="#worked-example-5" id="toc-worked-example-5" class="nav-link" data-scroll-target="#worked-example-5"> <span class="header-section-number">8.8.2</span> Worked example</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">
<span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Binomial generalized linear models</span>
</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header><div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Before you start
</div>
</div>
<div class="callout-body-container callout-body">
<p>Before reading this page, you should be comfortable with basic statistical theory, using R, continuous and categorical data, and <strong>linear models</strong>.</p>
<p>You should also review the basics of the Binomial distribution <a href="">here</a>. EGS: ADD LINK.</p>
</div>
</div>
<section id="introduction" class="level2" data-number="8.1"><h2 data-number="8.1" class="anchored" data-anchor-id="introduction">
<span class="header-section-number">8.1</span> Introduction</h2>
<p>In this section, we will look at how we can use a Binomial Generalized Linear Model (GLM) to analyze data with a binary response and numeric explanatory variables.</p>
<p></p>
<p>On this page, we focus on one particular type of GLM, the Binomial GLM. The Binomial GLM fits a non-linear line to:</p>
<ol type="1">
<li><p>estimate a relationship between <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span>, where <span class="math inline">\(Y\)</span> comes from binary trials and is therefore bounded between 0 and 1.</p></li>
<li><p>predict change in <span class="math inline">\(Y\)</span> from change in <span class="math inline">\(X\)</span>.</p></li>
</ol>
<p></p>
<p>But unlike a linear model, the Binomial GLM does not do this with a straight line (on the scale of the data). The Binomial GLM fits a curved line bounded between 0 and 1 on the y-axis. This is because the Binomial GLM (also called a logistic regression - more on why later), the response data are the result of binary trials e.g Yes or No, anything with only two outcomes. But it can represent the result of single trials or many e.g. did an individual survive or die (1 trial), or the proportion of survivors from a population of 10 (10 trials). The <span class="math inline">\(Y\)</span> values can only take values between 0 or 1 (either 0 and 1 themselves or proportions e.g.&nbsp;0.1).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-nonlinear-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-nonlinear-plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.1: Example of a fitted Binomial GLM model. The estimated relationship is plotted on the scale of the data.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>A large number of the models that are used in biological research are GLMs. This is because a lot of biological data would not meet the assumptions required for a linear model, for example survival data, occupancy data, or presence of a particular gene are all examples of binary responses.</p>
<p>As a result of their wide usage, GLMs are a key part of modern quantitative biology!</p>
<p></p>
</section><section id="which-questions" class="level2" data-number="8.2"><h2 data-number="8.2" class="anchored" data-anchor-id="which-questions">
<span class="header-section-number">8.2</span> <i class="far fa-question-circle"></i> Which questions?</h2>
<p>Example questions that can be answered with the Binomial GLM:</p>
<p>Inference</p>
<ul>
<li>How does body weight (kg) influence survival probability of sparrows?</li>
<li>How does forest cover (type) affect the occurrence probability of a plant species?</li>
</ul>
<p>Prediction</p>
<ul>
<li>What is the mortality of beetles exposed to different concentrations of carbon disulfide <span class="math inline">\(CS_2\)</span>?</li>
</ul>
<p></p>
</section><section id="type-of-data" class="level2 tabset tabset-fade" data-number="8.3"><h2 class="tabset tabset-fade anchored" data-number="8.3" data-anchor-id="type-of-data">
<span class="header-section-number">8.3</span> <i class="fas fa-table"></i> Type of data</h2>
<p></p>
<section id="theory" class="level3" data-number="8.3.1"><h3 data-number="8.3.1" class="anchored" data-anchor-id="theory">
<span class="header-section-number">8.3.1</span> Theory</h3>
<p>As mentioned above, in a Binomial GLM the response data are always the result of binary trials. The values of the response variable should have either only two outcomes or be the result of many trials (a number of successes and number of failures or a proportion). Examples of this type of data would be:</p>
<ul>
<li>Number that survived and number that died</li>
<li>Present/absent</li>
<li>Yes/no</li>
<li>On/off</li>
<li>Proportion of seedlings that germinated</li>
</ul>
<p>Just as was stated for the linear models, <strong>always remember to check that the way your variables are classified in R is the same as the format you expect.</strong></p>
<p>In this case, there are several options that will work. The response variable <span class="math inline">\(Y\)</span> can be stored as a factor, an integer, or as a continuous numeric variable.</p>
</section><section id="worked-example" class="level3" data-number="8.3.2"><h3 data-number="8.3.2" class="anchored" data-anchor-id="worked-example">
<span class="header-section-number">8.3.2</span> Worked example</h3>
<p>For this worked example, we will try to find out if/how body weight (kg) influences the survival probability of Soay sheep.</p>
<div class="cell">

</div>
<p>Biologically, it is expected that smaller sheep would find it harder to survive harsh winters as they have fewer reserves (such as body fat) than larger sheep.</p>
<section id="introduction-to-the-data" class="level4" data-number="8.3.2.1"><h4 data-number="8.3.2.1" class="anchored" data-anchor-id="introduction-to-the-data">
<span class="header-section-number">8.3.2.1</span> Introduction to the data</h4>
<p>This data for this example are from 1986 to 1996 for a population of Soay sheep from the St.&nbsp;Kilda islands in Scotland, UK.<br>
They have been monitored following a standardised protocol since 1985. Each year the sheep are caught, weighed, and the population size is counted. Information on the project can be found <a href="http://soaysheep.biology.ed.ac.uk/">here</a>. These data are open source and can be found in the appendix of this <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1600-0706.2012.00035.">paper</a> EGS: CITE COULSON 2012.</p>
<p>The data consists of five variables:</p>
<ul>
<li>Year - year of recording</li>
<li>Age - age of the sheep in years</li>
<li>Survival - whether the sheep survived until the next year or not</li>
<li>Weight - weight in kg</li>
<li>Population size (PopSize) - number of females counted that year</li>
</ul>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sheep_data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,328 × 5
    Year   Age Survival Weight PopSize
   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
 1  1987    11        1   26.5     331
 2  1988    12        0   27.2     457
 3  1987    10        1   26       331
 4  1988    11        0   26       457
 5  1986     8        1   26.7     211
 6  1987     9        0   24       331
 7  1986     8        1   26.5     211
 8  1987     9        1   25       331
 9  1988    10        0   23.2     457
10  1988    10        0   22.2     457
# … with 1,318 more rows</code></pre>
</div>
</div>
<p>All measures are for female sheep only.</p>
<p>You can find the data <a href="">here</a> if you want to follow along with the example. It is a <code>.csv</code> file with column headings.</p>
<p>To begin the analysis, let us first have a look at the data after it has been imported. We do this by looking at the raw data and making a plot. It can be easier to see the data if we jitter it because many data points sit on top of each other.</p>
<p>EGS: NEED TO ADD LINE TO READ IN DATA YOU CAN GET IT FROM THE PAPER LINK ABOVE THEN HOST IT.</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sheep_data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">Survival</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">geom_point</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"grey70"</span>, size <span class="op">=</span> <span class="fl">2</span>, alpha <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Weight of sheep (kg)"</span>,
      y <span class="op">=</span> <span class="st">"Survival of sheep"</span>,
      tag <span class="op">=</span> <span class="st">"a"</span><span class="op">)</span>

<span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sheep_data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">Survival</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">geom_jitter</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"grey70"</span>, size <span class="op">=</span> <span class="fl">2</span>, height <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span><span class="op">+</span>
 <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Weight of sheep (kg)"</span>,
      y <span class="op">=</span> <span class="st">"Survival of sheep"</span>,
      tag <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span>

<span class="va">plot1</span> <span class="op">+</span> <span class="va">plot2</span> <span class="op">+</span> <span class="fu">plot_layout</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep-data-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep-data-plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.2: Scatterplot of weight and against survival of sheep, plot a = raw data, plot b = jittered points</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Since we are interested in <strong>whether weight (kg) influences the survival of sheep</strong>, we will need two of the five variables in the data: <code>Weight</code> and <code>Survival</code>. We will not need <code>Year</code>, <code>PopSize</code> and <code>Age</code>.</p>
<p>The response variable then is <code>Survival</code>, which is a binary variable (it has two outcomes 0/1). The explanatory variable is <code>Weight</code>, which is continuous numeric.</p>
<p></p>
</section></section></section><section id="model-details" class="level2 tabset tabset-fade" data-number="8.4"><h2 class="tabset tabset-fade anchored" data-number="8.4" data-anchor-id="model-details">
<span class="header-section-number">8.4</span> <i class="fas fa-project-diagram"></i> Model details</h2>
<p></p>
<section id="theory-1" class="level3" data-number="8.4.1"><h3 data-number="8.4.1" class="anchored" data-anchor-id="theory-1">
<span class="header-section-number">8.4.1</span> Theory</h3>
<p></p>
<p>A Binomial GLM is just like all other statistical models, it aims to represent mathematically how the data that are being modelled were generated. In this case, it is assumed that the data were generated by a Binomial distribution and influenced by a relationship with some explanatory variables.</p>
<p>The sections below detail the three components of a GLM and describe what form they take for the Binomial GLM:</p>
<ul>
<li>The <span style="color:orange">systematic part</span>: <span class="math inline">\(\alpha\)</span> + <span class="math inline">\(\beta\)</span><span class="math inline">\(X_i\)</span>
</li>
<li>The <span style="color:purple">random part</span>: the distribution of the error</li>
<li>The <span style="color:magenta">link function</span>: that links them together</li>
</ul>
<p></p>
<p>In <i class="fab fa-r-project"></i> it looks like this:</p>
<p><code>glm(y ~ x, data, family = binomial(link = "logit"))</code></p>
<p>there are more details below in the <strong>fitting the model in <i class="fab fa-r-project"></i></strong> section.</p>
<p></p>
<p><strong>The <span style="color:orange">systematic part</span></strong></p>
<p>This is the part of the model that is hopefully the most familiar, it is a linear equation just like those used in linear models.</p>
<p>This is also known as the linear predictor, <span class="math inline">\(\eta\)</span>.</p>
<p>An example is <span class="math inline">\(\eta\)</span> = <span class="math inline">\(\alpha + \beta_1 X_{1} + \beta_2 X_{2}\)</span></p>
<p>where <span class="math inline">\(\alpha\)</span> = an intercept, and the <span class="math inline">\(\beta\)</span> values represent the change in <span class="math inline">\(\eta\)</span> with every unit change of <span class="math inline">\(X\)</span>.</p>
<p></p>
<p>In <i class="fab fa-r-project"></i> this is written as <code>y ~ x</code>.</p>
<p></p>
<p><strong>The <span style="color:purple">random part</span></strong></p>
<p>This part of the model represents the spread of the actual data points around the linear predictor. This random part is how we deal with the errors (residuals) of the model <span style="color:red"><span class="math inline">\(\varepsilon\)</span></span>. In linear models, this part was assumed to be Normal. In a Binomial GLM, it is assumed to be Binomially distributed (the clue is in the name).</p>
<p>The Binomial distribution is used to represent the number of successes (<span class="math inline">\(r\)</span>) from a number of independent trials (<span class="math inline">\(N\)</span>) when the probability of success (<span class="math inline">\(p\)</span>) is the same for each trial. We use it for data from binary trials.</p>
<p>The Binomial distribution has two parameters <span class="math inline">\(N\)</span> and <span class="math inline">\(p\)</span>. Usually, <span class="math inline">\(N\)</span> is known (it comes from the data), so there is only one unknown parameter that must be estimated. That is <span class="math inline">\(p\)</span>.</p>
<p></p>
<p>In <i class="fab fa-r-project"></i> the random part of the GLM is specified by the <code>family</code> argument for a Binomial GLM it is <code>family = binomial</code>.</p>
<p></p>
<p><strong>The <span style="color:magenta">Link function</span></strong></p>
<p>The link function transforms the <span style="color:orange">systematic part</span> of the model onto the scale of data (called the response scale), connecting it to the <span style="color:purple">random part</span>.</p>
<p>This step is necessary because the model is non-linear. If you were to directly estimate <span class="math inline">\(Y\)</span> (the response data) with a linear predictor as you do in a linear model, this would give a straight line. The straight line would not be bounded and could therefore predict values of <span class="math inline">\(Y\)</span> above 1 or below 0, which does not make any sense.</p>
<p>Instead, the Binomial GLM links the estimate from the linear predictor to <span class="math inline">\(Y\)</span> via a link function.</p>
<p>The default link function, or canonical function, for a Binomial distribution is the logit function (hence the alternative name of logistic regression). So, in a Binomial GLM the logit of the linear predictor is linked to <span class="math inline">\(Y\)</span> instead of the linear predictor itself.</p>
<p></p>
<p>In <i class="fab fa-r-project"></i> the link is specified within the <code>family</code> argument, but with an extra argument called <code>link</code>. For a Binomial GLM with a logit link it is <code>family = binomial(link = logit)</code>.</p>
<p></p>
<p>Using the logit link converts a probability to a log-odds. The log-odds is the natural logarithm of the odds ratio: <span class="math display">\[\frac{p}{1-p}\]</span>. Where <span class="math inline">\(p\)</span> = the probability of an event occurring.</p>
<p></p>
<p>The odds ratio is the probability of a event occurring (<span class="math inline">\(p\)</span>) over the probability it doesn’t occur (<span class="math inline">\(1-p\)</span>). Odds are often used in the things like betting, e.g.&nbsp;to represent the odds of a particular horse winning a race. An odds ratio of 10:1 means that the companies assume that for every 10 successes (for the company i.e.&nbsp;a horse loses and they keep the money) there will be 1 failure (the horse wins and the company has to pay).</p>
<p></p>
<p>The logit link takes the following form:</p>
<p><span class="math display">\[
\begin{aligned}
\mu = log(\frac{p}{1-p})
\end{aligned}
\]</span></p>
<p>Where <span class="math inline">\(\mu\)</span> = the log-odds, which is the output of the model.</p>
<p></p>
<p>The inverse of the logit link is:</p>
<p><span class="math display">\[
\begin{aligned}
p = \frac{e^{\mu}}{1+e^{\mu}}
\end{aligned}
\]</span></p>
<p>OR</p>
<p><span class="math display">\[
\begin{aligned}
p = \frac{1}{1+e^{-\mu}}
\end{aligned}
\]</span></p>
<p>Where <span class="math inline">\(p\)</span> = the probability of success.</p>
<p>Here <span class="math inline">\(\mu\)</span> is on the link scale and <span class="math inline">\(p\)</span> is on the response scale (the scale of the response variable).</p>
<p></p>
<p>Other possible link functions for a Binomial GLM are the <strong>probit</strong> and <strong>cloglog</strong> functions.</p>
For more information
<details><summary>
click here
</summary><p><strong>The Probit link</strong></p>
<p>One way of thinking of binomial problems is as a threshold:</p>
<p>e.g.&nbsp;imagine you have a dam, and if the water is too high, it will flow over the dam.</p>
<p>Therefore a model is needed that can capture when the dam overflows (it either does or does not). What is observed, and modelled, is whether the water was too high, or not. But, hidden underneath this is a variable of water height, which is not observed. But it is water height that controls if the dam overflows or not. An unobserved variable like this is called a <strong>latent variable</strong>.</p>
<p>This idea can be used in the modeling. If the water height was observed, it could be modelled with a simple linear regression of water height against rainfall, and assume the residuals are normally distributed. But, if only data on whether this value is above a certain threshold, or not is available. It turns out that this is the same as a Binomial GLM with a probit link!</p>
<p>Mathematically the model is <span class="math inline">\(Y_i = 1\)</span> if <span class="math inline">\(\mu_i &gt; 0\)</span>, where <span class="math inline">\(\mu_i\)</span> is the latent variable.</p>
<p>The threshold idea can be useful for interpreting models: if you think there is some unobserved variable that causes the binary response when it is above a threshold, it can be easier to understand the process.</p>
<p>In practice the estimates from the probit and logit link functions give almost the same results, even though they have different interpretations.</p>
<p>The probit uses an inverse normal link function where a higher mean = higher probability of success.</p>
<p><strong>Use when you want a threshold model because you have an unobserved variable that causes a binary response.</strong></p>
<p><strong>cloglog</strong></p>
<p>Using a cloglog link allows binary data to be linked to count data.</p>
<p>It is useful when the 0s and 1s actually come from counts, where the count is recorded as “0/zero” or “more than zero”. For example, the presence or absence of a species. In this case, the presence or absence is really a result of counting abundance of a species.</p>
<p>The cloglog link allows these binary data to be linked to a log(abundance) using the equations below.</p>
<p><span class="math display">\[
log(\lambda) = log(-log(1-p))
\]</span></p>
<p>Where <span class="math inline">\(\lambda\)</span> = the mean abundance and <span class="math inline">\(p\)</span> = probability of presence.</p>
<p><strong>Use when you want to link the binary data to abundance because they represent counts.</strong></p>
</details><p></p>
<p><strong>Assumptions</strong></p>
<p>There are several assumptions that should be met for the Binomial GLM to be valid.</p>
<ul>
<li><p>There are no outliers</p></li>
<li><p>Each value of Y is independent</p></li>
<li><p>The dispersion parameter is constant</p></li>
<li><p>The correct variance function is used (in a Binomial GLM this is assumed to be controlled by the mean)</p></li>
<li><p>The correct distribution is used (Binomial here - it is assumed that the random part of the model does follow a Binomial distribution)</p></li>
</ul>
<p></p>
<p>All of these assumptions should be met for the model to work properly and they ALWAYS need to be checked. Outliers, dispersion, and variance can be checked once the model has been fitted.</p>
<p>Independence of Y should be ensured during data collection and the correct distribution is determined by the characteristics of the data.</p>
<p></p>
<p><strong>Writing the model in <i class="fab fa-r-project"></i></strong></p>
<p>To fit the model in <i class="fab fa-r-project"></i>, we will use the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function.</p>
<p><code>glm(y ~ x, data, family = binomial(link = "logit"))</code></p>
<p></p>
<p>The <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function takes several arguments:</p>
<ul>
<li><p>formula in form: <code>y ~ x</code> (<span style="color:orange">systematic part</span>)</p></li>
<li><p>data: your data object.</p></li>
<li><p>family: specifies the distribution used for the <span style="color:purple">random part</span> of the model and the <span style="color:magenta">link function</span>. In this case, the model uses the binomial family with a logit link.</p></li>
</ul>
<p></p>
<p>The function will fit the GLM using <strong>maximum likelihood estimation</strong> and gives us the maximum likelihood estimates of <span style="color:orange"><span class="math inline">\(\alpha\)</span></span> and <span style="color:blue"><span class="math inline">\(\beta_is\)</span></span> as an output.</p>
<p>The formula part <code>y ~ x</code>, is the same as in the function <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code>. The <span class="math inline">\(y\)</span> is the response variable and <span class="math inline">\(x\)</span> is an explanatory variable. To run the GLM in <i class="fab fa-r-project"></i> first it is necessary to identify which variable is the response and which is/are explanatory.</p>
<p></p>
<p>There are two ways of fitting a Binomial GLM using the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function:</p>
<ul>
<li><p>Option 1: fitting response as a single factor (for single trials and proportions).</p></li>
<li><p>Option 2: fitting response as two columns (for multiple trials).</p></li>
</ul>
<p></p>
<p><strong>For data with results of single trials (0s and 1s) or proportions (0-1)</strong></p>
<p>The <span class="math inline">\(y\)</span> is the response variable and is a single vector of numbers.</p>
<p></p>
<p><strong>For data with results of multiple trials (number of successes and number of failures)</strong></p>
<p>The <span class="math inline">\(y\)</span> is still the response variable but instead of single column of 0s and 1s or proportions, there are two columns (one of successes and one of failures). The columns are combined using <code><a href="https://rdrr.io/r/base/cbind.html">cbind()</a></code> within the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function:</p>
<p><code>glm(cbind(success, failure) ~ x, data = data,  family = binomial(link = "logit"))</code></p>
<p>Doing this accounts for number of trials because <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> will also get information on the number of trials conducted (successes + failures).</p>
<p></p>
<p>Once you have an identified response and explanatory variable/variables, you can then plug these variables into the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function in the below format using the column names in place of <code>response</code> and <code>explanatory</code> and including your data frame name as the data argument.</p>
<p>Both modes of fitting are shown below.</p>
<p></p>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co">### single trial version</span>

<span class="va">glm_model_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">response</span> <span class="op">~</span> <span class="va">explanatory</span>, data <span class="op">=</span> <span class="va">your_data</span>,
                     family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">binomial</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"logit"</span><span class="op">)</span><span class="op">)</span>

<span class="co">### multiple trials version</span>

<span class="va">glm_multiple_trials</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">successes</span>, <span class="va">failures</span><span class="op">)</span> <span class="op">~</span> <span class="va">explanatory</span>, 
                           data <span class="op">=</span> <span class="va">example_data2</span>,
                     family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">binomial</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"logit"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Running the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> as illustrated above runs the GLM and saves the output as a <code>glm_model_object</code>.</p>
<p>We can then view our results from the model object by using the function <code><a href="https://rdrr.io/r/stats/coef.html">coef()</a></code>. This will take the output of the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>, the model object, as its argument and extracts the maximum likelihood estimates of <span style="color:orange"><span class="math inline">\(\alpha\)</span></span> and <span style="color:blue"><span class="math inline">\(\beta\)</span></span>.</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) explanatory 
 -1.5908333   0.2032877 </code></pre>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_multiple_trials</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> (Intercept)  explanatory 
 0.487107213 -0.005335974 </code></pre>
</div>
</div>
<p>You can see that the format of the output is the same regardless of whether it is a Binomial GLM on a set of single trials, or multiple trials.</p>
<p></p>
</section><section id="worked-example-1" class="level3" data-number="8.4.2"><h3 data-number="8.4.2" class="anchored" data-anchor-id="worked-example-1">
<span class="header-section-number">8.4.2</span> Worked example</h3>
<p></p>
<p>This worked example demonstrates how to fit a GLM in <span class="math inline">\(R\)</span> using the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function for the Soay sheep data example. This is an example of a single trial question.</p>
<p>In this example, we are asking: <strong>Does body weight affect the survival probability of sheep?</strong></p>
<p>The survival of the sheep (0 or 1) is the response (<span class="math inline">\(Y\)</span>) and body weight (kg) (<span class="math inline">\(X\)</span>) is the explanatory variable.</p>
<p>We put these variables in the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function in the below format.</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sheep_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">Survival</span> <span class="op">~</span> <span class="va">Weight</span>, data <span class="op">=</span> <span class="va">sheep_data</span>, 
                   family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">binomial</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"logit"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We have run the model and have assigned it to an object name. A logit link was used as these data do not come from a latent continuous variable or from counts, so we can use the default/canonical link function here.</p>
<p>Let us take a look at the maximum likelihood estimates of our model parameters (<span style="color:orange"><span class="math inline">\(\alpha\)</span></span>, <span style="color:blue"><span class="math inline">\(\beta\)</span></span>) using the <code><a href="https://rdrr.io/r/stats/coef.html">coef()</a></code>.</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept)      Weight 
 -2.1197077   0.1817796 </code></pre>
</div>
</div>
</section></section><section id="parameters" class="level2 tabset tabset-fade" data-number="8.5"><h2 class="tabset tabset-fade anchored" data-number="8.5" data-anchor-id="parameters">
<span class="header-section-number">8.5</span> <i class="fas fa-laptop"></i> Parameters</h2>
<p></p>
<section id="theory-2" class="level3" data-number="8.5.1"><h3 data-number="8.5.1" class="anchored" data-anchor-id="theory-2">
<span class="header-section-number">8.5.1</span> Theory</h3>
<p>We introduced the model parameters of a Binomial GLM in the model theory section above: <span style="color:orange"><span class="math inline">\(\alpha\)</span></span> = the intercept; <span style="color:blue"><span class="math inline">\(\beta_i\)</span></span> = the slope of the model line (steepness/gradient).</p>
<p></p>
<p><strong>But what do these parameters really mean?</strong></p>
<p>To fully understand the parameters and what they mean, we have to note that there is a non-linear relationship between the explanatory and response variable on the scale of the observed data.</p>
<p>To interpret the parameters, we have to recall the linear predictor and the link function.</p>
<p>The linear predictor is <span class="math inline">\(\eta = \color{orange}\alpha + \color{blue}{\beta_i}X_i\)</span>. The <span class="math inline">\(\eta\)</span> in this case is the log-odds of the probability of success (<span class="math inline">\(p\)</span>).<br><span class="math inline">\(p\)</span> is what we are trying to estimate.</p>
<p></p>
<p>That is:</p>
<p><span class="math display">\[
log\bigg(\frac{p}{1-p}\bigg) = \eta = \color{orange}\alpha + \color{blue}{\beta}X_i.
\]</span></p>
<p>You might recognise the first part of this formula from the introduction of the logit link function above, it is the logit link. Note that on the scale of the link, the model line will be straight. However, on the response scale will be curved.</p>
<p>To relate the linear predictor back to the response scale of the data and <span class="math inline">\(p\)</span>. It is necessary to take the inverse of the link function like using the below equation:</p>
<p><span class="math display">\[
p = \frac{1}{1 + e^{- \eta}} = \frac{1}{1 + e^{-(\color{orange}\alpha + \color{blue}{\beta}X_i)}}.
\]</span></p>
<p>Therefore, it is possible to interpret the parameters on the response scale or on the link scale. We will discuss both options here.</p>
<p></p>
<p><strong>Parameters on the logit scale</strong></p>
<p>On the logit scale, the response will be the log odds of <span class="math inline">\(Y\)</span>.</p>
<p>To get to the log odds there are a few steps.</p>
<ol type="1">
<li><p>We need the probability of success: <span class="math inline">\(\frac{1}{(1+10)}\)</span> = <span class="math inline">\(0.09\)</span></p></li>
<li><p>Next we use that probability to find the odds: <span class="math inline">\(\frac{0.09}{(1-0.09)} = 0.0989\)</span></p></li>
<li><p>To get to the log odds, we take the log of 2: <span class="math inline">\(log(\frac{0.09}{(1-0.09)}) = -2.31\)</span></p></li>
</ol>
<p>Step 3 should look familiar. It is the same as the link function in the Binomial GLM.</p>
<p></p>
<p><strong><span style="color:orange"><span class="math inline">\(\alpha\)</span>, the intercept</span></strong></p>
<p>This is the parameter that gives the value of the log odds of <span class="math inline">\(Y\)</span> when <span class="math inline">\(X = 0\)</span>.</p>
<p><strong><span style="color:blue"><span class="math inline">\(\beta\)</span>, the slope</span></strong></p>
<p>This gives the amount of change in the log odds of <span class="math inline">\(Y\)</span> as the value of <span class="math inline">\(X\)</span> changes by one unit.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-log-odds-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-log-odds-plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.3: Example of a Binomial GLM model line on the logit scale. Intercept in orange and slope in blue</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p><strong>Parameters on the response scale</strong></p>
<p>On the response scale, the response will be <span class="math inline">\(Y\)</span>, which is a probability of success (e.g.&nbsp;survival, presence etc).</p>
<p><strong><span style="color:orange"><span class="math inline">\(\alpha\)</span>, the intercept</span></strong></p>
<p>To find the expected value of <span class="math inline">\(Y\)</span> when <span class="math inline">\(X = 0\)</span>. You need to convert the estimate of <span class="math inline">\(\alpha\)</span> from the logit scale back to the response scale using the equation for the inverse of the logit link: e.g.</p>
<p><span class="math display">\[
p = \frac{1}{1+e^{-(\alpha)}}
\]</span></p>
<p>This can be done in R.</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># estimate of the intercept on logit scale (log odds of y)</span>

<span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) 
  -1.590833 </code></pre>
</div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># convert to response scale (a probability of success)</span>

<span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) 
  0.1692667 </code></pre>
</div>
</div>
<p></p>
<p><strong><span style="color:blue"><span class="math inline">\(\beta\)</span>, the slope</span></strong></p>
<p>On the response scale the slope parameter has a somewhat different meaning. On the logit scale, the slope captured a linear relationship. However, on the response scale, the relationship between <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> is not linear. As a result, the gradient of the model line will be different for different <span class="math inline">\(X\)</span> values (see <a href="#fig-response-scale-example">Figure&nbsp;<span>8.4</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-response-scale-example" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-response-scale-example-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.4: Example of a Binomial GLM model line on the scale of the response data. Intercept in orange and slope in blue</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>The estimate of the <span class="math inline">\(\beta\)</span> coefficient on the logit scale, will tell you about the strength and direction of the relationship between <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span>. This is the case even though this relationship will not be linear on the response scale.</p>
<p>To see the change in slope of the model line on the response scale you can predict values of <span class="math inline">\(Y\)</span> on the response scale and either plot the relationship (like above in <a href="#fig-response-scale-example">Figure&nbsp;<span>8.4</span></a>) or simply look at the change between two <span class="math inline">\(X\)</span> values.</p>
<p>e.g.</p>
<div class="cell">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># predict log odds of Y when X = 2 or X = 3</span>
<span class="co"># use a linear equation for this</span>

<span class="va">odds_y_x2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">+</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">*</span><span class="fl">2</span><span class="op">)</span>
<span class="va">odds_y_x3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">+</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">*</span><span class="fl">3</span><span class="op">)</span>

<span class="co"># estimate of probability of success when X = 2</span>
<span class="co"># take inverse of the logit link</span>

<span class="va">prob_X2</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="op">-</span><span class="va">odds_y_x2</span><span class="op">)</span><span class="op">)</span>

<span class="co"># estimate of probability of success when X = 3</span>

<span class="va">prob_X3</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="op">-</span><span class="va">odds_y_x3</span><span class="op">)</span><span class="op">)</span>

<span class="co"># slope from X = 2 to X = 3 on response scale</span>

<span class="va">prob_X3</span> <span class="op">-</span> <span class="va">prob_X2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) 
 0.03841184 </code></pre>
</div>
</div>
<p>If you were to repeat the example above for <span class="math inline">\(X\)</span> = 10 and <span class="math inline">\(X\)</span> = 11, you would not get the same slope on the response scale. Even though it is the same on the logit scale.</p>
<p></p>
</section><section id="worked-example-2" class="level3" data-number="8.5.2"><h3 data-number="8.5.2" class="anchored" data-anchor-id="worked-example-2">
<span class="header-section-number">8.5.2</span> Worked example</h3>
<p>In the previous section, we fit a Binomial GLM using the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> function and looked at the estimates of some parameters using the <code><a href="https://rdrr.io/r/stats/coef.html">coef()</a></code> function. In this section, we will use the model theory to interpret what those parameters mean.</p>
<section id="the-intercept" class="level4" data-number="8.5.2.1"><h4 data-number="8.5.2.1" class="anchored" data-anchor-id="the-intercept">
<span class="header-section-number">8.5.2.1</span> <span style="color:orange">The intercept</span>
</h4>
<p>For our sheep data model, the estimate of the intercept is:</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) 
  -2.119708 </code></pre>
</div>
</div>
<p>This is the estimate of the log odds of survival for a sheep which has a body weight of 0 kg. The intercept estimate is a log odds of -2.1197077 of survival for a sheep that weighs 0 kg.</p>
<p>To get the probability of survival for a sheep with a body weight of 0 kg we need to convert these estimates to the response scale using the code below:</p>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>(Intercept) 
   0.107196 </code></pre>
</div>
</div>
<p>In other words, the survival probability of a sheep is 0.107196 when their weight is 0 kg.</p>
<p>In this example, the intercept is not very interesting, as it does not make a lot of biological sense to know the expected survival probability of sheep when their weight is 0 kg. A sheep born at anytime will weigh more than 0 kg.</p>
</section><section id="the-slope" class="level4" data-number="8.5.2.2"><h4 data-number="8.5.2.2" class="anchored" data-anchor-id="the-slope">
<span class="header-section-number">8.5.2.2</span> <span style="color:blue">The slope</span>
</h4>
<p>The estimate of the slope of the relationship between body weight and survival probability is interesting to us since it can tell us the direction and the strength of the effect of body weight on survival of the sheep.</p>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Weight 
0.1817796 </code></pre>
</div>
</div>
<p>In this case, our model estimates that the log odds of the survival of the sheep increases by 0.1817796 when the weight of the sheep increases by 1 kg. This shows a positive relationship between body weight and survival probability.</p>
<p>To see what this effect looks like on the response scale, it is easiest to plot the results using predictions on the response scale (response scale). This is what we do below.</p>
</section><section id="plotting-the-results" class="level4" data-number="8.5.2.3"><h4 data-number="8.5.2.3" class="anchored" data-anchor-id="plotting-the-results">
<span class="header-section-number">8.5.2.3</span> Plotting the Results</h4>
<p>As well as looking at the maximum likelihood estimates of the parameters from the Binomial GLM, we can also plot the results. This is especially useful for interpreting the results on the response data scale.</p>
<p>To do this, we use the <code>ggplot()</code> with <code>geom_line()</code> and the <code><a href="https://rdrr.io/r/stats/predict.html">predict()</a></code> function.</p>
<p>To make the first plot, we will only need to use two arguments:</p>
<ul>
<li>
<code>object</code> = your model object</li>
<li>
<code>type</code> = “link”, which means predict on the link scale or “response”, which means predict on the response scale.</li>
</ul>
<p>We will make predictions on both the link and response scale so that we can compare the results.</p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sheep_data</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">sheep_data</span>, 
                     predictions_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>, 
                                            type <span class="op">=</span> <span class="st">"link"</span><span class="op">)</span>,
                     predictions_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>, 
                                            type <span class="op">=</span> <span class="st">"response"</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once we have created predictions of <span class="math inline">\(Y\)</span> from the model object, we can plot these using <code>geom_line()</code> as in the code below.</p>
<div class="cell">
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">link</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sheep_data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Weight</span>, y<span class="op">=</span><span class="va">predictions_link</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">predictions_link</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Weight (kg)"</span>, y <span class="op">=</span> <span class="st">"Log Odds of Survival"</span><span class="op">)</span>

<span class="va">response</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sheep_data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Weight</span>, y<span class="op">=</span><span class="va">Survival</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"grey70"</span>, size <span class="op">=</span> <span class="fl">2</span>,
              height <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">predictions_response</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Weight (kg)"</span>, y <span class="op">=</span><span class="st">"Survival Probability"</span><span class="op">)</span>

<span class="va">link</span> <span class="op">+</span> <span class="va">response</span> <span class="op">+</span> <span class="fu">plot_layout</span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep-preditions1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep-preditions1-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.5: a) Plot of fitted GLM line on the logit scale b) Plot of jittered raw data and fitted GLM line for the soay sheep data on the response scale</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>In the next section, we will look at how to add uncertainty to these plots and our interpretation.</p>
<p></p>
</section></section></section><section id="quantify-uncertainty" class="level2 tabset tabset-fade" data-number="8.6"><h2 class="tabset tabset-fade anchored" data-number="8.6" data-anchor-id="quantify-uncertainty">
<span class="header-section-number">8.6</span> <i class="fas fa-arrows-alt-h"></i> Quantify uncertainty</h2>
<p></p>
<section id="theory-3" class="level3" data-number="8.6.1"><h3 data-number="8.6.1" class="anchored" data-anchor-id="theory-3">
<span class="header-section-number">8.6.1</span> Theory</h3>
<p>As was discussed for linear models, statistics does not give a single correct answer. When we estimate the parameters in our statistical model, there will be many plausible parameters that could have produced our observed data. In these cases, some of the parameters will be more likely than the others to have generated the observed data.</p>
<p>The Binomial GLM is no exception. We will have to consider and present the uncertainty in the parameters we estimate.</p>
<p>The <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code>function uses maximum likelihood estimation for the parameter estimation. Therefore, our consideration of uncertainty for these models are discussed <a href="link%20to%20ML%20page">here</a>. We will therefore quantify uncertainty using <em>standard errors</em>, <em>confidence intervals</em> and <em>prediction intervals</em> which should be familiar to you but head to the <a href="">uncertainty</a> pages if you need a recap.</p>
<p>Just as with linear models there are two different types of uncertainty we will look at <em>uncertainty in the model line of parameters <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span></em> and <em>uncertainty in a prediction of <span class="math inline">\(Y\)</span></em>.</p>
<p></p>
<p><strong>Uncertainty in the estimates of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span></strong></p>
<p>To find the standard errors for the estimates of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> we can use <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> function. This takes the model object from the <code><a href="https://rdrr.io/r/stats/glm.html">glm()</a></code> as its argument. It outputs a big table with lots of information. The first line gives the formula used for the model object. The second line shows the summary of the residuals of the model and the standard errors are shown in the second column of the third part, <code>Coefficients:</code>.</p>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = response ~ explanatory, family = binomial(link = "logit"), 
    data = your_data)

Deviance Residuals: 
     Min        1Q    Median        3Q       Max  
-1.98905  -1.04140   0.00328   1.01222   1.75292  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -1.59083    0.52309  -3.041  0.00236 ** 
explanatory  0.20329    0.06106   3.329  0.00087 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 138.63  on 99  degrees of freedom
Residual deviance: 124.79  on 98  degrees of freedom
AIC: 128.79

Number of Fisher Scoring iterations: 4</code></pre>
</div>
</div>
<p></p>
<p>By taking the <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> of the example model, we can see the standard error of the intercept (<span class="math inline">\(\alpha\)</span>) is 0.52, and the standard error for the slope (<span class="math inline">\(\beta\)</span>) is 0.06.</p>
<p>We also observe that the standard error is smaller than the estimated effects. Actually, to indicate a clear direction of the effect, the standard error should be less than half the size of the estimated coefficient. This would ensure that 95% confidence intervals for the effect (see below) do not span 0.</p>
<p></p>
<p><strong>Confidence intervals</strong></p>
<p>For interpretation of the uncertainty, it can be easier to use the standard error to calculate confidence intervals. Confidence intervals indicate the range of plausible values for a parameter. <strong>They represent an interval, that if you were to collect a sample and run the analysis, then repeat that many many times AND each time draw a confidence interval, on average 95% of the time, the true population value of the parameter would be found in within the confidence interval.</strong></p>
<p></p>
<p>To calculate a confidence interval from a standard error you need to use the formulas (this is exactly the same as for linear models):</p>
<p><span class="math display">\[
\begin{aligned}
UpperCI = estimate + (1.96 SE) \\
LowerCI = estimate - (1.96 SE) \\
\end{aligned}
\]</span></p>
<p>Remember that the estimates of the coefficients are for the effect of <span class="math inline">\(X\)</span> on the log odds of <span class="math inline">\(Y\)</span> (i.e.&nbsp;on the link scale). Therefore any measures of uncertainty are also on the link scale. After estimating the upper and lower confidence intervals of these parameters, it might be useful to convert them to the response scale to understand their effect. This is especially true for the intercept (<span class="math inline">\(\alpha\)</span>).</p>
<p></p>
<p>Using the <code><a href="https://rdrr.io/r/stats/confint.html">confint()</a></code> function it is possible to display the 95% confidence intervals for the intercept and slope.</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/confint.html">confint</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  2.5 %     97.5 %
(Intercept) -2.68608847 -0.6222264
explanatory  0.09118142  0.3320830</code></pre>
</div>
</div>
<p></p>
<p><strong>Plotting uncertainty in <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span></strong></p>
<p>The above section has shown how you can quantify uncertainty in the line parameters of a Binomial GLM as numbers. But often in science, it is clearer to show these things visually.</p>
<p>We can add the confidence intervals to <a href="#fig-log-odds-plot">Figure&nbsp;<span>8.3</span></a> showing our GLM fitted line. This can be done either on the link scale or the response scale.</p>
<p>To do this, we need to generate new predictions with the <code><a href="https://rdrr.io/r/stats/predict.html">predict()</a></code> function. We also need to calculate the upper and lower confidence interval bounds using the standard error.</p>
<p>This is a little different to using <code><a href="https://rdrr.io/r/stats/predict.html">predict()</a></code> for a linear model object. For GLMs there are some different arguments. The key arguments are:</p>
<ul>
<li>
<code>type</code>. This argument can take values “link” (gives predictions on the link scale), “response” (gives predictions on the response scale) or “terms” (gives a matrix of the fitted values of each term in the model formula on the link scale).</li>
<li>
<code>se.fit</code>. This argument can be <code>TRUE</code> or <code>FALSE</code> and indicates whether the standard error for the prediction should be given in addition to the mean prediction. The standard error will be on the scale of the mean prediction i.e. either link scale or response dependent on which <code>type</code> was specified.</li>
</ul>
<p></p>
<p>The <code>interval</code> argument that was used for linear model objects to specify a prediction or a confidence interval does not work for predictions on a GLM object. Therefore, the standard errors presented are for a 95% confidence interval.</p>
<div class="cell">
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># First create the new data to predict for </span>
<span class="co"># This example generates 100 values of the explanatory variable from min to max</span>

<span class="va">newdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>explanatory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">$</span><span class="va">explanatory</span><span class="op">)</span>,
                                        <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">$</span><span class="va">explanatory</span><span class="op">)</span>,
                                        length.out <span class="op">=</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Then add predictions to the newdata for the link and response scale</span>
<span class="co"># REMEMBER the confidence intervals too, you can access these by taking the </span>
<span class="co"># $se.fit of the results of `predict`, $fit is the mean prediction</span>

<span class="va">predictions</span> <span class="op">&lt;-</span> <span class="va">newdata</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>predictions_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">glm_model_object</span>, 
                                                         newdata <span class="op">=</span> <span class="va">newdata</span>, 
                                                         type<span class="op">=</span><span class="st">"link"</span>, 
                                                         se.fit<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span>,
                              SE_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">glm_model_object</span>, 
                                                         newdata <span class="op">=</span> <span class="va">newdata</span>, 
                                                         type<span class="op">=</span><span class="st">"link"</span>, 
                                                         se.fit<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span>,
                              predictions_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">glm_model_object</span>, 
                                                         newdata <span class="op">=</span> <span class="va">newdata</span>, 
                                                         type<span class="op">=</span><span class="st">"response"</span>, se.fit<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span>,
                              SE_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">glm_model_object</span>, 
                                                         newdata <span class="op">=</span> <span class="va">newdata</span>, 
                                                         type<span class="op">=</span><span class="st">"response"</span>,
                                                         se.fit<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span><span class="op">)</span> <span class="op">%&gt;%</span>
                <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span>



<span class="va">predictions</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 100 × 5
   explanatory predictions_link SE_link predictions_response SE_response
         &lt;dbl&gt;            &lt;dbl&gt;   &lt;dbl&gt;                &lt;dbl&gt;       &lt;dbl&gt;
 1       -4.12            -2.43   0.759               0.0811      0.0566
 2       -3.88            -2.38   0.745               0.0848      0.0578
 3       -3.64            -2.33   0.731               0.0887      0.0591
 4       -3.40            -2.28   0.717               0.0927      0.0603
 5       -3.16            -2.23   0.703               0.0968      0.0615
 6       -2.92            -2.18   0.690               0.101       0.0627
 7       -2.68            -2.14   0.676               0.106       0.0639
 8       -2.44            -2.09   0.662               0.110       0.0650
 9       -2.20            -2.04   0.648               0.115       0.0661
10       -1.96            -1.99   0.634               0.120       0.0671
# … with 90 more rows</code></pre>
</div>
</div>
<p></p>
<p>You should notice that all predictions on the response scale around bounded between 0 and 1, but this is not the case on the link scale.</p>
<p></p>
<p>Then you can plot these. Remember these are confidence intervals not prediction intervals. The <code>predict</code> function will not give prediction intervals for a GLM. In fact, it is not always an easy or sensible thing to create in this context, especially for a Binomial GLM. When the mean prediction is close to 0 or 1, the prediction interval can easily fill the whole possible value space (i.e.&nbsp;span from 0 right to 1), which is not very logical and is hard to interpret.</p>
<p>To get the upper and lower bounds of the confidence interval from a GLM using <code><a href="https://rdrr.io/r/stats/predict.html">predict()</a></code> you need to calculate the bounds yourself. You need to add or subtract 1.96 times the standard error from the mean prediction.</p>
<div class="cell">
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="va">predictions</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>upper_ci_link <span class="op">=</span> <span class="va">predictions_link</span> <span class="op">+</span> <span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="va">SE_link</span><span class="op">)</span>,
                                      lower_ci_link <span class="op">=</span> <span class="va">predictions_link</span> <span class="op">-</span> <span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="va">SE_link</span><span class="op">)</span>,
                                      upper_ci_response <span class="op">=</span> <span class="va">predictions_response</span> <span class="op">+</span> <span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="va">SE_response</span><span class="op">)</span>,
                                      lower_ci_response <span class="op">=</span> <span class="va">predictions_response</span> <span class="op">-</span> <span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="va">SE_response</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot predictions on the link scale </span>

<span class="va">example_predictions_link</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">predictions</span>, 
                                 <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">explanatory</span>, 
                                     y <span class="op">=</span> <span class="va">predictions_link</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span>color<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_ribbon</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">lower_ci_link</span>, 
                  ymax <span class="op">=</span> <span class="va">upper_ci_link</span><span class="op">)</span>, 
              alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"A. Predictions on link scale"</span>,
       y <span class="op">=</span> <span class="st">"Log odds of response"</span>,
       x <span class="op">=</span> <span class="st">"Explanatory"</span><span class="op">)</span>

<span class="co"># For this plot, the base can be the response data</span>
<span class="va">example_predictions_response</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span>data <span class="op">=</span> <span class="va">example_data</span>, 
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">explanatory</span>, y <span class="op">=</span> <span class="va">response</span><span class="op">)</span>, 
              colour <span class="op">=</span> <span class="st">"grey"</span>, size <span class="op">=</span> <span class="fl">2</span>, height <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_ribbon</span><span class="op">(</span>data <span class="op">=</span> <span class="va">predictions</span>,
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">explanatory</span>,
                  ymin <span class="op">=</span> <span class="va">lower_ci_response</span>, 
                  ymax <span class="op">=</span> <span class="va">upper_ci_response</span><span class="op">)</span>, 
              alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span>data <span class="op">=</span> <span class="va">predictions</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">explanatory</span>, y <span class="op">=</span> <span class="va">predictions_response</span><span class="op">)</span>, 
            colour <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"B. Predictions on response scale"</span>,
       y <span class="op">=</span> <span class="st">"Probability"</span>,
       x <span class="op">=</span> <span class="st">"Explanatory"</span><span class="op">)</span>

<span class="va">example_predictions_link</span> <span class="op">+</span> <span class="va">example_predictions_response</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-plot_ci_example" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-plot_ci_example-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.6: Examples of plots of fitted GLM line and 95% confidence intervals (grey shaded area) on the link scale (A) and on the response scale (B)</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section><section id="worked-example-3" class="level3" data-number="8.6.2"><h3 data-number="8.6.2" class="anchored" data-anchor-id="worked-example-3">
<span class="header-section-number">8.6.2</span> Worked example</h3>
<p>At the end of the last section, we created of our sheep model and looked at the estimated model line. Now we will add uncertainty to that plot.</p>
<p>First, we should look at the confidence intervals of our parameter estimates.</p>
<div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/confint.html">confint</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span>,<span class="fl">2</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            2.5 % 97.5 %
(Intercept) -2.63  -1.62
Weight       0.15   0.21</code></pre>
</div>
</div>
<p>The confidence interval has been rounded to two decimal places to make it easier to read.</p>
<p>To add these intervals to the plot, we need to make new predictions including the upper and lower bounds of the prediction interval. We will do this on the link scale and the response response scale.</p>
<div class="cell">
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># first make some new data to predict for</span>
<span class="va">newdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Weight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">sheep_data</span><span class="op">$</span><span class="va">Weight</span><span class="op">)</span>,
               <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">sheep_data</span><span class="op">$</span><span class="va">Weight</span><span class="op">)</span>,
               length.out <span class="op">=</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span>

<span class="va">sheep_predictions</span> <span class="op">&lt;-</span> <span class="va">newdata</span> <span class="op">%&gt;%</span> 
  <span class="fu">mutate</span><span class="op">(</span>predictions_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"link"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span>,
            <span class="co"># the next lines add the lower and upper ci bounds</span>
            lower_ci_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"link"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span> <span class="op">-</span> 
                                    <span class="op">(</span><span class="fl">1.96</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"link"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span><span class="op">)</span>,
            upper_ci_link <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>, 
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"link"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span> <span class="op">+</span> 
                                    <span class="op">(</span><span class="fl">1.96</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"link"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span><span class="op">)</span>,
            <span class="co"># then predict on the response scale</span>
            predictions_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"response"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span>,
            <span class="co"># and add the confidence interval bounds</span>
            lower_ci_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>, 
                                    type<span class="op">=</span><span class="st">"response"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span> <span class="op">-</span> 
                                    <span class="op">(</span><span class="fl">1.96</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"response"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span><span class="op">)</span>,
            upper_ci_response <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>, 
                                    type<span class="op">=</span><span class="st">"response"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">fit</span> <span class="op">+</span> 
                                    <span class="op">(</span><span class="fl">1.96</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">sheep_model</span>,
                                    <span class="va">newdata</span>,
                                    type<span class="op">=</span><span class="st">"response"</span>, 
                                    se.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">se.fit</span><span class="op">)</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p></p>
<p>Once we have created predictions of <span class="math inline">\(Y\)</span> from the model object, we can plot them using <code>geom_line()</code> and <code>geom_ribbon()</code> as in the code below.</p>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="co"># Plot of Confidence Intervals on the link scale</span>

<span class="va">sheep_predictions_link</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_predictions</span>, 
                                 <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, 
                                     y <span class="op">=</span> <span class="va">predictions_link</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span>color<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_ribbon</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">lower_ci_link</span>, 
                  ymax <span class="op">=</span> <span class="va">upper_ci_link</span><span class="op">)</span>, 
              alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"A. Predictions on link scale"</span>,
       y <span class="op">=</span> <span class="st">"Log odds of survival"</span>,
       x <span class="op">=</span> <span class="st">"Body weight (kg)"</span><span class="op">)</span>

<span class="co"># For this plot, the base can be the response data</span>
<span class="va">sheep_predictions_response</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_data</span>, 
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">Survival</span><span class="op">)</span>, 
              colour <span class="op">=</span> <span class="st">"grey"</span>, size <span class="op">=</span> <span class="fl">2</span>, height <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_ribbon</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_predictions</span>,
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>,
                  ymin <span class="op">=</span> <span class="va">lower_ci_response</span>, 
                  ymax <span class="op">=</span> <span class="va">upper_ci_response</span><span class="op">)</span>, 
              alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_predictions</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">predictions_response</span><span class="op">)</span>, 
            colour <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"B. Predictions on response scale"</span>,
       y <span class="op">=</span> <span class="st">"Probability of survival"</span>,
       x <span class="op">=</span> <span class="st">"Body weight (kg)"</span><span class="op">)</span>


<span class="va">sheep_predictions_link</span> <span class="op">+</span> <span class="va">sheep_predictions_response</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep_predictions_plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep_predictions_plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.7: A. plot of the fitted GLM line on the link scale including 95% confidence interval (grey), B. plot of the fitted GLM line on the response scale including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>It can be observed from the figure above that the confidence intervals for the predictions around <span class="math inline">\(1\)</span> are very narrower. This means that as we get closer to the true classification (0 or 1), we are able to estimate them with much certainty.</p>
<p></p>
</section></section><section id="model-checking" class="level2 tabset tabset-fade" data-number="8.7"><h2 class="tabset tabset-fade anchored" data-number="8.7" data-anchor-id="model-checking">
<span class="header-section-number">8.7</span> <i class="fas fa-tasks"></i> Model checking</h2>
<p></p>
<section id="theory-4" class="level3" data-number="8.7.1"><h3 data-number="8.7.1" class="anchored" data-anchor-id="theory-4">
<span class="header-section-number">8.7.1</span> Theory</h3>
<p>Until this point, we have run a Binomial GLM, got parameter estimates and quantified uncertainty. Next, <strong>we will look at how to know if the model is any good.</strong> To do this, we need to check if the assumptions for a Binomial GLM are met for our data.</p>
<p></p>
<p>To check the assumptions of a Binomial GLM, we will follow the same procedure as for GLMs more broadly. For a reminder of this, go back to the <a href="">Introduction to GLMs page</a>.</p>
<p></p>
<p>The first step is to:</p>
<p><strong>Plot the model predictions and the data together</strong></p>
<p>From this plot, it is possible to test:</p>
<ul>
<li>that the fitted model line is capturing the relationship between x and y on the response scale</li>
</ul>
<p></p>
<p>To do this, we can re-plot the example plot from the section above.</p>
<div class="cell">
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">example_predictions_response</span> <span class="op">+</span>
    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Model predictions and data on response scale"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-example_predictions_plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-example_predictions_plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.8: plot of the fitted GLM line on the response scale including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>If you look carefully at <a href="#fig-example_predictions_plot">Figure&nbsp;<span>8.8</span></a> you can see that there are more data points that are 0 for the response for negative values of the explanatory variable. As the values of the explanatory variable increase, we find progressively more 1 values for the response. Therefore, we would expect a model line that that an intercept of close to 0 on the response scale and then shows a positive relationship between x and y, reaching an asymptote (levelling off) at a response value of 1 at the extreme end of the observed values of the explanatory variable (around 20). This is exactly what the fitted model line does! Therefore, in this case, it is possible to see that the model fits the data quite well.</p>
<p></p>
<p>Note that if you had proportional data or those from multiple trials that the data points might be distributed around the fitted line.</p>
<p>Then it would be possible to also test:</p>
<ul>
<li>whether the data points are evenly distributed around the fitted line</li>
</ul>
<div class="cell">
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">explanatory</span>, y <span class="op">=</span> <span class="va">successes</span><span class="op">/</span><span class="op">(</span><span class="va">successes</span> <span class="op">+</span> <span class="va">failures</span><span class="op">)</span><span class="op">)</span>, 
       data <span class="op">=</span> <span class="va">example_data2</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">geom_smooth</span><span class="op">(</span>method<span class="op">=</span><span class="fu">stats</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span>, method.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span>, 
              se<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-example_predictions_plot_proportion" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-example_predictions_plot_proportion-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.9: plot of the fitted GLM line on the response scale with proportional response, including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>In this example, the fit is much worse. Although there are an even number of data points above and below the fitted model line, they are very widely spread away from it. The model line does not seem to be explaining the data well, that is partly because there is so little data (only 10 data points). But hopefully shows what a less well fitting model looks like.</p>
<p> </p>
<p>The second step for checking model assumptions is:</p>
<p></p>
<p><strong>Overdispersion</strong></p>
<p>Checking for overdispersion tests if these assumptions are met:</p>
<ul>
<li><p>Correct variance function is used</p></li>
<li><p>Dispersion parameter is constant</p></li>
</ul>
<p>In a Binomial GLM it is assumed that the variance is controlled by the mean. But this is not always true. It is something we need to check as part of the model checking process.</p>
<p></p>
<p>To check for overdispersion in GLMs we need to calculate the deviance ratio. This is done by taking the ratio of residual deviance and residual degrees of freedom.</p>
<p>The assumption is that the deviance = 1 for Binomial GLMs. Values higher than 2 indicates that the residuals likely have more dispersion than expected. But there is no fixed threshold for when overdispersion is occurring, each researcher has to decide on their own how high a deviance ratio is a problem.</p>
<p></p>
<p>The deviance values are found in <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> at the bottom of the table</p>
<div class="cell">
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glm_model_object</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = response ~ explanatory, family = binomial(link = "logit"), 
    data = your_data)

Deviance Residuals: 
     Min        1Q    Median        3Q       Max  
-1.98905  -1.04140   0.00328   1.01222   1.75292  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -1.59083    0.52309  -3.041  0.00236 ** 
explanatory  0.20329    0.06106   3.329  0.00087 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 138.63  on 99  degrees of freedom
Residual deviance: 124.79  on 98  degrees of freedom
AIC: 128.79

Number of Fisher Scoring iterations: 4</code></pre>
</div>
</div>
<p></p>
<p>For the example above the <strong>Deviance ratio = 124.79/98 = 1.2733673.</strong></p>
<p>This value is a bit above 1, but much lower than 2, so the dispersion assumption seems to be met in the example.</p>
<p></p>
</section><section id="worked-example-4" class="level3" data-number="8.7.2"><h3 data-number="8.7.2" class="anchored" data-anchor-id="worked-example-4">
<span class="header-section-number">8.7.2</span> Worked example</h3>
<p></p>
<p>Using the theory covered in the previous section, we can now check our sheepModel to ensure that it meets the assumptions of a Binomial GLM.</p>
<p></p>
<p>First, we will look again at <strong>the plot of the model predictions and the data together.</strong></p>
<p></p>
<div class="cell">
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sheep_predictions_response</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Model predictions on response scale"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep_predictions_plot2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep_predictions_plot2-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.10: Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>The fit of this model is perhaps a little harder than the example in the theory section. This is partly caused because there is more data, so it overlaps a lot.</p>
<div class="cell">
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_data</span>, 
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">Survival</span><span class="op">)</span>, 
              colour <span class="op">=</span> <span class="st">"grey70"</span>, size <span class="op">=</span> <span class="fl">2</span>, height <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_ribbon</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_predictions</span>,
              <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>,
                  ymin <span class="op">=</span> <span class="va">lower_ci_response</span>, 
                  ymax <span class="op">=</span> <span class="va">upper_ci_response</span><span class="op">)</span>, 
              alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sheep_predictions</span>, 
            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Weight</span>, y <span class="op">=</span> <span class="va">predictions_response</span><span class="op">)</span>, 
            colour <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Model predictions on response scale"</span>,
       y <span class="op">=</span> <span class="st">"Probability of survival"</span>,
       x <span class="op">=</span> <span class="st">"Body weight (kg)"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep_predictions_plot3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep_predictions_plot3-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.11: Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>If we increase the jitter of the points from 0.01 to 0.1, it becomes a bit easier to see the pattern. Now it is possible to see there are many more data points with a response value of 1, than there are data points with a response value of 0. There are especially few response values of 0 (death) for sheep with a body weight above 25kg. Very few of these big sheep die.</p>
<p>So, we would expect a model line to fit well, if it captures small sheep having a lower survival change than big sheep. It seems that the model does capture the data quite well in this case.</p>
<p></p>
<p>Next, we should <strong>check overdispersion.</strong></p>
<div class="cell">
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">sheep_model</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = Survival ~ Weight, family = binomial(link = "logit"), 
    data = sheep_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.7799   0.2793   0.4854   0.7191   1.6256  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -2.11971    0.25498  -8.313   &lt;2e-16 ***
Weight       0.18178    0.01434  12.680   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1445.6  on 1327  degrees of freedom
Residual deviance: 1251.5  on 1326  degrees of freedom
AIC: 1255.5

Number of Fisher Scoring iterations: 4</code></pre>
</div>
</div>
<p>The deviance ratio for the sheep model is:</p>
<p>1251.5/1326 = 0.943816.</p>
<p>This is actually slightly below the expected dispersion of 1, but it is pretty close. So, it seems the dispersion assumption is met for the sheep model.</p>
<p></p>
<p>In the next section we will interpret our results.</p>
<p></p>
</section></section><section id="draw-conclusions" class="level2 tabset tabset-fade page-columns page-full" data-number="8.8"><h2 class="tabset tabset-fade anchored" data-number="8.8" data-anchor-id="draw-conclusions">
<span class="header-section-number">8.8</span> <i class="far fa-lightbulb"></i> Draw conclusions</h2>
<section id="theory-5" class="level3" data-number="8.8.1"><h3 data-number="8.8.1" class="anchored" data-anchor-id="theory-5">
<span class="header-section-number">8.8.1</span> Theory</h3>
<p></p>
<p>In the previous sections you learned how to run a Binomial GLM, what the parameters of the model mean, how to quantify uncertainty in the parameters, and how to check the assumptions of the model. Now, we can bring everything together to draw some conclusions.</p>
<p></p>
<p>There are several components required in drawing a conclusion:</p>
<ul>
<li>statement of the maximum likelihood estimate of the parameters of interest (including strength and direction).</li>
<li>statement of the uncertainty in the estimate</li>
<li>link the results to biology and the question asked</li>
<li>Discussion of next directions</li>
</ul>
<p></p>
</section><section id="worked-example-5" class="level3" data-number="8.8.2"><h3 data-number="8.8.2" class="anchored" data-anchor-id="worked-example-5">
<span class="header-section-number">8.8.2</span> Worked example</h3>
<p></p>
<p>This is the final section of our analysis of the data on survival of sheep. We will now bring together all of the results we have obtained and draw a conclusion following the same format as the in the theory section.</p>
<p>A reminder, we were asking: <strong>Does weight influence survival of Soay sheep?</strong></p>
<p></p>
<p>The maximum likelihood estimate of the relationship between weight and survival probability is 0.1817796. In other words, for every 1 kg increase in body weight, the log-odds of the survival of the sheep increases by 0.18.</p>
<p>On the response scale this increase is non-linear and is better shown with a graph.</p>
<div class="cell">
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span class="va">sheep_predictions_response</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Model predictions on response scale"</span><span class="op">)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-sheep_predictions_plot4" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="2.X_binomial_GLM_files/figure-html/fig-sheep_predictions_plot4-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8.12: Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p>
<p>Here it is clearer to see the increase in survival probability with body weight. The increase in survival is steepest for smaller sheep, levelling off after around 25kg at a 0.95 probability of survival in a given year for these big sheep.</p>
<p></p>
<p>When we look at the uncertainty in this estimate, we see the 95% confidence interval is 0.15, 0.21. The confidence interval limits are same sign, meaning that 0 is not included as a plausible value for the strength of the relationship. Therefore, we can conclude that weight has seems to have a positive impact on the survival of sheep.</p>
<p></p>
<p>In model checking, all the assumptions of the Binomial GLMs were met. The model was not overdispersed, the data are Binomial (coming from single trials) and the fitted model line captured the pattern of the data well.</p>
<p>It should be noted that there is also data on Age in the Soay sheep dataset. It is reasonable to assume that Age and Body weight have some relationship. Older sheep are likely to be bigger than new lambs. Therefore, it might be good to also include Age in the model possibly conducting model selection to decide if this variable is needed.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
What’s next?
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>
<strong>Poisson Generalized linear models</strong> for analyses when your response variable is not normally distributed.</li>
</ul>
</div>
</div>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Further reading
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
</section>

<!-- -->

<div class="no-row-height column-margin column-container"><section id="contributors" class="level3 unlisted unnumbered"><h3 class="unlisted unnumbered anchored" data-anchor-id="contributors">Contributors</h3>
<ul>
<li>Kwaku Peprah Adjei (lead)</li>
<li>Emily G. Simmonds</li>
<li>Bob O’Hara</li>
</ul></section></div></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
    } else {
      disableStylesheet(alternateStylesheets);
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./2.X_introduction_to_GLMs.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Introduction to generalised linear models</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./2.X_mle.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Maximum likelihood estimation”</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb46" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Binomial generalized linear models</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, eval=TRUE, include=FALSE}</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co"># add all packages that need loading</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fontawesome)</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="co"># source figure settings</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../StatisticsInR/Files/biostats_theme.R"</span>)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup-data, include=FALSE, echo=FALSE}</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a><span class="co"># load the sheep data</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>sheep_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../StatisticsInR/Files/sheep_data.csv"</span>)</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>sheep_data <span class="ot">&lt;-</span> sheep_data[<span class="fu">is.na</span>(sheep_data<span class="sc">$</span>Age)<span class="sc">==</span><span class="cn">FALSE</span>,]</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="co"># create some example data</span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2020</span>)</span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>your_data <span class="ot">&lt;-</span> example_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">explanatory =</span> <span class="fu">c</span>(<span class="fu">rnorm</span>(<span class="dv">50</span>, <span class="dv">5</span>, <span class="dv">3</span>), </span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>                                                    <span class="fu">rnorm</span>(<span class="dv">50</span>, <span class="dv">10</span>, <span class="dv">3</span>)),</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>                           <span class="at">response =</span> (<span class="fu">c</span>(<span class="fu">sample</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="dv">50</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>, </span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.2</span>)), </span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">sample</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="dv">50</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.8</span>)))),</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>                           <span class="at">group =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"a"</span>, <span class="st">"b"</span>, <span class="st">"c"</span>, <span class="st">"d"</span>, <span class="st">"e"</span>,</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">"f"</span>, <span class="st">"g"</span>, <span class="st">"h"</span>, <span class="st">"i"</span>, <span class="st">"j"</span>), <span class="at">each =</span> <span class="dv">10</span>))</span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>example_data2 <span class="ot">&lt;-</span> example_data <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(group, response) <span class="sc">%&gt;%</span> </span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"response"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">explanatory =</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">10</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">successes =</span> <span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="at">failures =</span> <span class="st">`</span><span class="at">1</span><span class="st">`</span>)</span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a><span class="fu">## Before you start</span></span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>Before reading this page, you should be comfortable with basic statistical </span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>theory, using R, continuous and categorical data, and **linear models**.</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>You should also review the basics of the Binomial distribution <span class="co">[</span><span class="ot">here</span><span class="co">]()</span>. EGS: ADD LINK.</span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a><span class="fu">## Introduction</span></span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>In this section, we will look at how we can use a Binomial Generalized Linear </span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>Model (GLM) </span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>to analyze data with a binary response and numeric explanatory variables.</span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a>On this page, we focus on one particular type of GLM, the Binomial GLM. The</span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>Binomial GLM fits a non-linear line to:</span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a>1) estimate a relationship between $X$ and $Y$, where $Y$ comes from binary </span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>trials and is</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a>therefore bounded between 0 and 1. </span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a>2) predict change in $Y$ from change in $X$. </span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a>But unlike a linear model, the Binomial GLM does not do this with a straight</span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a>line (on the scale of the data). The Binomial GLM fits a curved line bounded</span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a>between 0 and 1 on the y-axis. This is because the Binomial GLM (also called a </span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a>logistic regression - more on why later), </span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a>the response data are the result of binary trials e.g Yes or No, anything with </span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a>only two outcomes. But it can represent the result of single trials or many e.g.</span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a>did an individual survive or die (1 trial), or the proportion of survivors from a </span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a>population of 10 (10 trials). </span>
<span id="cb46-80"><a href="#cb46-80" aria-hidden="true" tabindex="-1"></a>The $Y$ values can only </span>
<span id="cb46-81"><a href="#cb46-81" aria-hidden="true" tabindex="-1"></a>take values between 0 or 1 (either 0 and 1 themselves or proportions e.g. 0.1).  </span>
<span id="cb46-82"><a href="#cb46-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-83"><a href="#cb46-83" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-nonlinear-plot, include = TRUE, echo = FALSE, fig.cap = "Example of a fitted Binomial GLM model. The estimated relationship is plotted on the scale of the data."}</span></span>
<span id="cb46-84"><a href="#cb46-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-85"><a href="#cb46-85" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> response), <span class="at">data =</span> example_data)<span class="sc">+</span></span>
<span id="cb46-86"><a href="#cb46-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>)<span class="sc">+</span></span>
<span id="cb46-87"><a href="#cb46-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span>stats<span class="sc">::</span>glm, <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span><span class="st">"binomial"</span>), </span>
<span id="cb46-88"><a href="#cb46-88" aria-hidden="true" tabindex="-1"></a>              <span class="at">se=</span><span class="cn">TRUE</span>)</span>
<span id="cb46-89"><a href="#cb46-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-90"><a href="#cb46-90" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-91"><a href="#cb46-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-92"><a href="#cb46-92" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-93"><a href="#cb46-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-94"><a href="#cb46-94" aria-hidden="true" tabindex="-1"></a>A large number of the models that are used in biological research are GLMs. </span>
<span id="cb46-95"><a href="#cb46-95" aria-hidden="true" tabindex="-1"></a>This</span>
<span id="cb46-96"><a href="#cb46-96" aria-hidden="true" tabindex="-1"></a>is because a lot of biological data would not meet the assumptions required for</span>
<span id="cb46-97"><a href="#cb46-97" aria-hidden="true" tabindex="-1"></a>a linear model, for example survival data, occupancy data, or presence of a </span>
<span id="cb46-98"><a href="#cb46-98" aria-hidden="true" tabindex="-1"></a>particular gene are all examples of binary responses. </span>
<span id="cb46-99"><a href="#cb46-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-100"><a href="#cb46-100" aria-hidden="true" tabindex="-1"></a>As a result of their wide usage, </span>
<span id="cb46-101"><a href="#cb46-101" aria-hidden="true" tabindex="-1"></a>GLMs are a key part of modern quantitative biology!</span>
<span id="cb46-102"><a href="#cb46-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-103"><a href="#cb46-103" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-104"><a href="#cb46-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-105"><a href="#cb46-105" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="far fa-question-circle"&gt;&lt;/i&gt; Which questions? </span></span>
<span id="cb46-106"><a href="#cb46-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-107"><a href="#cb46-107" aria-hidden="true" tabindex="-1"></a>Example questions that can be answered with the Binomial GLM:</span>
<span id="cb46-108"><a href="#cb46-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-109"><a href="#cb46-109" aria-hidden="true" tabindex="-1"></a>Inference</span>
<span id="cb46-110"><a href="#cb46-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-111"><a href="#cb46-111" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>How does body weight (kg) influence survival probability of sparrows?</span>
<span id="cb46-112"><a href="#cb46-112" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>How does forest cover (type) affect the occurrence probability</span>
<span id="cb46-113"><a href="#cb46-113" aria-hidden="true" tabindex="-1"></a> of a plant species?</span>
<span id="cb46-114"><a href="#cb46-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-115"><a href="#cb46-115" aria-hidden="true" tabindex="-1"></a>Prediction</span>
<span id="cb46-116"><a href="#cb46-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-117"><a href="#cb46-117" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>What is the mortality of beetles exposed to different concentrations of </span>
<span id="cb46-118"><a href="#cb46-118" aria-hidden="true" tabindex="-1"></a> carbon disulfide $CS_2$?</span>
<span id="cb46-119"><a href="#cb46-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-120"><a href="#cb46-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-121"><a href="#cb46-121" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-122"><a href="#cb46-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-123"><a href="#cb46-123" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="fas fa-table"&gt;&lt;/i&gt; Type of data {.tabset .tabset-fade}</span></span>
<span id="cb46-124"><a href="#cb46-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-125"><a href="#cb46-125" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-126"><a href="#cb46-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-127"><a href="#cb46-127" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-128"><a href="#cb46-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-129"><a href="#cb46-129" aria-hidden="true" tabindex="-1"></a>As mentioned above, in a Binomial GLM the response data are always the result of</span>
<span id="cb46-130"><a href="#cb46-130" aria-hidden="true" tabindex="-1"></a>binary trials. </span>
<span id="cb46-131"><a href="#cb46-131" aria-hidden="true" tabindex="-1"></a>The values of the response variable should have either only two outcomes or</span>
<span id="cb46-132"><a href="#cb46-132" aria-hidden="true" tabindex="-1"></a>be the result of many trials (a number of successes and number of failures or</span>
<span id="cb46-133"><a href="#cb46-133" aria-hidden="true" tabindex="-1"></a>a proportion). Examples of </span>
<span id="cb46-134"><a href="#cb46-134" aria-hidden="true" tabindex="-1"></a>this type of data would be:</span>
<span id="cb46-135"><a href="#cb46-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-136"><a href="#cb46-136" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Number that survived and number that died</span>
<span id="cb46-137"><a href="#cb46-137" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Present/absent</span>
<span id="cb46-138"><a href="#cb46-138" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Yes/no</span>
<span id="cb46-139"><a href="#cb46-139" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>On/off</span>
<span id="cb46-140"><a href="#cb46-140" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Proportion of seedlings that germinated</span>
<span id="cb46-141"><a href="#cb46-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-142"><a href="#cb46-142" aria-hidden="true" tabindex="-1"></a>Just as was stated for the linear models, </span>
<span id="cb46-143"><a href="#cb46-143" aria-hidden="true" tabindex="-1"></a>**always remember to check that the way your variables are classified in R </span>
<span id="cb46-144"><a href="#cb46-144" aria-hidden="true" tabindex="-1"></a>is the same as the format you expect.** </span>
<span id="cb46-145"><a href="#cb46-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-146"><a href="#cb46-146" aria-hidden="true" tabindex="-1"></a>In this case, there are several options that will work. The response variable </span>
<span id="cb46-147"><a href="#cb46-147" aria-hidden="true" tabindex="-1"></a>$Y$ can be stored as a factor, an integer, or as a continuous numeric variable. </span>
<span id="cb46-148"><a href="#cb46-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-149"><a href="#cb46-149" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-150"><a href="#cb46-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-151"><a href="#cb46-151" aria-hidden="true" tabindex="-1"></a>For this worked example, we will try to find out if/how body weight (kg)</span>
<span id="cb46-152"><a href="#cb46-152" aria-hidden="true" tabindex="-1"></a>influences the survival probability of Soay sheep. </span>
<span id="cb46-153"><a href="#cb46-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-154"><a href="#cb46-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep, echo=FALSE, eval=TRUE, fig.cap="Illustration of sheep by Emily G. Simmonds", fig.height=150}</span></span>
<span id="cb46-155"><a href="#cb46-155" aria-hidden="true" tabindex="-1"></a><span class="co">#include_graphics("./Figures/sheep.png")</span></span>
<span id="cb46-156"><a href="#cb46-156" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-157"><a href="#cb46-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-158"><a href="#cb46-158" aria-hidden="true" tabindex="-1"></a>Biologically, it is expected that smaller sheep would find it harder to survive</span>
<span id="cb46-159"><a href="#cb46-159" aria-hidden="true" tabindex="-1"></a>harsh winters as they have fewer reserves (such as body fat) than larger sheep. </span>
<span id="cb46-160"><a href="#cb46-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-161"><a href="#cb46-161" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Introduction to the data</span></span>
<span id="cb46-162"><a href="#cb46-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-163"><a href="#cb46-163" aria-hidden="true" tabindex="-1"></a>This data for this example are from 1986 to 1996 for a</span>
<span id="cb46-164"><a href="#cb46-164" aria-hidden="true" tabindex="-1"></a>population of Soay sheep from the St. Kilda islands in Scotland, UK.  </span>
<span id="cb46-165"><a href="#cb46-165" aria-hidden="true" tabindex="-1"></a>They have been monitored following a standardised protocol since 1985. </span>
<span id="cb46-166"><a href="#cb46-166" aria-hidden="true" tabindex="-1"></a>Each year the sheep</span>
<span id="cb46-167"><a href="#cb46-167" aria-hidden="true" tabindex="-1"></a>are caught, weighed, and the population size is counted. Information on the </span>
<span id="cb46-168"><a href="#cb46-168" aria-hidden="true" tabindex="-1"></a>project can be found <span class="co">[</span><span class="ot">here</span><span class="co">](http://soaysheep.biology.ed.ac.uk/)</span>. These data are</span>
<span id="cb46-169"><a href="#cb46-169" aria-hidden="true" tabindex="-1"></a>open source and can be found in the appendix of this <span class="co">[</span><span class="ot">paper</span><span class="co">](https://onlinelibrary.wiley.com/doi/full/10.1111/j.1600-0706.2012.00035.)</span> EGS: CITE COULSON 2012.</span>
<span id="cb46-170"><a href="#cb46-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-171"><a href="#cb46-171" aria-hidden="true" tabindex="-1"></a>The data consists of five variables: </span>
<span id="cb46-172"><a href="#cb46-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-173"><a href="#cb46-173" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Year - year of recording</span>
<span id="cb46-174"><a href="#cb46-174" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Age - age of the sheep in years</span>
<span id="cb46-175"><a href="#cb46-175" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Survival - whether the sheep survived until the next year or not</span>
<span id="cb46-176"><a href="#cb46-176" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Weight - weight in kg </span>
<span id="cb46-177"><a href="#cb46-177" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Population size (PopSize) - number of females counted that year</span>
<span id="cb46-178"><a href="#cb46-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-179"><a href="#cb46-179" aria-hidden="true" tabindex="-1"></a><span class="in">```{r show-data, include = TRUE, echo = TRUE, }</span></span>
<span id="cb46-180"><a href="#cb46-180" aria-hidden="true" tabindex="-1"></a>sheep_data</span>
<span id="cb46-181"><a href="#cb46-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-182"><a href="#cb46-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-183"><a href="#cb46-183" aria-hidden="true" tabindex="-1"></a>All measures are for female sheep only. </span>
<span id="cb46-184"><a href="#cb46-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-185"><a href="#cb46-185" aria-hidden="true" tabindex="-1"></a>You can find the data <span class="co">[</span><span class="ot">here</span><span class="co">]()</span> if you want to follow along with the example. </span>
<span id="cb46-186"><a href="#cb46-186" aria-hidden="true" tabindex="-1"></a>It is a <span class="in">`.csv`</span> file with column headings.</span>
<span id="cb46-187"><a href="#cb46-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-188"><a href="#cb46-188" aria-hidden="true" tabindex="-1"></a>To begin the analysis, let us first have a look at the data after it </span>
<span id="cb46-189"><a href="#cb46-189" aria-hidden="true" tabindex="-1"></a>has been imported. We do this by looking at the raw data and making a plot. It</span>
<span id="cb46-190"><a href="#cb46-190" aria-hidden="true" tabindex="-1"></a>can be easier to see the data if we jitter it because many data points sit on </span>
<span id="cb46-191"><a href="#cb46-191" aria-hidden="true" tabindex="-1"></a>top of each other. </span>
<span id="cb46-192"><a href="#cb46-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-193"><a href="#cb46-193" aria-hidden="true" tabindex="-1"></a>EGS: NEED TO ADD LINE TO READ IN DATA YOU CAN GET IT FROM THE PAPER LINK ABOVE THEN HOST IT.</span>
<span id="cb46-194"><a href="#cb46-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-195"><a href="#cb46-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep-data-plot, include = TRUE, echo = TRUE, fig.cap = "Scatterplot of weight and against survival of sheep, plot a = raw data, plot b = jittered points"}</span></span>
<span id="cb46-196"><a href="#cb46-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-197"><a href="#cb46-197" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sheep_data, <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> Survival))<span class="sc">+</span></span>
<span id="cb46-198"><a href="#cb46-198" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"grey70"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.25</span>)<span class="sc">+</span></span>
<span id="cb46-199"><a href="#cb46-199" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Weight of sheep (kg)"</span>,</span>
<span id="cb46-200"><a href="#cb46-200" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Survival of sheep"</span>,</span>
<span id="cb46-201"><a href="#cb46-201" aria-hidden="true" tabindex="-1"></a>      <span class="at">tag =</span> <span class="st">"a"</span>)</span>
<span id="cb46-202"><a href="#cb46-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-203"><a href="#cb46-203" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sheep_data, <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> Survival))<span class="sc">+</span></span>
<span id="cb46-204"><a href="#cb46-204" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_jitter</span>(<span class="at">colour =</span> <span class="st">"grey70"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">height =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.25</span>)<span class="sc">+</span></span>
<span id="cb46-205"><a href="#cb46-205" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Weight of sheep (kg)"</span>,</span>
<span id="cb46-206"><a href="#cb46-206" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Survival of sheep"</span>,</span>
<span id="cb46-207"><a href="#cb46-207" aria-hidden="true" tabindex="-1"></a>      <span class="at">tag =</span> <span class="st">"b"</span>)</span>
<span id="cb46-208"><a href="#cb46-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-209"><a href="#cb46-209" aria-hidden="true" tabindex="-1"></a>plot1 <span class="sc">+</span> plot2 <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb46-210"><a href="#cb46-210" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-211"><a href="#cb46-211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-212"><a href="#cb46-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-213"><a href="#cb46-213" aria-hidden="true" tabindex="-1"></a>Since we are interested in </span>
<span id="cb46-214"><a href="#cb46-214" aria-hidden="true" tabindex="-1"></a>**whether weight (kg) influences the survival of sheep**, </span>
<span id="cb46-215"><a href="#cb46-215" aria-hidden="true" tabindex="-1"></a>we will need two of the five variables in the data: <span class="in">`Weight`</span> and <span class="in">`Survival`</span>. </span>
<span id="cb46-216"><a href="#cb46-216" aria-hidden="true" tabindex="-1"></a>We will not need <span class="in">`Year`</span>, <span class="in">`PopSize`</span> and <span class="in">`Age`</span>. </span>
<span id="cb46-217"><a href="#cb46-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-218"><a href="#cb46-218" aria-hidden="true" tabindex="-1"></a>The response variable then is <span class="in">`Survival`</span>, </span>
<span id="cb46-219"><a href="#cb46-219" aria-hidden="true" tabindex="-1"></a>which is a binary variable (it has two outcomes 0/1). </span>
<span id="cb46-220"><a href="#cb46-220" aria-hidden="true" tabindex="-1"></a>The explanatory variable is <span class="in">`Weight`</span>, which is continuous numeric. </span>
<span id="cb46-221"><a href="#cb46-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-222"><a href="#cb46-222" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-223"><a href="#cb46-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-224"><a href="#cb46-224" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="fas fa-project-diagram"&gt;&lt;/i&gt; Model details {.tabset .tabset-fade}</span></span>
<span id="cb46-225"><a href="#cb46-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-226"><a href="#cb46-226" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-227"><a href="#cb46-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-228"><a href="#cb46-228" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-229"><a href="#cb46-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-230"><a href="#cb46-230" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-231"><a href="#cb46-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-232"><a href="#cb46-232" aria-hidden="true" tabindex="-1"></a>A Binomial GLM is just like all other statistical models, it aims to represent</span>
<span id="cb46-233"><a href="#cb46-233" aria-hidden="true" tabindex="-1"></a>mathematically how the data that are being modelled were generated. </span>
<span id="cb46-234"><a href="#cb46-234" aria-hidden="true" tabindex="-1"></a>In this case,</span>
<span id="cb46-235"><a href="#cb46-235" aria-hidden="true" tabindex="-1"></a>it is assumed that the data were generated by a Binomial distribution and </span>
<span id="cb46-236"><a href="#cb46-236" aria-hidden="true" tabindex="-1"></a>influenced by a relationship with some explanatory variables. </span>
<span id="cb46-237"><a href="#cb46-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-238"><a href="#cb46-238" aria-hidden="true" tabindex="-1"></a>The sections below detail the three components of a GLM and describe what form </span>
<span id="cb46-239"><a href="#cb46-239" aria-hidden="true" tabindex="-1"></a>they take for the Binomial GLM: </span>
<span id="cb46-240"><a href="#cb46-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-241"><a href="#cb46-241" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>The <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>systematic part<span class="kw">&lt;/span&gt;</span>: $\alpha$ + $\beta$$X_i$</span>
<span id="cb46-242"><a href="#cb46-242" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>The <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:purple"</span><span class="kw">&gt;</span>random part<span class="kw">&lt;/span&gt;</span>: the distribution of the error</span>
<span id="cb46-243"><a href="#cb46-243" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>The <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:magenta"</span><span class="kw">&gt;</span>link function<span class="kw">&lt;/span&gt;</span>: that links them together</span>
<span id="cb46-244"><a href="#cb46-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-245"><a href="#cb46-245" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-246"><a href="#cb46-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-247"><a href="#cb46-247" aria-hidden="true" tabindex="-1"></a>In <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span> it looks like this:</span>
<span id="cb46-248"><a href="#cb46-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-249"><a href="#cb46-249" aria-hidden="true" tabindex="-1"></a><span class="in">`glm(y ~ x, data, family = binomial(link = "logit"))`</span></span>
<span id="cb46-250"><a href="#cb46-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-251"><a href="#cb46-251" aria-hidden="true" tabindex="-1"></a>there are more details below in the **fitting the model in &lt;i class="fab fa-r-project"&gt;&lt;/i&gt;** section. </span>
<span id="cb46-252"><a href="#cb46-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-253"><a href="#cb46-253" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-254"><a href="#cb46-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-255"><a href="#cb46-255" aria-hidden="true" tabindex="-1"></a>**The &lt;span style="color:orange"&gt;systematic part&lt;/span&gt;**</span>
<span id="cb46-256"><a href="#cb46-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-257"><a href="#cb46-257" aria-hidden="true" tabindex="-1"></a>This is the part of the model that is hopefully the most familiar, it is a</span>
<span id="cb46-258"><a href="#cb46-258" aria-hidden="true" tabindex="-1"></a>linear equation just like those used in linear models. </span>
<span id="cb46-259"><a href="#cb46-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-260"><a href="#cb46-260" aria-hidden="true" tabindex="-1"></a>This is also known as the linear predictor, $\eta$. </span>
<span id="cb46-261"><a href="#cb46-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-262"><a href="#cb46-262" aria-hidden="true" tabindex="-1"></a>An example is </span>
<span id="cb46-263"><a href="#cb46-263" aria-hidden="true" tabindex="-1"></a>$\eta$ = $\alpha + \beta_1 X_{1} + \beta_2 X_{2}$</span>
<span id="cb46-264"><a href="#cb46-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-265"><a href="#cb46-265" aria-hidden="true" tabindex="-1"></a>where $\alpha$ = an intercept, and the $\beta$ values represent the change in </span>
<span id="cb46-266"><a href="#cb46-266" aria-hidden="true" tabindex="-1"></a>$\eta$ with every unit change of $X$. </span>
<span id="cb46-267"><a href="#cb46-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-268"><a href="#cb46-268" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-269"><a href="#cb46-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-270"><a href="#cb46-270" aria-hidden="true" tabindex="-1"></a>In <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span> this is written as <span class="in">`y ~ x`</span>. </span>
<span id="cb46-271"><a href="#cb46-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-272"><a href="#cb46-272" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-273"><a href="#cb46-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-274"><a href="#cb46-274" aria-hidden="true" tabindex="-1"></a>**The &lt;span style="color:purple"&gt;random part&lt;/span&gt;**</span>
<span id="cb46-275"><a href="#cb46-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-276"><a href="#cb46-276" aria-hidden="true" tabindex="-1"></a>This part of the model represents the spread of the actual data points around</span>
<span id="cb46-277"><a href="#cb46-277" aria-hidden="true" tabindex="-1"></a>the linear predictor. </span>
<span id="cb46-278"><a href="#cb46-278" aria-hidden="true" tabindex="-1"></a>This random part is how we deal with the errors (residuals) of the model</span>
<span id="cb46-279"><a href="#cb46-279" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:red"</span><span class="kw">&gt;</span>$\varepsilon$<span class="kw">&lt;/span&gt;</span>. </span>
<span id="cb46-280"><a href="#cb46-280" aria-hidden="true" tabindex="-1"></a>In linear models, this part was assumed to be Normal. </span>
<span id="cb46-281"><a href="#cb46-281" aria-hidden="true" tabindex="-1"></a>In a Binomial GLM, it is assumed to be Binomially distributed (the clue is in </span>
<span id="cb46-282"><a href="#cb46-282" aria-hidden="true" tabindex="-1"></a>the name). </span>
<span id="cb46-283"><a href="#cb46-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-284"><a href="#cb46-284" aria-hidden="true" tabindex="-1"></a>The Binomial distribution is used to represent the number of </span>
<span id="cb46-285"><a href="#cb46-285" aria-hidden="true" tabindex="-1"></a>successes ($r$) from a number of independent trials ($N$) when</span>
<span id="cb46-286"><a href="#cb46-286" aria-hidden="true" tabindex="-1"></a>the probability of success ($p$) is the same for each trial. We</span>
<span id="cb46-287"><a href="#cb46-287" aria-hidden="true" tabindex="-1"></a>use it for data from binary trials.</span>
<span id="cb46-288"><a href="#cb46-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-289"><a href="#cb46-289" aria-hidden="true" tabindex="-1"></a>The Binomial distribution has two parameters $N$ and $p$. Usually,</span>
<span id="cb46-290"><a href="#cb46-290" aria-hidden="true" tabindex="-1"></a>$N$ is known (it comes from the data), </span>
<span id="cb46-291"><a href="#cb46-291" aria-hidden="true" tabindex="-1"></a>so there is only one unknown parameter that must be estimated.</span>
<span id="cb46-292"><a href="#cb46-292" aria-hidden="true" tabindex="-1"></a>That is $p$. </span>
<span id="cb46-293"><a href="#cb46-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-294"><a href="#cb46-294" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-295"><a href="#cb46-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-296"><a href="#cb46-296" aria-hidden="true" tabindex="-1"></a>In <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span> the random part of the GLM</span>
<span id="cb46-297"><a href="#cb46-297" aria-hidden="true" tabindex="-1"></a>is specified by the <span class="in">`family`</span> argument for a Binomial GLM it is </span>
<span id="cb46-298"><a href="#cb46-298" aria-hidden="true" tabindex="-1"></a><span class="in">`family = binomial`</span>. </span>
<span id="cb46-299"><a href="#cb46-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-300"><a href="#cb46-300" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-301"><a href="#cb46-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-302"><a href="#cb46-302" aria-hidden="true" tabindex="-1"></a>**The &lt;span style="color:magenta"&gt;Link function&lt;/span&gt;**</span>
<span id="cb46-303"><a href="#cb46-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-304"><a href="#cb46-304" aria-hidden="true" tabindex="-1"></a>The link function transforms the <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>systematic part<span class="kw">&lt;/span&gt;</span> </span>
<span id="cb46-305"><a href="#cb46-305" aria-hidden="true" tabindex="-1"></a>of the model onto the scale of data (called the response scale), connecting it to the </span>
<span id="cb46-306"><a href="#cb46-306" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:purple"</span><span class="kw">&gt;</span>random part<span class="kw">&lt;/span&gt;</span>. </span>
<span id="cb46-307"><a href="#cb46-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-308"><a href="#cb46-308" aria-hidden="true" tabindex="-1"></a>This step is necessary because the model is non-linear. If you were to directly</span>
<span id="cb46-309"><a href="#cb46-309" aria-hidden="true" tabindex="-1"></a>estimate $Y$ (the response data) with a linear predictor as you do in a linear</span>
<span id="cb46-310"><a href="#cb46-310" aria-hidden="true" tabindex="-1"></a>model, this would give a straight line. The straight line would not be bounded</span>
<span id="cb46-311"><a href="#cb46-311" aria-hidden="true" tabindex="-1"></a>and could therefore predict values of $Y$ above 1 or below 0, which does not</span>
<span id="cb46-312"><a href="#cb46-312" aria-hidden="true" tabindex="-1"></a>make any sense. </span>
<span id="cb46-313"><a href="#cb46-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-314"><a href="#cb46-314" aria-hidden="true" tabindex="-1"></a>Instead, the Binomial GLM links the estimate from the linear predictor to $Y$ </span>
<span id="cb46-315"><a href="#cb46-315" aria-hidden="true" tabindex="-1"></a>via a link function. </span>
<span id="cb46-316"><a href="#cb46-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-317"><a href="#cb46-317" aria-hidden="true" tabindex="-1"></a>The default link function, or canonical function, for a Binomial distribution</span>
<span id="cb46-318"><a href="#cb46-318" aria-hidden="true" tabindex="-1"></a>is the logit function (hence the alternative name of logistic regression). </span>
<span id="cb46-319"><a href="#cb46-319" aria-hidden="true" tabindex="-1"></a>So,</span>
<span id="cb46-320"><a href="#cb46-320" aria-hidden="true" tabindex="-1"></a>in a Binomial GLM the logit of the linear predictor is linked to $Y$ instead</span>
<span id="cb46-321"><a href="#cb46-321" aria-hidden="true" tabindex="-1"></a>of the linear predictor itself.</span>
<span id="cb46-322"><a href="#cb46-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-323"><a href="#cb46-323" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-324"><a href="#cb46-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-325"><a href="#cb46-325" aria-hidden="true" tabindex="-1"></a>In <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span> the link</span>
<span id="cb46-326"><a href="#cb46-326" aria-hidden="true" tabindex="-1"></a>is specified within the <span class="in">`family`</span> argument, but with an extra argument called</span>
<span id="cb46-327"><a href="#cb46-327" aria-hidden="true" tabindex="-1"></a><span class="in">`link`</span>. For a Binomial GLM with a logit link it is</span>
<span id="cb46-328"><a href="#cb46-328" aria-hidden="true" tabindex="-1"></a><span class="in">`family = binomial(link = logit)`</span>. </span>
<span id="cb46-329"><a href="#cb46-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-330"><a href="#cb46-330" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-331"><a href="#cb46-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-332"><a href="#cb46-332" aria-hidden="true" tabindex="-1"></a>Using the logit link converts a probability to a log-odds. The log-odds is the </span>
<span id="cb46-333"><a href="#cb46-333" aria-hidden="true" tabindex="-1"></a>natural logarithm of the odds ratio: $$\frac{p}{1-p}$$. Where $p$ = the </span>
<span id="cb46-334"><a href="#cb46-334" aria-hidden="true" tabindex="-1"></a>probability of an event occurring. </span>
<span id="cb46-335"><a href="#cb46-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-336"><a href="#cb46-336" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-337"><a href="#cb46-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-338"><a href="#cb46-338" aria-hidden="true" tabindex="-1"></a>The odds ratio</span>
<span id="cb46-339"><a href="#cb46-339" aria-hidden="true" tabindex="-1"></a>is the probability of a event occurring ($p$) over the probability it doesn't </span>
<span id="cb46-340"><a href="#cb46-340" aria-hidden="true" tabindex="-1"></a>occur ($1-p$). Odds are often used in the things like betting, e.g. to represent the </span>
<span id="cb46-341"><a href="#cb46-341" aria-hidden="true" tabindex="-1"></a>odds of a particular horse winning a race. An odds ratio of 10:1 means that the </span>
<span id="cb46-342"><a href="#cb46-342" aria-hidden="true" tabindex="-1"></a>companies assume that for every 10 successes (for the company i.e. a horse</span>
<span id="cb46-343"><a href="#cb46-343" aria-hidden="true" tabindex="-1"></a>loses and they keep the money) there will be 1 failure (the horse wins and the</span>
<span id="cb46-344"><a href="#cb46-344" aria-hidden="true" tabindex="-1"></a>company has to pay).</span>
<span id="cb46-345"><a href="#cb46-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-346"><a href="#cb46-346" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-347"><a href="#cb46-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-348"><a href="#cb46-348" aria-hidden="true" tabindex="-1"></a>The logit link takes the following form:</span>
<span id="cb46-349"><a href="#cb46-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-350"><a href="#cb46-350" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-351"><a href="#cb46-351" aria-hidden="true" tabindex="-1"></a>\begin{aligned}</span>
<span id="cb46-352"><a href="#cb46-352" aria-hidden="true" tabindex="-1"></a>\mu = log(\frac{p}{1-p})</span>
<span id="cb46-353"><a href="#cb46-353" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb46-354"><a href="#cb46-354" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-355"><a href="#cb46-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-356"><a href="#cb46-356" aria-hidden="true" tabindex="-1"></a>Where $\mu$ = the log-odds, which is the output of the model. </span>
<span id="cb46-357"><a href="#cb46-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-358"><a href="#cb46-358" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-359"><a href="#cb46-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-360"><a href="#cb46-360" aria-hidden="true" tabindex="-1"></a>The inverse of the logit link is:</span>
<span id="cb46-361"><a href="#cb46-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-362"><a href="#cb46-362" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-363"><a href="#cb46-363" aria-hidden="true" tabindex="-1"></a>\begin{aligned}</span>
<span id="cb46-364"><a href="#cb46-364" aria-hidden="true" tabindex="-1"></a>p = \frac{e^{\mu}}{1+e^{\mu}} </span>
<span id="cb46-365"><a href="#cb46-365" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb46-366"><a href="#cb46-366" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-367"><a href="#cb46-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-368"><a href="#cb46-368" aria-hidden="true" tabindex="-1"></a>OR</span>
<span id="cb46-369"><a href="#cb46-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-370"><a href="#cb46-370" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-371"><a href="#cb46-371" aria-hidden="true" tabindex="-1"></a>\begin{aligned}</span>
<span id="cb46-372"><a href="#cb46-372" aria-hidden="true" tabindex="-1"></a>p = \frac{1}{1+e^{-\mu}} </span>
<span id="cb46-373"><a href="#cb46-373" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb46-374"><a href="#cb46-374" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-375"><a href="#cb46-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-376"><a href="#cb46-376" aria-hidden="true" tabindex="-1"></a>Where $p$ = the probability of success.</span>
<span id="cb46-377"><a href="#cb46-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-378"><a href="#cb46-378" aria-hidden="true" tabindex="-1"></a>Here $\mu$ is on the link scale and $p$ is on the response scale (the scale of</span>
<span id="cb46-379"><a href="#cb46-379" aria-hidden="true" tabindex="-1"></a>the response variable). </span>
<span id="cb46-380"><a href="#cb46-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-381"><a href="#cb46-381" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-382"><a href="#cb46-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-383"><a href="#cb46-383" aria-hidden="true" tabindex="-1"></a>Other possible link functions for a Binomial GLM are the </span>
<span id="cb46-384"><a href="#cb46-384" aria-hidden="true" tabindex="-1"></a>**probit** and **cloglog** functions.</span>
<span id="cb46-385"><a href="#cb46-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-386"><a href="#cb46-386" aria-hidden="true" tabindex="-1"></a>For more information <span class="kw">&lt;details&gt;&lt;summary&gt;</span>click here<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb46-387"><a href="#cb46-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-388"><a href="#cb46-388" aria-hidden="true" tabindex="-1"></a>**The Probit link**</span>
<span id="cb46-389"><a href="#cb46-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-390"><a href="#cb46-390" aria-hidden="true" tabindex="-1"></a>One way of thinking of binomial problems is as a threshold:</span>
<span id="cb46-391"><a href="#cb46-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-392"><a href="#cb46-392" aria-hidden="true" tabindex="-1"></a>e.g. imagine you have a dam, and if the water is too high, it will flow over the dam. </span>
<span id="cb46-393"><a href="#cb46-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-394"><a href="#cb46-394" aria-hidden="true" tabindex="-1"></a>Therefore a model is needed that can capture when the dam overflows (it either does or does not). </span>
<span id="cb46-395"><a href="#cb46-395" aria-hidden="true" tabindex="-1"></a>What is observed, and modelled, is whether the water was too high, or not. </span>
<span id="cb46-396"><a href="#cb46-396" aria-hidden="true" tabindex="-1"></a>But, hidden underneath this is a variable of water height, which is not observed. </span>
<span id="cb46-397"><a href="#cb46-397" aria-hidden="true" tabindex="-1"></a>But it is water height that controls if the dam overflows or not. </span>
<span id="cb46-398"><a href="#cb46-398" aria-hidden="true" tabindex="-1"></a>An unobserved variable like this is called a **latent variable**.</span>
<span id="cb46-399"><a href="#cb46-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-400"><a href="#cb46-400" aria-hidden="true" tabindex="-1"></a>This idea can be used in the modeling. </span>
<span id="cb46-401"><a href="#cb46-401" aria-hidden="true" tabindex="-1"></a>If the water height was observed, it could be modelled with a simple linear regression </span>
<span id="cb46-402"><a href="#cb46-402" aria-hidden="true" tabindex="-1"></a>of water height against rainfall, and assume the residuals are normally distributed. </span>
<span id="cb46-403"><a href="#cb46-403" aria-hidden="true" tabindex="-1"></a>But, if only data on whether this value is above a certain threshold, or not is available. </span>
<span id="cb46-404"><a href="#cb46-404" aria-hidden="true" tabindex="-1"></a>It turns out that this is the same as a Binomial GLM with a probit link!</span>
<span id="cb46-405"><a href="#cb46-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-406"><a href="#cb46-406" aria-hidden="true" tabindex="-1"></a>Mathematically the model is $Y_i = 1$ if $\mu_i &gt; 0$, where $\mu_i$ is the latent variable. </span>
<span id="cb46-407"><a href="#cb46-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-408"><a href="#cb46-408" aria-hidden="true" tabindex="-1"></a>The threshold idea can be useful for interpreting models: </span>
<span id="cb46-409"><a href="#cb46-409" aria-hidden="true" tabindex="-1"></a>if you think there is some unobserved variable that causes the binary response when it is above a threshold, </span>
<span id="cb46-410"><a href="#cb46-410" aria-hidden="true" tabindex="-1"></a>it can be easier to understand the process. </span>
<span id="cb46-411"><a href="#cb46-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-412"><a href="#cb46-412" aria-hidden="true" tabindex="-1"></a>In practice the estimates from the probit and logit link functions give almost the same results, </span>
<span id="cb46-413"><a href="#cb46-413" aria-hidden="true" tabindex="-1"></a>even though they have different interpretations.</span>
<span id="cb46-414"><a href="#cb46-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-415"><a href="#cb46-415" aria-hidden="true" tabindex="-1"></a>The probit uses an inverse normal link function where a higher mean =</span>
<span id="cb46-416"><a href="#cb46-416" aria-hidden="true" tabindex="-1"></a>higher probability of success.</span>
<span id="cb46-417"><a href="#cb46-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-418"><a href="#cb46-418" aria-hidden="true" tabindex="-1"></a>**Use when you want a threshold model because you have an unobserved variable that causes a binary response.**</span>
<span id="cb46-419"><a href="#cb46-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-420"><a href="#cb46-420" aria-hidden="true" tabindex="-1"></a>**cloglog**</span>
<span id="cb46-421"><a href="#cb46-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-422"><a href="#cb46-422" aria-hidden="true" tabindex="-1"></a>Using a cloglog link allows binary data to be linked to count data.</span>
<span id="cb46-423"><a href="#cb46-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-424"><a href="#cb46-424" aria-hidden="true" tabindex="-1"></a>It is useful when the 0s and 1s actually come from counts, where the count is recorded as "0/zero" or "more than zero". </span>
<span id="cb46-425"><a href="#cb46-425" aria-hidden="true" tabindex="-1"></a>For example, the presence or absence of a species. </span>
<span id="cb46-426"><a href="#cb46-426" aria-hidden="true" tabindex="-1"></a>In this case, the presence or absence is really a result of counting abundance of</span>
<span id="cb46-427"><a href="#cb46-427" aria-hidden="true" tabindex="-1"></a>a species. </span>
<span id="cb46-428"><a href="#cb46-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-429"><a href="#cb46-429" aria-hidden="true" tabindex="-1"></a>The cloglog link allows these binary data to be linked to a log(abundance) </span>
<span id="cb46-430"><a href="#cb46-430" aria-hidden="true" tabindex="-1"></a>using the equations below.</span>
<span id="cb46-431"><a href="#cb46-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-432"><a href="#cb46-432" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-433"><a href="#cb46-433" aria-hidden="true" tabindex="-1"></a>log(\lambda) = log(-log(1-p))</span>
<span id="cb46-434"><a href="#cb46-434" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-435"><a href="#cb46-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-436"><a href="#cb46-436" aria-hidden="true" tabindex="-1"></a>Where $\lambda$ = the mean abundance and $p$ = probability of </span>
<span id="cb46-437"><a href="#cb46-437" aria-hidden="true" tabindex="-1"></a>presence. </span>
<span id="cb46-438"><a href="#cb46-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-439"><a href="#cb46-439" aria-hidden="true" tabindex="-1"></a>**Use when you want to link the binary data to abundance because they represent counts.**</span>
<span id="cb46-440"><a href="#cb46-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-441"><a href="#cb46-441" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb46-442"><a href="#cb46-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-443"><a href="#cb46-443" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-444"><a href="#cb46-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-445"><a href="#cb46-445" aria-hidden="true" tabindex="-1"></a>**Assumptions**</span>
<span id="cb46-446"><a href="#cb46-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-447"><a href="#cb46-447" aria-hidden="true" tabindex="-1"></a>There are several assumptions that should be met for the Binomial GLM to be </span>
<span id="cb46-448"><a href="#cb46-448" aria-hidden="true" tabindex="-1"></a>valid.</span>
<span id="cb46-449"><a href="#cb46-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-450"><a href="#cb46-450" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>There are no outliers</span>
<span id="cb46-451"><a href="#cb46-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-452"><a href="#cb46-452" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Each value of Y is independent</span>
<span id="cb46-453"><a href="#cb46-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-454"><a href="#cb46-454" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>The dispersion parameter is constant</span>
<span id="cb46-455"><a href="#cb46-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-456"><a href="#cb46-456" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>The correct variance function is used (in a Binomial GLM this is assumed</span>
<span id="cb46-457"><a href="#cb46-457" aria-hidden="true" tabindex="-1"></a> to be controlled by the mean)</span>
<span id="cb46-458"><a href="#cb46-458" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-459"><a href="#cb46-459" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>The correct distribution is used (Binomial here - it is assumed that the </span>
<span id="cb46-460"><a href="#cb46-460" aria-hidden="true" tabindex="-1"></a> random part of the model does follow a Binomial distribution)</span>
<span id="cb46-461"><a href="#cb46-461" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-462"><a href="#cb46-462" aria-hidden="true" tabindex="-1"></a> <span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-463"><a href="#cb46-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-464"><a href="#cb46-464" aria-hidden="true" tabindex="-1"></a>All of these assumptions should be met for the model to work properly and </span>
<span id="cb46-465"><a href="#cb46-465" aria-hidden="true" tabindex="-1"></a>they ALWAYS need to be checked. Outliers, dispersion, and variance can be </span>
<span id="cb46-466"><a href="#cb46-466" aria-hidden="true" tabindex="-1"></a>checked once the model has been fitted. </span>
<span id="cb46-467"><a href="#cb46-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-468"><a href="#cb46-468" aria-hidden="true" tabindex="-1"></a>Independence of Y should be ensured during data collection and the correct</span>
<span id="cb46-469"><a href="#cb46-469" aria-hidden="true" tabindex="-1"></a>distribution is determined by the characteristics of the data.</span>
<span id="cb46-470"><a href="#cb46-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-471"><a href="#cb46-471" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-472"><a href="#cb46-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-473"><a href="#cb46-473" aria-hidden="true" tabindex="-1"></a>**Writing the model in &lt;i class="fab fa-r-project"&gt;&lt;/i&gt;**</span>
<span id="cb46-474"><a href="#cb46-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-475"><a href="#cb46-475" aria-hidden="true" tabindex="-1"></a>To fit the model in <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span>, </span>
<span id="cb46-476"><a href="#cb46-476" aria-hidden="true" tabindex="-1"></a>we will use the <span class="in">`glm()`</span> function. </span>
<span id="cb46-477"><a href="#cb46-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-478"><a href="#cb46-478" aria-hidden="true" tabindex="-1"></a><span class="in">`glm(y ~ x, data, family = binomial(link = "logit"))`</span></span>
<span id="cb46-479"><a href="#cb46-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-480"><a href="#cb46-480" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-481"><a href="#cb46-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-482"><a href="#cb46-482" aria-hidden="true" tabindex="-1"></a>The <span class="in">`glm()`</span> function takes several arguments:</span>
<span id="cb46-483"><a href="#cb46-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-484"><a href="#cb46-484" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>formula in form: <span class="in">`y ~ x`</span> (<span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>systematic part<span class="kw">&lt;/span&gt;</span>)</span>
<span id="cb46-485"><a href="#cb46-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-486"><a href="#cb46-486" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>data: your data object.</span>
<span id="cb46-487"><a href="#cb46-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-488"><a href="#cb46-488" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>family: specifies the distribution used for the </span>
<span id="cb46-489"><a href="#cb46-489" aria-hidden="true" tabindex="-1"></a> <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:purple"</span><span class="kw">&gt;</span>random part<span class="kw">&lt;/span&gt;</span> of the model and the </span>
<span id="cb46-490"><a href="#cb46-490" aria-hidden="true" tabindex="-1"></a> <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:magenta"</span><span class="kw">&gt;</span>link function<span class="kw">&lt;/span&gt;</span>. </span>
<span id="cb46-491"><a href="#cb46-491" aria-hidden="true" tabindex="-1"></a> In this case, </span>
<span id="cb46-492"><a href="#cb46-492" aria-hidden="true" tabindex="-1"></a> the model uses the binomial family with a logit link. </span>
<span id="cb46-493"><a href="#cb46-493" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-494"><a href="#cb46-494" aria-hidden="true" tabindex="-1"></a> <span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-495"><a href="#cb46-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-496"><a href="#cb46-496" aria-hidden="true" tabindex="-1"></a>The function will fit the GLM using **maximum likelihood estimation**</span>
<span id="cb46-497"><a href="#cb46-497" aria-hidden="true" tabindex="-1"></a>and gives us the maximum likelihood estimates of </span>
<span id="cb46-498"><a href="#cb46-498" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>$\alpha$<span class="kw">&lt;/span&gt;</span> and</span>
<span id="cb46-499"><a href="#cb46-499" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:blue"</span><span class="kw">&gt;</span>$\beta_is$<span class="kw">&lt;/span&gt;</span> as an output. </span>
<span id="cb46-500"><a href="#cb46-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-501"><a href="#cb46-501" aria-hidden="true" tabindex="-1"></a>The formula part <span class="in">`y ~ x`</span>, is the same as in the function <span class="in">`lm()`</span>.</span>
<span id="cb46-502"><a href="#cb46-502" aria-hidden="true" tabindex="-1"></a>The $y$ is the response variable and $x$ is an explanatory variable. </span>
<span id="cb46-503"><a href="#cb46-503" aria-hidden="true" tabindex="-1"></a>To run the GLM in <span class="kw">&lt;i</span> <span class="er">class</span><span class="ot">=</span><span class="st">"fab fa-r-project"</span><span class="kw">&gt;&lt;/i&gt;</span> first it is necessary to</span>
<span id="cb46-504"><a href="#cb46-504" aria-hidden="true" tabindex="-1"></a>identify which variable is the response and which is/are </span>
<span id="cb46-505"><a href="#cb46-505" aria-hidden="true" tabindex="-1"></a>explanatory.</span>
<span id="cb46-506"><a href="#cb46-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-507"><a href="#cb46-507" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-508"><a href="#cb46-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-509"><a href="#cb46-509" aria-hidden="true" tabindex="-1"></a>There are two ways of fitting a Binomial GLM using the <span class="in">`glm()`</span> function:</span>
<span id="cb46-510"><a href="#cb46-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-511"><a href="#cb46-511" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Option 1: fitting response as a single factor (for single trials and proportions).</span>
<span id="cb46-512"><a href="#cb46-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-513"><a href="#cb46-513" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Option 2: fitting response as two columns (for multiple trials).</span>
<span id="cb46-514"><a href="#cb46-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-515"><a href="#cb46-515" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-516"><a href="#cb46-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-517"><a href="#cb46-517" aria-hidden="true" tabindex="-1"></a>**For data with results of single trials (0s and 1s) or proportions (0-1)**</span>
<span id="cb46-518"><a href="#cb46-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-519"><a href="#cb46-519" aria-hidden="true" tabindex="-1"></a>The $y$ is the response variable and is a single vector of numbers.</span>
<span id="cb46-520"><a href="#cb46-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-521"><a href="#cb46-521" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-522"><a href="#cb46-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-523"><a href="#cb46-523" aria-hidden="true" tabindex="-1"></a>**For data with results of multiple trials (number of successes and number</span>
<span id="cb46-524"><a href="#cb46-524" aria-hidden="true" tabindex="-1"></a>of failures)**</span>
<span id="cb46-525"><a href="#cb46-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-526"><a href="#cb46-526" aria-hidden="true" tabindex="-1"></a>The $y$ is still the response variable but instead of single column of 0s and 1s </span>
<span id="cb46-527"><a href="#cb46-527" aria-hidden="true" tabindex="-1"></a>or proportions, there are two columns (one of successes and one of failures). The</span>
<span id="cb46-528"><a href="#cb46-528" aria-hidden="true" tabindex="-1"></a>columns are combined using <span class="in">`cbind()`</span> within the <span class="in">`glm()`</span> function:</span>
<span id="cb46-529"><a href="#cb46-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-530"><a href="#cb46-530" aria-hidden="true" tabindex="-1"></a>`glm(cbind(success, failure) ~ x, data = data, </span>
<span id="cb46-531"><a href="#cb46-531" aria-hidden="true" tabindex="-1"></a>family = binomial(link = "logit"))`</span>
<span id="cb46-532"><a href="#cb46-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-533"><a href="#cb46-533" aria-hidden="true" tabindex="-1"></a>Doing this accounts for number of trials because <span class="in">`glm()`</span> will also get information</span>
<span id="cb46-534"><a href="#cb46-534" aria-hidden="true" tabindex="-1"></a>on the number of trials conducted (successes + failures).</span>
<span id="cb46-535"><a href="#cb46-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-536"><a href="#cb46-536" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-537"><a href="#cb46-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-538"><a href="#cb46-538" aria-hidden="true" tabindex="-1"></a>Once you have an identified response and explanatory variable/variables, you</span>
<span id="cb46-539"><a href="#cb46-539" aria-hidden="true" tabindex="-1"></a>can then plug these variables into the <span class="in">`glm()`</span> function in the below format </span>
<span id="cb46-540"><a href="#cb46-540" aria-hidden="true" tabindex="-1"></a>using the column names in place of <span class="in">`response`</span> and <span class="in">`explanatory`</span> and </span>
<span id="cb46-541"><a href="#cb46-541" aria-hidden="true" tabindex="-1"></a>including your data frame name as the data argument. </span>
<span id="cb46-542"><a href="#cb46-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-543"><a href="#cb46-543" aria-hidden="true" tabindex="-1"></a>Both modes of fitting are shown below. </span>
<span id="cb46-544"><a href="#cb46-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-545"><a href="#cb46-545" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-546"><a href="#cb46-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-547"><a href="#cb46-547" aria-hidden="true" tabindex="-1"></a><span class="in">```{r model-example, include=TRUE, echo=TRUE, eval=TRUE}</span></span>
<span id="cb46-548"><a href="#cb46-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-549"><a href="#cb46-549" aria-hidden="true" tabindex="-1"></a><span class="do">### single trial version</span></span>
<span id="cb46-550"><a href="#cb46-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-551"><a href="#cb46-551" aria-hidden="true" tabindex="-1"></a>glm_model_object <span class="ot">&lt;-</span> <span class="fu">glm</span>(response <span class="sc">~</span> explanatory, <span class="at">data =</span> your_data,</span>
<span id="cb46-552"><a href="#cb46-552" aria-hidden="true" tabindex="-1"></a>                     <span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">"logit"</span>))</span>
<span id="cb46-553"><a href="#cb46-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-554"><a href="#cb46-554" aria-hidden="true" tabindex="-1"></a><span class="do">### multiple trials version</span></span>
<span id="cb46-555"><a href="#cb46-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-556"><a href="#cb46-556" aria-hidden="true" tabindex="-1"></a>glm_multiple_trials <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(successes, failures) <span class="sc">~</span> explanatory, </span>
<span id="cb46-557"><a href="#cb46-557" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> example_data2,</span>
<span id="cb46-558"><a href="#cb46-558" aria-hidden="true" tabindex="-1"></a>                     <span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">"logit"</span>))</span>
<span id="cb46-559"><a href="#cb46-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-560"><a href="#cb46-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-561"><a href="#cb46-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-562"><a href="#cb46-562" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-563"><a href="#cb46-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-564"><a href="#cb46-564" aria-hidden="true" tabindex="-1"></a>Running the <span class="in">`glm()`</span> as illustrated above runs the GLM </span>
<span id="cb46-565"><a href="#cb46-565" aria-hidden="true" tabindex="-1"></a>and saves the output as a <span class="in">`glm_model_object`</span>.</span>
<span id="cb46-566"><a href="#cb46-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-567"><a href="#cb46-567" aria-hidden="true" tabindex="-1"></a>We can then view our results from the model object by using the function <span class="in">`coef()`</span>. </span>
<span id="cb46-568"><a href="#cb46-568" aria-hidden="true" tabindex="-1"></a>This will take the output of the <span class="in">`glm()`</span>, the model object, </span>
<span id="cb46-569"><a href="#cb46-569" aria-hidden="true" tabindex="-1"></a>as its argument and extracts the maximum likelihood estimates of </span>
<span id="cb46-570"><a href="#cb46-570" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>$\alpha$<span class="kw">&lt;/span&gt;</span> and </span>
<span id="cb46-571"><a href="#cb46-571" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:blue"</span><span class="kw">&gt;</span>$\beta$<span class="kw">&lt;/span&gt;</span>.</span>
<span id="cb46-572"><a href="#cb46-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-573"><a href="#cb46-573" aria-hidden="true" tabindex="-1"></a><span class="in">```{r coef-example, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-574"><a href="#cb46-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-575"><a href="#cb46-575" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(glm_model_object)</span>
<span id="cb46-576"><a href="#cb46-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-577"><a href="#cb46-577" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(glm_multiple_trials)</span>
<span id="cb46-578"><a href="#cb46-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-579"><a href="#cb46-579" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-580"><a href="#cb46-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-581"><a href="#cb46-581" aria-hidden="true" tabindex="-1"></a>You can see that the format of the output is the same regardless of whether it</span>
<span id="cb46-582"><a href="#cb46-582" aria-hidden="true" tabindex="-1"></a>is a Binomial GLM on a set of single trials, or multiple trials. </span>
<span id="cb46-583"><a href="#cb46-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-584"><a href="#cb46-584" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-585"><a href="#cb46-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-586"><a href="#cb46-586" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-587"><a href="#cb46-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-588"><a href="#cb46-588" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-589"><a href="#cb46-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-590"><a href="#cb46-590" aria-hidden="true" tabindex="-1"></a>This worked example demonstrates how to fit a GLM in $R$ </span>
<span id="cb46-591"><a href="#cb46-591" aria-hidden="true" tabindex="-1"></a>using the <span class="in">`glm()`</span> function for the Soay sheep data example. This is an example</span>
<span id="cb46-592"><a href="#cb46-592" aria-hidden="true" tabindex="-1"></a>of a single trial question. </span>
<span id="cb46-593"><a href="#cb46-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-594"><a href="#cb46-594" aria-hidden="true" tabindex="-1"></a>In this example, we are asking:</span>
<span id="cb46-595"><a href="#cb46-595" aria-hidden="true" tabindex="-1"></a>**Does body weight affect the survival probability of sheep?**</span>
<span id="cb46-596"><a href="#cb46-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-597"><a href="#cb46-597" aria-hidden="true" tabindex="-1"></a>The survival of the sheep (0 or 1) is the response ($Y$) and </span>
<span id="cb46-598"><a href="#cb46-598" aria-hidden="true" tabindex="-1"></a>body weight (kg) ($X$) is the explanatory variable. </span>
<span id="cb46-599"><a href="#cb46-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-600"><a href="#cb46-600" aria-hidden="true" tabindex="-1"></a>We put these variables in the <span class="in">`glm()`</span> function in the below format.</span>
<span id="cb46-601"><a href="#cb46-601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-602"><a href="#cb46-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r sheep-model, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-603"><a href="#cb46-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-604"><a href="#cb46-604" aria-hidden="true" tabindex="-1"></a>sheep_model <span class="ot">&lt;-</span> <span class="fu">glm</span>(Survival <span class="sc">~</span> Weight, <span class="at">data =</span> sheep_data, </span>
<span id="cb46-605"><a href="#cb46-605" aria-hidden="true" tabindex="-1"></a>                   <span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">"logit"</span>))</span>
<span id="cb46-606"><a href="#cb46-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-607"><a href="#cb46-607" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-608"><a href="#cb46-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-609"><a href="#cb46-609" aria-hidden="true" tabindex="-1"></a>We have run the model and have assigned it to an object name. A logit link was</span>
<span id="cb46-610"><a href="#cb46-610" aria-hidden="true" tabindex="-1"></a>used as these data do not come from a latent continuous variable or from counts, </span>
<span id="cb46-611"><a href="#cb46-611" aria-hidden="true" tabindex="-1"></a>so we can use the default/canonical link function here. </span>
<span id="cb46-612"><a href="#cb46-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-613"><a href="#cb46-613" aria-hidden="true" tabindex="-1"></a>Let us take a look at the maximum likelihood estimates of our model parameters (<span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>$\alpha$<span class="kw">&lt;/span&gt;</span>, </span>
<span id="cb46-614"><a href="#cb46-614" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:blue"</span><span class="kw">&gt;</span>$\beta$<span class="kw">&lt;/span&gt;</span>) </span>
<span id="cb46-615"><a href="#cb46-615" aria-hidden="true" tabindex="-1"></a>using the <span class="in">`coef()`</span>.</span>
<span id="cb46-616"><a href="#cb46-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-617"><a href="#cb46-617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r coef-sheep-model, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-618"><a href="#cb46-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-619"><a href="#cb46-619" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(sheep_model)</span>
<span id="cb46-620"><a href="#cb46-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-621"><a href="#cb46-621" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-622"><a href="#cb46-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-623"><a href="#cb46-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-624"><a href="#cb46-624" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="fas fa-laptop"&gt;&lt;/i&gt; Parameters {.tabset .tabset-fade}</span></span>
<span id="cb46-625"><a href="#cb46-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-626"><a href="#cb46-626" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-627"><a href="#cb46-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-628"><a href="#cb46-628" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-629"><a href="#cb46-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-630"><a href="#cb46-630" aria-hidden="true" tabindex="-1"></a>We introduced the model parameters of a Binomial GLM in the model theory</span>
<span id="cb46-631"><a href="#cb46-631" aria-hidden="true" tabindex="-1"></a>section above: </span>
<span id="cb46-632"><a href="#cb46-632" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:orange"</span><span class="kw">&gt;</span>$\alpha$<span class="kw">&lt;/span&gt;</span> = </span>
<span id="cb46-633"><a href="#cb46-633" aria-hidden="true" tabindex="-1"></a>the intercept; <span class="kw">&lt;span</span> <span class="er">style</span><span class="ot">=</span><span class="st">"color:blue"</span><span class="kw">&gt;</span>$\beta_i$<span class="kw">&lt;/span&gt;</span> = </span>
<span id="cb46-634"><a href="#cb46-634" aria-hidden="true" tabindex="-1"></a>the slope of the model line (steepness/gradient).</span>
<span id="cb46-635"><a href="#cb46-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-636"><a href="#cb46-636" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-637"><a href="#cb46-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-638"><a href="#cb46-638" aria-hidden="true" tabindex="-1"></a>**But what do these parameters really mean?**</span>
<span id="cb46-639"><a href="#cb46-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-640"><a href="#cb46-640" aria-hidden="true" tabindex="-1"></a>To fully understand the parameters and what they mean, we have to note that </span>
<span id="cb46-641"><a href="#cb46-641" aria-hidden="true" tabindex="-1"></a>there is a non-linear relationship between the explanatory and response </span>
<span id="cb46-642"><a href="#cb46-642" aria-hidden="true" tabindex="-1"></a>variable on the scale of the observed data. </span>
<span id="cb46-643"><a href="#cb46-643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-644"><a href="#cb46-644" aria-hidden="true" tabindex="-1"></a>To interpret the parameters, </span>
<span id="cb46-645"><a href="#cb46-645" aria-hidden="true" tabindex="-1"></a>we have to recall the linear predictor and the link function.</span>
<span id="cb46-646"><a href="#cb46-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-647"><a href="#cb46-647" aria-hidden="true" tabindex="-1"></a>The linear predictor is $\eta = \color{orange}\alpha + \color{blue}{\beta_i}X_i$. </span>
<span id="cb46-648"><a href="#cb46-648" aria-hidden="true" tabindex="-1"></a>The $\eta$ in this case is the log-odds of the probability of success ($p$).  </span>
<span id="cb46-649"><a href="#cb46-649" aria-hidden="true" tabindex="-1"></a>$p$ is what we are trying to estimate. </span>
<span id="cb46-650"><a href="#cb46-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-651"><a href="#cb46-651" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-652"><a href="#cb46-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-653"><a href="#cb46-653" aria-hidden="true" tabindex="-1"></a>That is:</span>
<span id="cb46-654"><a href="#cb46-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-655"><a href="#cb46-655" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-656"><a href="#cb46-656" aria-hidden="true" tabindex="-1"></a>log\bigg(\frac{p}{1-p}\bigg) = \eta = \color{orange}\alpha + \color{blue}{\beta}X_i.</span>
<span id="cb46-657"><a href="#cb46-657" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-658"><a href="#cb46-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-659"><a href="#cb46-659" aria-hidden="true" tabindex="-1"></a>You might recognise the first part of this formula from the introduction of the</span>
<span id="cb46-660"><a href="#cb46-660" aria-hidden="true" tabindex="-1"></a>logit link function above, it is the logit link. Note that on the scale of the</span>
<span id="cb46-661"><a href="#cb46-661" aria-hidden="true" tabindex="-1"></a>link, the model line will be straight. However, on the response scale will be </span>
<span id="cb46-662"><a href="#cb46-662" aria-hidden="true" tabindex="-1"></a>curved. </span>
<span id="cb46-663"><a href="#cb46-663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-664"><a href="#cb46-664" aria-hidden="true" tabindex="-1"></a>To relate the linear predictor back to the response scale of the data and $p$. </span>
<span id="cb46-665"><a href="#cb46-665" aria-hidden="true" tabindex="-1"></a>It is necessary to take the inverse of the link function like using </span>
<span id="cb46-666"><a href="#cb46-666" aria-hidden="true" tabindex="-1"></a>the below equation:</span>
<span id="cb46-667"><a href="#cb46-667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-668"><a href="#cb46-668" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-669"><a href="#cb46-669" aria-hidden="true" tabindex="-1"></a>p = \frac{1}{1 + e^{- \eta}} = \frac{1}{1 + e^{-(\color{orange}\alpha + \color{blue}{\beta}X_i)}}.</span>
<span id="cb46-670"><a href="#cb46-670" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-671"><a href="#cb46-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-672"><a href="#cb46-672" aria-hidden="true" tabindex="-1"></a>Therefore, it is possible to interpret the parameters on the response scale </span>
<span id="cb46-673"><a href="#cb46-673" aria-hidden="true" tabindex="-1"></a>or on the link scale. We will discuss both options here. </span>
<span id="cb46-674"><a href="#cb46-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-675"><a href="#cb46-675" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-676"><a href="#cb46-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-677"><a href="#cb46-677" aria-hidden="true" tabindex="-1"></a>**Parameters on the logit scale**</span>
<span id="cb46-678"><a href="#cb46-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-679"><a href="#cb46-679" aria-hidden="true" tabindex="-1"></a>On the logit scale, the response will be the log odds of $Y$. </span>
<span id="cb46-680"><a href="#cb46-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-681"><a href="#cb46-681" aria-hidden="true" tabindex="-1"></a>To get to the log odds there are a few steps.</span>
<span id="cb46-682"><a href="#cb46-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-683"><a href="#cb46-683" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>We need the probability of success: $\frac{1}{(1+10)}$ = $0.09$</span>
<span id="cb46-684"><a href="#cb46-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-685"><a href="#cb46-685" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Next we use that probability to find the odds:</span>
<span id="cb46-686"><a href="#cb46-686" aria-hidden="true" tabindex="-1"></a>$\frac{0.09}{(1-0.09)} = 0.0989$</span>
<span id="cb46-687"><a href="#cb46-687" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-688"><a href="#cb46-688" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>To get to the log odds, we take the log of 2:</span>
<span id="cb46-689"><a href="#cb46-689" aria-hidden="true" tabindex="-1"></a>$log(\frac{0.09}{(1-0.09)}) = -2.31$</span>
<span id="cb46-690"><a href="#cb46-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-691"><a href="#cb46-691" aria-hidden="true" tabindex="-1"></a>Step 3 should look familiar. It is the same as the link </span>
<span id="cb46-692"><a href="#cb46-692" aria-hidden="true" tabindex="-1"></a>function in the Binomial GLM. </span>
<span id="cb46-693"><a href="#cb46-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-694"><a href="#cb46-694" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-695"><a href="#cb46-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-696"><a href="#cb46-696" aria-hidden="true" tabindex="-1"></a>**&lt;span style="color:orange"&gt;$\alpha$, the intercept&lt;/span&gt;**</span>
<span id="cb46-697"><a href="#cb46-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-698"><a href="#cb46-698" aria-hidden="true" tabindex="-1"></a>This is the parameter that gives the value of the log odds of $Y$ when $X = 0$.</span>
<span id="cb46-699"><a href="#cb46-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-700"><a href="#cb46-700" aria-hidden="true" tabindex="-1"></a>**&lt;span style="color:blue"&gt;$\beta$, the slope&lt;/span&gt;**</span>
<span id="cb46-701"><a href="#cb46-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-702"><a href="#cb46-702" aria-hidden="true" tabindex="-1"></a>This gives the amount of change in the log odds of $Y$ as the value of $X$ </span>
<span id="cb46-703"><a href="#cb46-703" aria-hidden="true" tabindex="-1"></a>changes by one unit. </span>
<span id="cb46-704"><a href="#cb46-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-707"><a href="#cb46-707" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb46-708"><a href="#cb46-708" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-log-odds-plot</span></span>
<span id="cb46-709"><a href="#cb46-709" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: true</span></span>
<span id="cb46-710"><a href="#cb46-710" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb46-711"><a href="#cb46-711" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Example of a Binomial GLM model line on the logit scale. Intercept in orange and slope in blue"</span></span>
<span id="cb46-712"><a href="#cb46-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-713"><a href="#cb46-713" aria-hidden="true" tabindex="-1"></a>example_data <span class="ot">&lt;-</span> <span class="fu">mutate</span>(example_data,</span>
<span id="cb46-714"><a href="#cb46-714" aria-hidden="true" tabindex="-1"></a>                       <span class="at">logity =</span> <span class="fu">predict.glm</span>(glm_model_object, </span>
<span id="cb46-715"><a href="#cb46-715" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">newdata =</span> example_data,</span>
<span id="cb46-716"><a href="#cb46-716" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">type=</span><span class="st">"link"</span>))</span>
<span id="cb46-717"><a href="#cb46-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-718"><a href="#cb46-718" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_data, <span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> logity))<span class="sc">+</span></span>
<span id="cb46-719"><a href="#cb46-719" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"x"</span>,</span>
<span id="cb46-720"><a href="#cb46-720" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"log odds of y"</span>,</span>
<span id="cb46-721"><a href="#cb46-721" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="st">"Example of results of GLM on logit scale"</span>)<span class="sc">+</span></span>
<span id="cb46-722"><a href="#cb46-722" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> logity),</span>
<span id="cb46-723"><a href="#cb46-723" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-724"><a href="#cb46-724" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>],</span>
<span id="cb46-725"><a href="#cb46-725" aria-hidden="true" tabindex="-1"></a>                 <span class="at">colour =</span> <span class="st">"orange"</span>), <span class="at">size =</span> <span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb46-726"><a href="#cb46-726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb46-727"><a href="#cb46-727" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-728"><a href="#cb46-728" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-729"><a href="#cb46-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-730"><a href="#cb46-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-731"><a href="#cb46-731" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-732"><a href="#cb46-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-733"><a href="#cb46-733" aria-hidden="true" tabindex="-1"></a>**Parameters on the response scale**</span>
<span id="cb46-734"><a href="#cb46-734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-735"><a href="#cb46-735" aria-hidden="true" tabindex="-1"></a>On the response scale, the response will be $Y$, which is a probability of </span>
<span id="cb46-736"><a href="#cb46-736" aria-hidden="true" tabindex="-1"></a>success (e.g. survival, presence etc). </span>
<span id="cb46-737"><a href="#cb46-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-738"><a href="#cb46-738" aria-hidden="true" tabindex="-1"></a>**&lt;span style="color:orange"&gt;$\alpha$, the intercept&lt;/span&gt;**</span>
<span id="cb46-739"><a href="#cb46-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-740"><a href="#cb46-740" aria-hidden="true" tabindex="-1"></a>To find the expected value of $Y$ when $X = 0$. You need to convert the </span>
<span id="cb46-741"><a href="#cb46-741" aria-hidden="true" tabindex="-1"></a>estimate of $\alpha$ from the logit scale back to the response scale using the</span>
<span id="cb46-742"><a href="#cb46-742" aria-hidden="true" tabindex="-1"></a>equation for the inverse of the logit link: e.g.</span>
<span id="cb46-743"><a href="#cb46-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-744"><a href="#cb46-744" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-745"><a href="#cb46-745" aria-hidden="true" tabindex="-1"></a>p = \frac{1}{1+e^{-(\alpha)}}</span>
<span id="cb46-746"><a href="#cb46-746" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-747"><a href="#cb46-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-748"><a href="#cb46-748" aria-hidden="true" tabindex="-1"></a>This can be done in R.</span>
<span id="cb46-749"><a href="#cb46-749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-750"><a href="#cb46-750" aria-hidden="true" tabindex="-1"></a><span class="in">```{r intercept-convert, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-751"><a href="#cb46-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-752"><a href="#cb46-752" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate of the intercept on logit scale (log odds of y)</span></span>
<span id="cb46-753"><a href="#cb46-753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-754"><a href="#cb46-754" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>]</span>
<span id="cb46-755"><a href="#cb46-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-756"><a href="#cb46-756" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to response scale (a probability of success)</span></span>
<span id="cb46-757"><a href="#cb46-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-758"><a href="#cb46-758" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(<span class="sc">-</span>(<span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>])))</span>
<span id="cb46-759"><a href="#cb46-759" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-760"><a href="#cb46-760" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-761"><a href="#cb46-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-762"><a href="#cb46-762" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-763"><a href="#cb46-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-764"><a href="#cb46-764" aria-hidden="true" tabindex="-1"></a>**&lt;span style="color:blue"&gt;$\beta$, the slope&lt;/span&gt;**</span>
<span id="cb46-765"><a href="#cb46-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-766"><a href="#cb46-766" aria-hidden="true" tabindex="-1"></a>On the response scale the slope parameter has a somewhat different meaning. On</span>
<span id="cb46-767"><a href="#cb46-767" aria-hidden="true" tabindex="-1"></a>the logit scale, the slope captured a linear relationship. </span>
<span id="cb46-768"><a href="#cb46-768" aria-hidden="true" tabindex="-1"></a>However, on the </span>
<span id="cb46-769"><a href="#cb46-769" aria-hidden="true" tabindex="-1"></a>response scale, the relationship between $X$ and $Y$ is not linear. </span>
<span id="cb46-770"><a href="#cb46-770" aria-hidden="true" tabindex="-1"></a>As a result,</span>
<span id="cb46-771"><a href="#cb46-771" aria-hidden="true" tabindex="-1"></a>the gradient of the model line will be different for different $X$ values </span>
<span id="cb46-772"><a href="#cb46-772" aria-hidden="true" tabindex="-1"></a>(see @fig-response-scale-example). </span>
<span id="cb46-773"><a href="#cb46-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-776"><a href="#cb46-776" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb46-777"><a href="#cb46-777" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-response-scale-example</span></span>
<span id="cb46-778"><a href="#cb46-778" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: true</span></span>
<span id="cb46-779"><a href="#cb46-779" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb46-780"><a href="#cb46-780" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Example of a Binomial GLM model line on the scale of the response data. Intercept in orange and slope in blue"</span></span>
<span id="cb46-781"><a href="#cb46-781" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb46-782"><a href="#cb46-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-783"><a href="#cb46-783" aria-hidden="true" tabindex="-1"></a>example_data <span class="ot">&lt;-</span> <span class="fu">mutate</span>(example_data, </span>
<span id="cb46-784"><a href="#cb46-784" aria-hidden="true" tabindex="-1"></a>                       <span class="at">predictions =</span> <span class="fu">predict</span>(glm_model_object,</span>
<span id="cb46-785"><a href="#cb46-785" aria-hidden="true" tabindex="-1"></a>                                             example_data,</span>
<span id="cb46-786"><a href="#cb46-786" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">type =</span> <span class="st">"response"</span>))</span>
<span id="cb46-787"><a href="#cb46-787" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-788"><a href="#cb46-788" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(example_data, <span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> response))<span class="sc">+</span></span>
<span id="cb46-789"><a href="#cb46-789" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> response),</span>
<span id="cb46-790"><a href="#cb46-790" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"grey70"</span>, <span class="at">size =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb46-791"><a href="#cb46-791" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"x"</span>,</span>
<span id="cb46-792"><a href="#cb46-792" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"y"</span>,</span>
<span id="cb46-793"><a href="#cb46-793" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="st">"Example of results of GLM on response"</span>)<span class="sc">+</span></span>
<span id="cb46-794"><a href="#cb46-794" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> predictions),</span>
<span id="cb46-795"><a href="#cb46-795" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-796"><a href="#cb46-796" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>(<span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>]))),</span>
<span id="cb46-797"><a href="#cb46-797" aria-hidden="true" tabindex="-1"></a>                 <span class="at">colour =</span> <span class="st">"orange"</span>), <span class="at">size =</span> <span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb46-798"><a href="#cb46-798" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb46-799"><a href="#cb46-799" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-800"><a href="#cb46-800" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-801"><a href="#cb46-801" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-802"><a href="#cb46-802" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-803"><a href="#cb46-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-804"><a href="#cb46-804" aria-hidden="true" tabindex="-1"></a>The estimate of the $\beta$ coefficient on the logit scale, will tell you about</span>
<span id="cb46-805"><a href="#cb46-805" aria-hidden="true" tabindex="-1"></a>the strength and direction of the relationship between $X$ and $Y$. </span>
<span id="cb46-806"><a href="#cb46-806" aria-hidden="true" tabindex="-1"></a>This is the</span>
<span id="cb46-807"><a href="#cb46-807" aria-hidden="true" tabindex="-1"></a>case even though this relationship will not be linear on the response scale.</span>
<span id="cb46-808"><a href="#cb46-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-809"><a href="#cb46-809" aria-hidden="true" tabindex="-1"></a>To see the change in slope of the model line on the response scale you can</span>
<span id="cb46-810"><a href="#cb46-810" aria-hidden="true" tabindex="-1"></a>predict values of $Y$ on the response scale and either plot the relationship </span>
<span id="cb46-811"><a href="#cb46-811" aria-hidden="true" tabindex="-1"></a>(like above in @fig-response-scale-example) or</span>
<span id="cb46-812"><a href="#cb46-812" aria-hidden="true" tabindex="-1"></a>simply look at the change between two $X$ values. </span>
<span id="cb46-813"><a href="#cb46-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-814"><a href="#cb46-814" aria-hidden="true" tabindex="-1"></a>e.g.</span>
<span id="cb46-815"><a href="#cb46-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-816"><a href="#cb46-816" aria-hidden="true" tabindex="-1"></a><span class="in">```{r slope-convert, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-817"><a href="#cb46-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-818"><a href="#cb46-818" aria-hidden="true" tabindex="-1"></a><span class="co"># predict log odds of Y when X = 2 or X = 3</span></span>
<span id="cb46-819"><a href="#cb46-819" aria-hidden="true" tabindex="-1"></a><span class="co"># use a linear equation for this</span></span>
<span id="cb46-820"><a href="#cb46-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-821"><a href="#cb46-821" aria-hidden="true" tabindex="-1"></a>odds_y_x2 <span class="ot">&lt;-</span> <span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>] <span class="sc">+</span> (<span class="fu">coef</span>(glm_model_object)[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">2</span>)</span>
<span id="cb46-822"><a href="#cb46-822" aria-hidden="true" tabindex="-1"></a>odds_y_x3 <span class="ot">&lt;-</span> <span class="fu">coef</span>(glm_model_object)[<span class="dv">1</span>] <span class="sc">+</span> (<span class="fu">coef</span>(glm_model_object)[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">3</span>)</span>
<span id="cb46-823"><a href="#cb46-823" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-824"><a href="#cb46-824" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate of probability of success when X = 2</span></span>
<span id="cb46-825"><a href="#cb46-825" aria-hidden="true" tabindex="-1"></a><span class="co"># take inverse of the logit link</span></span>
<span id="cb46-826"><a href="#cb46-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-827"><a href="#cb46-827" aria-hidden="true" tabindex="-1"></a>prob_X2 <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(<span class="sc">-</span>odds_y_x2))</span>
<span id="cb46-828"><a href="#cb46-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-829"><a href="#cb46-829" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate of probability of success when X = 3</span></span>
<span id="cb46-830"><a href="#cb46-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-831"><a href="#cb46-831" aria-hidden="true" tabindex="-1"></a>prob_X3 <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(<span class="sc">-</span>odds_y_x3))</span>
<span id="cb46-832"><a href="#cb46-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-833"><a href="#cb46-833" aria-hidden="true" tabindex="-1"></a><span class="co"># slope from X = 2 to X = 3 on response scale</span></span>
<span id="cb46-834"><a href="#cb46-834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-835"><a href="#cb46-835" aria-hidden="true" tabindex="-1"></a>prob_X3 <span class="sc">-</span> prob_X2</span>
<span id="cb46-836"><a href="#cb46-836" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-837"><a href="#cb46-837" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-838"><a href="#cb46-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-839"><a href="#cb46-839" aria-hidden="true" tabindex="-1"></a>If you were to repeat the example above for $X$ = 10 and $X$ = 11, you would not</span>
<span id="cb46-840"><a href="#cb46-840" aria-hidden="true" tabindex="-1"></a>get the same slope on the response scale. </span>
<span id="cb46-841"><a href="#cb46-841" aria-hidden="true" tabindex="-1"></a>Even though it is the same on the logit scale. </span>
<span id="cb46-842"><a href="#cb46-842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-843"><a href="#cb46-843" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-844"><a href="#cb46-844" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-845"><a href="#cb46-845" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-846"><a href="#cb46-846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-847"><a href="#cb46-847" aria-hidden="true" tabindex="-1"></a>In the previous section, we fit a Binomial GLM using the <span class="in">`glm()`</span> function and </span>
<span id="cb46-848"><a href="#cb46-848" aria-hidden="true" tabindex="-1"></a>looked at the estimates of some parameters using the <span class="in">`coef()`</span> function. </span>
<span id="cb46-849"><a href="#cb46-849" aria-hidden="true" tabindex="-1"></a>In this section, we will use the model theory to interpret what those parameters</span>
<span id="cb46-850"><a href="#cb46-850" aria-hidden="true" tabindex="-1"></a>mean. </span>
<span id="cb46-851"><a href="#cb46-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-852"><a href="#cb46-852" aria-hidden="true" tabindex="-1"></a><span class="fu">#### &lt;span style="color:orange"&gt;The intercept&lt;/span&gt;</span></span>
<span id="cb46-853"><a href="#cb46-853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-854"><a href="#cb46-854" aria-hidden="true" tabindex="-1"></a>For our sheep data model, the estimate of the intercept is:</span>
<span id="cb46-855"><a href="#cb46-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-856"><a href="#cb46-856" aria-hidden="true" tabindex="-1"></a><span class="in">```{r coef-sheep-model2, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-857"><a href="#cb46-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-858"><a href="#cb46-858" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(sheep_model)[<span class="dv">1</span>]</span>
<span id="cb46-859"><a href="#cb46-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-860"><a href="#cb46-860" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-861"><a href="#cb46-861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-862"><a href="#cb46-862" aria-hidden="true" tabindex="-1"></a>This is the estimate of the log odds of survival for a sheep which has a body weight of 0 kg. </span>
<span id="cb46-863"><a href="#cb46-863" aria-hidden="true" tabindex="-1"></a>The intercept estimate is a log odds of <span class="in">`r coef(sheep_model)[1]`</span> of survival </span>
<span id="cb46-864"><a href="#cb46-864" aria-hidden="true" tabindex="-1"></a>for a sheep that weighs 0 kg. </span>
<span id="cb46-865"><a href="#cb46-865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-866"><a href="#cb46-866" aria-hidden="true" tabindex="-1"></a>To get the probability of survival for a sheep with a body</span>
<span id="cb46-867"><a href="#cb46-867" aria-hidden="true" tabindex="-1"></a>weight of 0 kg we need to convert these estimates to the response scale using</span>
<span id="cb46-868"><a href="#cb46-868" aria-hidden="true" tabindex="-1"></a>the code below:</span>
<span id="cb46-869"><a href="#cb46-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-870"><a href="#cb46-870" aria-hidden="true" tabindex="-1"></a><span class="in">```{r back-transform-sheep, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-871"><a href="#cb46-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-872"><a href="#cb46-872" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span><span class="fu">coef</span>(sheep_model)[<span class="dv">1</span>]))</span>
<span id="cb46-873"><a href="#cb46-873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-874"><a href="#cb46-874" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-875"><a href="#cb46-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-876"><a href="#cb46-876" aria-hidden="true" tabindex="-1"></a>In other words, </span>
<span id="cb46-877"><a href="#cb46-877" aria-hidden="true" tabindex="-1"></a>the survival probability of a sheep is <span class="in">`r 1/(1+exp(-coef(sheep_model)[1]))`</span> </span>
<span id="cb46-878"><a href="#cb46-878" aria-hidden="true" tabindex="-1"></a>when their weight is 0 kg. </span>
<span id="cb46-879"><a href="#cb46-879" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-880"><a href="#cb46-880" aria-hidden="true" tabindex="-1"></a>In this example, the intercept is not very interesting, </span>
<span id="cb46-881"><a href="#cb46-881" aria-hidden="true" tabindex="-1"></a>as it does not make a lot of biological sense to know the expected survival </span>
<span id="cb46-882"><a href="#cb46-882" aria-hidden="true" tabindex="-1"></a>probability of sheep when their weight is 0 kg. </span>
<span id="cb46-883"><a href="#cb46-883" aria-hidden="true" tabindex="-1"></a>A sheep born at anytime will weigh more than 0 kg. </span>
<span id="cb46-884"><a href="#cb46-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-885"><a href="#cb46-885" aria-hidden="true" tabindex="-1"></a><span class="fu">#### &lt;span style="color:blue"&gt;The slope&lt;/span&gt;</span></span>
<span id="cb46-886"><a href="#cb46-886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-887"><a href="#cb46-887" aria-hidden="true" tabindex="-1"></a>The estimate of the slope of the relationship between body weight and </span>
<span id="cb46-888"><a href="#cb46-888" aria-hidden="true" tabindex="-1"></a>survival probability is interesting to us since it can tell us the direction</span>
<span id="cb46-889"><a href="#cb46-889" aria-hidden="true" tabindex="-1"></a>and the strength of the effect of body weight on survival of the sheep. </span>
<span id="cb46-890"><a href="#cb46-890" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-891"><a href="#cb46-891" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-892"><a href="#cb46-892" aria-hidden="true" tabindex="-1"></a><span class="in">```{r coef-sheep-model3, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-893"><a href="#cb46-893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-894"><a href="#cb46-894" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(sheep_model)[<span class="dv">2</span>]</span>
<span id="cb46-895"><a href="#cb46-895" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-896"><a href="#cb46-896" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-897"><a href="#cb46-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-898"><a href="#cb46-898" aria-hidden="true" tabindex="-1"></a>In this case, our model estimates that the log odds of the survival of the sheep</span>
<span id="cb46-899"><a href="#cb46-899" aria-hidden="true" tabindex="-1"></a>increases by <span class="in">`r coef(sheep_model)[2]`</span> when the weight of the sheep increases by</span>
<span id="cb46-900"><a href="#cb46-900" aria-hidden="true" tabindex="-1"></a>1 kg. This shows a positive relationship between body weight and survival </span>
<span id="cb46-901"><a href="#cb46-901" aria-hidden="true" tabindex="-1"></a>probability. </span>
<span id="cb46-902"><a href="#cb46-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-903"><a href="#cb46-903" aria-hidden="true" tabindex="-1"></a>To see what this effect looks like on the response scale, it is easiest to</span>
<span id="cb46-904"><a href="#cb46-904" aria-hidden="true" tabindex="-1"></a>plot the results using predictions on the response scale (response scale). This</span>
<span id="cb46-905"><a href="#cb46-905" aria-hidden="true" tabindex="-1"></a>is what we do below. </span>
<span id="cb46-906"><a href="#cb46-906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-907"><a href="#cb46-907" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Plotting the Results </span></span>
<span id="cb46-908"><a href="#cb46-908" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-909"><a href="#cb46-909" aria-hidden="true" tabindex="-1"></a>As well as looking at the maximum likelihood estimates of the parameters from </span>
<span id="cb46-910"><a href="#cb46-910" aria-hidden="true" tabindex="-1"></a>the Binomial GLM, we can also plot the results. This is especially useful for</span>
<span id="cb46-911"><a href="#cb46-911" aria-hidden="true" tabindex="-1"></a>interpreting the results on the response data scale. </span>
<span id="cb46-912"><a href="#cb46-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-913"><a href="#cb46-913" aria-hidden="true" tabindex="-1"></a>To do this, we use the <span class="in">`ggplot()`</span> with <span class="in">`geom_line()`</span> and the <span class="in">`predict()`</span></span>
<span id="cb46-914"><a href="#cb46-914" aria-hidden="true" tabindex="-1"></a>function.</span>
<span id="cb46-915"><a href="#cb46-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-916"><a href="#cb46-916" aria-hidden="true" tabindex="-1"></a>To make the first plot, we will only need to use two arguments:</span>
<span id="cb46-917"><a href="#cb46-917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-918"><a href="#cb46-918" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span><span class="in">`object`</span> = your model object</span>
<span id="cb46-919"><a href="#cb46-919" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span><span class="in">`type`</span> = "link", which means predict on the link scale or</span>
<span id="cb46-920"><a href="#cb46-920" aria-hidden="true" tabindex="-1"></a> "response", which means predict on the response scale. </span>
<span id="cb46-921"><a href="#cb46-921" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-922"><a href="#cb46-922" aria-hidden="true" tabindex="-1"></a>We will make predictions on both the link and response scale so that we </span>
<span id="cb46-923"><a href="#cb46-923" aria-hidden="true" tabindex="-1"></a>can compare the results. </span>
<span id="cb46-924"><a href="#cb46-924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-925"><a href="#cb46-925" aria-hidden="true" tabindex="-1"></a><span class="in">```{r sheep-predictions1, include = TRUE, echo = TRUE}</span></span>
<span id="cb46-926"><a href="#cb46-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-927"><a href="#cb46-927" aria-hidden="true" tabindex="-1"></a>sheep_data <span class="ot">&lt;-</span> <span class="fu">mutate</span>(sheep_data, </span>
<span id="cb46-928"><a href="#cb46-928" aria-hidden="true" tabindex="-1"></a>                     <span class="at">predictions_link =</span> <span class="fu">predict</span>(sheep_model, </span>
<span id="cb46-929"><a href="#cb46-929" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">"link"</span>),</span>
<span id="cb46-930"><a href="#cb46-930" aria-hidden="true" tabindex="-1"></a>                     <span class="at">predictions_response =</span> <span class="fu">predict</span>(sheep_model, </span>
<span id="cb46-931"><a href="#cb46-931" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">"response"</span>))</span>
<span id="cb46-932"><a href="#cb46-932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-933"><a href="#cb46-933" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-934"><a href="#cb46-934" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-935"><a href="#cb46-935" aria-hidden="true" tabindex="-1"></a>Once we have created predictions of $Y$ from the model object, </span>
<span id="cb46-936"><a href="#cb46-936" aria-hidden="true" tabindex="-1"></a>we can plot these using <span class="in">`geom_line()`</span> as in the code below.</span>
<span id="cb46-937"><a href="#cb46-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-938"><a href="#cb46-938" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep-preditions1, include = TRUE, echo = TRUE, fig.cap = "a) Plot of fitted GLM line on the logit scale b) Plot of jittered raw data and fitted GLM line for the soay sheep data on the response scale"}</span></span>
<span id="cb46-939"><a href="#cb46-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-940"><a href="#cb46-940" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sheep_data, <span class="fu">aes</span>(<span class="at">x=</span>Weight, <span class="at">y=</span>predictions_link))<span class="sc">+</span></span>
<span id="cb46-941"><a href="#cb46-941" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>predictions_link), <span class="at">colour =</span> <span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-942"><a href="#cb46-942" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Weight (kg)"</span>, <span class="at">y =</span> <span class="st">"Log Odds of Survival"</span>)</span>
<span id="cb46-943"><a href="#cb46-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-944"><a href="#cb46-944" aria-hidden="true" tabindex="-1"></a>response <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sheep_data, <span class="fu">aes</span>(<span class="at">x=</span>Weight, <span class="at">y=</span>Survival))<span class="sc">+</span></span>
<span id="cb46-945"><a href="#cb46-945" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">colour =</span> <span class="st">"grey70"</span>, <span class="at">size =</span> <span class="dv">2</span>,</span>
<span id="cb46-946"><a href="#cb46-946" aria-hidden="true" tabindex="-1"></a>              <span class="at">height =</span> <span class="fl">0.05</span>)<span class="sc">+</span></span>
<span id="cb46-947"><a href="#cb46-947" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>predictions_response), <span class="at">colour =</span> <span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-948"><a href="#cb46-948" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Weight (kg)"</span>, <span class="at">y =</span><span class="st">"Survival Probability"</span>)</span>
<span id="cb46-949"><a href="#cb46-949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-950"><a href="#cb46-950" aria-hidden="true" tabindex="-1"></a>link <span class="sc">+</span> response <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb46-951"><a href="#cb46-951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-952"><a href="#cb46-952" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-953"><a href="#cb46-953" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-954"><a href="#cb46-954" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-955"><a href="#cb46-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-956"><a href="#cb46-956" aria-hidden="true" tabindex="-1"></a>In the next section, we will look at how to add uncertainty to these plots and </span>
<span id="cb46-957"><a href="#cb46-957" aria-hidden="true" tabindex="-1"></a>our interpretation. </span>
<span id="cb46-958"><a href="#cb46-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-959"><a href="#cb46-959" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-960"><a href="#cb46-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-961"><a href="#cb46-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-962"><a href="#cb46-962" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="fas fa-arrows-alt-h"&gt;&lt;/i&gt; Quantify uncertainty {.tabset .tabset-fade}</span></span>
<span id="cb46-963"><a href="#cb46-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-964"><a href="#cb46-964" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-965"><a href="#cb46-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-966"><a href="#cb46-966" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-967"><a href="#cb46-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-968"><a href="#cb46-968" aria-hidden="true" tabindex="-1"></a>As was discussed for linear models, </span>
<span id="cb46-969"><a href="#cb46-969" aria-hidden="true" tabindex="-1"></a>statistics does not give a single correct answer. </span>
<span id="cb46-970"><a href="#cb46-970" aria-hidden="true" tabindex="-1"></a>When we estimate the parameters in our statistical model, </span>
<span id="cb46-971"><a href="#cb46-971" aria-hidden="true" tabindex="-1"></a>there will be many plausible parameters that could have produced our </span>
<span id="cb46-972"><a href="#cb46-972" aria-hidden="true" tabindex="-1"></a>observed data. In these cases, some of the parameters will be more likely</span>
<span id="cb46-973"><a href="#cb46-973" aria-hidden="true" tabindex="-1"></a>than the others to have generated the observed data.</span>
<span id="cb46-974"><a href="#cb46-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-975"><a href="#cb46-975" aria-hidden="true" tabindex="-1"></a>The Binomial GLM is no exception. </span>
<span id="cb46-976"><a href="#cb46-976" aria-hidden="true" tabindex="-1"></a>We will have to consider and present the uncertainty in the parameters we </span>
<span id="cb46-977"><a href="#cb46-977" aria-hidden="true" tabindex="-1"></a>estimate.</span>
<span id="cb46-978"><a href="#cb46-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-979"><a href="#cb46-979" aria-hidden="true" tabindex="-1"></a>The <span class="in">`glm()`</span>function uses maximum likelihood estimation for the parameter </span>
<span id="cb46-980"><a href="#cb46-980" aria-hidden="true" tabindex="-1"></a>estimation. Therefore, our consideration of uncertainty for these models are </span>
<span id="cb46-981"><a href="#cb46-981" aria-hidden="true" tabindex="-1"></a>discussed <span class="co">[</span><span class="ot">here</span><span class="co">](link to ML page)</span>. </span>
<span id="cb46-982"><a href="#cb46-982" aria-hidden="true" tabindex="-1"></a>We will therefore quantify uncertainty using *standard errors*, </span>
<span id="cb46-983"><a href="#cb46-983" aria-hidden="true" tabindex="-1"></a>*confidence intervals* and *prediction intervals* </span>
<span id="cb46-984"><a href="#cb46-984" aria-hidden="true" tabindex="-1"></a>which should be familiar to you but head to the <span class="co">[</span><span class="ot">uncertainty</span><span class="co">]()</span> </span>
<span id="cb46-985"><a href="#cb46-985" aria-hidden="true" tabindex="-1"></a>pages if you need a recap.</span>
<span id="cb46-986"><a href="#cb46-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-987"><a href="#cb46-987" aria-hidden="true" tabindex="-1"></a>Just as with linear models there are two different types of uncertainty </span>
<span id="cb46-988"><a href="#cb46-988" aria-hidden="true" tabindex="-1"></a>we will look </span>
<span id="cb46-989"><a href="#cb46-989" aria-hidden="true" tabindex="-1"></a>at *uncertainty in the model line of parameters $\alpha$ and $\beta$* and </span>
<span id="cb46-990"><a href="#cb46-990" aria-hidden="true" tabindex="-1"></a>*uncertainty in a prediction of $Y$*.</span>
<span id="cb46-991"><a href="#cb46-991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-992"><a href="#cb46-992" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-993"><a href="#cb46-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-994"><a href="#cb46-994" aria-hidden="true" tabindex="-1"></a>**Uncertainty in the estimates of $\alpha$ and $\beta$**</span>
<span id="cb46-995"><a href="#cb46-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-996"><a href="#cb46-996" aria-hidden="true" tabindex="-1"></a>To find the standard errors for the estimates of $\alpha$ and $\beta$ </span>
<span id="cb46-997"><a href="#cb46-997" aria-hidden="true" tabindex="-1"></a>we can use <span class="in">`summary()`</span> function. </span>
<span id="cb46-998"><a href="#cb46-998" aria-hidden="true" tabindex="-1"></a>This takes the model object from the <span class="in">`glm()`</span> as its argument. </span>
<span id="cb46-999"><a href="#cb46-999" aria-hidden="true" tabindex="-1"></a>It outputs a big table with lots of information. </span>
<span id="cb46-1000"><a href="#cb46-1000" aria-hidden="true" tabindex="-1"></a>The first line gives the formula used for the model object. </span>
<span id="cb46-1001"><a href="#cb46-1001" aria-hidden="true" tabindex="-1"></a>The second line shows the summary of the residuals of the model and the </span>
<span id="cb46-1002"><a href="#cb46-1002" aria-hidden="true" tabindex="-1"></a>standard errors are shown in the second column of the third part, </span>
<span id="cb46-1003"><a href="#cb46-1003" aria-hidden="true" tabindex="-1"></a><span class="in">`Coefficients:`</span>.</span>
<span id="cb46-1004"><a href="#cb46-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1005"><a href="#cb46-1005" aria-hidden="true" tabindex="-1"></a><span class="in">```{r summary_glm_example}</span></span>
<span id="cb46-1006"><a href="#cb46-1006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1007"><a href="#cb46-1007" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_model_object)</span>
<span id="cb46-1008"><a href="#cb46-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1009"><a href="#cb46-1009" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1010"><a href="#cb46-1010" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1011"><a href="#cb46-1011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1012"><a href="#cb46-1012" aria-hidden="true" tabindex="-1"></a>By taking the <span class="in">`summary()`</span> of the example model, we can see the standard error of </span>
<span id="cb46-1013"><a href="#cb46-1013" aria-hidden="true" tabindex="-1"></a>the intercept ($\alpha$) is <span class="in">`r round(summary(glm_model_object)$coefficients[1,2],2)`</span>, </span>
<span id="cb46-1014"><a href="#cb46-1014" aria-hidden="true" tabindex="-1"></a>and the standard error for the slope </span>
<span id="cb46-1015"><a href="#cb46-1015" aria-hidden="true" tabindex="-1"></a>($\beta$) is <span class="in">`r round(summary(glm_model_object)$coefficients[2,2], 2)`</span>.</span>
<span id="cb46-1016"><a href="#cb46-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1017"><a href="#cb46-1017" aria-hidden="true" tabindex="-1"></a>We also observe that the standard error is smaller than the estimated effects. </span>
<span id="cb46-1018"><a href="#cb46-1018" aria-hidden="true" tabindex="-1"></a>Actually, to indicate a clear direction of the effect, the standard error should</span>
<span id="cb46-1019"><a href="#cb46-1019" aria-hidden="true" tabindex="-1"></a>be less than half the size of the estimated coefficient. This would ensure that</span>
<span id="cb46-1020"><a href="#cb46-1020" aria-hidden="true" tabindex="-1"></a>95% confidence intervals for the effect (see below) do not span 0. </span>
<span id="cb46-1021"><a href="#cb46-1021" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1022"><a href="#cb46-1022" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1023"><a href="#cb46-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1024"><a href="#cb46-1024" aria-hidden="true" tabindex="-1"></a>**Confidence intervals**</span>
<span id="cb46-1025"><a href="#cb46-1025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1026"><a href="#cb46-1026" aria-hidden="true" tabindex="-1"></a>For interpretation of the uncertainty, it can be easier to use the standard </span>
<span id="cb46-1027"><a href="#cb46-1027" aria-hidden="true" tabindex="-1"></a>error to calculate confidence intervals. </span>
<span id="cb46-1028"><a href="#cb46-1028" aria-hidden="true" tabindex="-1"></a>Confidence intervals indicate the range of plausible values for a parameter. </span>
<span id="cb46-1029"><a href="#cb46-1029" aria-hidden="true" tabindex="-1"></a>**They represent an interval, that if you were to collect a sample and run the</span>
<span id="cb46-1030"><a href="#cb46-1030" aria-hidden="true" tabindex="-1"></a>analysis, then</span>
<span id="cb46-1031"><a href="#cb46-1031" aria-hidden="true" tabindex="-1"></a>repeat that many many times AND each time</span>
<span id="cb46-1032"><a href="#cb46-1032" aria-hidden="true" tabindex="-1"></a>draw a confidence interval, on average 95% of the time, </span>
<span id="cb46-1033"><a href="#cb46-1033" aria-hidden="true" tabindex="-1"></a>the true population value of the parameter would be found in </span>
<span id="cb46-1034"><a href="#cb46-1034" aria-hidden="true" tabindex="-1"></a>within the confidence interval.** </span>
<span id="cb46-1035"><a href="#cb46-1035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1036"><a href="#cb46-1036" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1037"><a href="#cb46-1037" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1038"><a href="#cb46-1038" aria-hidden="true" tabindex="-1"></a>To calculate a confidence interval from a standard error you need to use the </span>
<span id="cb46-1039"><a href="#cb46-1039" aria-hidden="true" tabindex="-1"></a>formulas (this is exactly the same as for linear models):</span>
<span id="cb46-1040"><a href="#cb46-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1041"><a href="#cb46-1041" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-1042"><a href="#cb46-1042" aria-hidden="true" tabindex="-1"></a>\begin{aligned}</span>
<span id="cb46-1043"><a href="#cb46-1043" aria-hidden="true" tabindex="-1"></a>UpperCI = estimate + (1.96 SE) <span class="sc">\\</span></span>
<span id="cb46-1044"><a href="#cb46-1044" aria-hidden="true" tabindex="-1"></a>LowerCI = estimate - (1.96 SE) <span class="sc">\\</span></span>
<span id="cb46-1045"><a href="#cb46-1045" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb46-1046"><a href="#cb46-1046" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb46-1047"><a href="#cb46-1047" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1048"><a href="#cb46-1048" aria-hidden="true" tabindex="-1"></a>Remember that the estimates of the coefficients are for the effect of </span>
<span id="cb46-1049"><a href="#cb46-1049" aria-hidden="true" tabindex="-1"></a>$X$ on the log odds of $Y$ (i.e. on the link scale). Therefore any measures of </span>
<span id="cb46-1050"><a href="#cb46-1050" aria-hidden="true" tabindex="-1"></a>uncertainty are also on the link scale. </span>
<span id="cb46-1051"><a href="#cb46-1051" aria-hidden="true" tabindex="-1"></a>After estimating the upper and lower confidence intervals of these parameters, </span>
<span id="cb46-1052"><a href="#cb46-1052" aria-hidden="true" tabindex="-1"></a>it might be useful to convert them to the response scale to understand their</span>
<span id="cb46-1053"><a href="#cb46-1053" aria-hidden="true" tabindex="-1"></a>effect. This is especially true for the intercept ($\alpha$). </span>
<span id="cb46-1054"><a href="#cb46-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1055"><a href="#cb46-1055" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1056"><a href="#cb46-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1057"><a href="#cb46-1057" aria-hidden="true" tabindex="-1"></a>Using the <span class="in">`confint()`</span> function it is possible to display the 95% confidence </span>
<span id="cb46-1058"><a href="#cb46-1058" aria-hidden="true" tabindex="-1"></a>intervals for the intercept and</span>
<span id="cb46-1059"><a href="#cb46-1059" aria-hidden="true" tabindex="-1"></a>slope.</span>
<span id="cb46-1060"><a href="#cb46-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1061"><a href="#cb46-1061" aria-hidden="true" tabindex="-1"></a><span class="in">```{r confint_glm_example}</span></span>
<span id="cb46-1062"><a href="#cb46-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1063"><a href="#cb46-1063" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(glm_model_object)</span>
<span id="cb46-1064"><a href="#cb46-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1065"><a href="#cb46-1065" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1066"><a href="#cb46-1066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1067"><a href="#cb46-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1068"><a href="#cb46-1068" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1069"><a href="#cb46-1069" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1070"><a href="#cb46-1070" aria-hidden="true" tabindex="-1"></a>**Plotting uncertainty in $\alpha$ and $\beta$**</span>
<span id="cb46-1071"><a href="#cb46-1071" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1072"><a href="#cb46-1072" aria-hidden="true" tabindex="-1"></a>The above section has shown how you can quantify uncertainty in the line </span>
<span id="cb46-1073"><a href="#cb46-1073" aria-hidden="true" tabindex="-1"></a>parameters of a Binomial GLM as numbers. </span>
<span id="cb46-1074"><a href="#cb46-1074" aria-hidden="true" tabindex="-1"></a>But often in science, it is clearer to show these things visually.</span>
<span id="cb46-1075"><a href="#cb46-1075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1076"><a href="#cb46-1076" aria-hidden="true" tabindex="-1"></a>We can add the confidence intervals to @fig-log-odds-plot </span>
<span id="cb46-1077"><a href="#cb46-1077" aria-hidden="true" tabindex="-1"></a>showing our GLM fitted line. This can be done either on the link scale or the</span>
<span id="cb46-1078"><a href="#cb46-1078" aria-hidden="true" tabindex="-1"></a>response scale. </span>
<span id="cb46-1079"><a href="#cb46-1079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1080"><a href="#cb46-1080" aria-hidden="true" tabindex="-1"></a>To do this, we need to generate new predictions with the <span class="in">`predict()`</span> function. </span>
<span id="cb46-1081"><a href="#cb46-1081" aria-hidden="true" tabindex="-1"></a>We also need to calculate the upper and lower confidence interval bounds using</span>
<span id="cb46-1082"><a href="#cb46-1082" aria-hidden="true" tabindex="-1"></a>the standard error. </span>
<span id="cb46-1083"><a href="#cb46-1083" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1084"><a href="#cb46-1084" aria-hidden="true" tabindex="-1"></a>This is a little different to using <span class="in">`predict()`</span> for a linear model object. For</span>
<span id="cb46-1085"><a href="#cb46-1085" aria-hidden="true" tabindex="-1"></a>GLMs there are some different arguments. </span>
<span id="cb46-1086"><a href="#cb46-1086" aria-hidden="true" tabindex="-1"></a>The key arguments are:</span>
<span id="cb46-1087"><a href="#cb46-1087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1088"><a href="#cb46-1088" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`type`</span>. This argument can take values "link" (gives predictions on the link</span>
<span id="cb46-1089"><a href="#cb46-1089" aria-hidden="true" tabindex="-1"></a>scale), "response" (gives predictions on the response scale) or "terms"</span>
<span id="cb46-1090"><a href="#cb46-1090" aria-hidden="true" tabindex="-1"></a>(gives a matrix of the fitted values of each term in the model formula on the</span>
<span id="cb46-1091"><a href="#cb46-1091" aria-hidden="true" tabindex="-1"></a>link scale).</span>
<span id="cb46-1092"><a href="#cb46-1092" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`se.fit`</span>. This argument can be <span class="in">`TRUE`</span> or <span class="in">`FALSE`</span> and indicates whether the </span>
<span id="cb46-1093"><a href="#cb46-1093" aria-hidden="true" tabindex="-1"></a>standard error for the prediction should be given in addition to the mean </span>
<span id="cb46-1094"><a href="#cb46-1094" aria-hidden="true" tabindex="-1"></a>prediction. The standard error will be on the scale of the mean prediction i.e.</span>
<span id="cb46-1095"><a href="#cb46-1095" aria-hidden="true" tabindex="-1"></a>either link scale or response dependent on which <span class="in">`type`</span> was specified. </span>
<span id="cb46-1096"><a href="#cb46-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1097"><a href="#cb46-1097" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1098"><a href="#cb46-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1099"><a href="#cb46-1099" aria-hidden="true" tabindex="-1"></a>The <span class="in">`interval`</span> argument that was used for linear model objects to specify a</span>
<span id="cb46-1100"><a href="#cb46-1100" aria-hidden="true" tabindex="-1"></a>prediction or a confidence interval does not work for predictions on a GLM</span>
<span id="cb46-1101"><a href="#cb46-1101" aria-hidden="true" tabindex="-1"></a>object. Therefore, the standard errors presented are for a 95% confidence </span>
<span id="cb46-1102"><a href="#cb46-1102" aria-hidden="true" tabindex="-1"></a>interval. </span>
<span id="cb46-1103"><a href="#cb46-1103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1104"><a href="#cb46-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r preditions_example, echo = TRUE}</span></span>
<span id="cb46-1105"><a href="#cb46-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1106"><a href="#cb46-1106" aria-hidden="true" tabindex="-1"></a><span class="co"># First create the new data to predict for </span></span>
<span id="cb46-1107"><a href="#cb46-1107" aria-hidden="true" tabindex="-1"></a><span class="co"># This example generates 100 values of the explanatory variable from min to max</span></span>
<span id="cb46-1108"><a href="#cb46-1108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1109"><a href="#cb46-1109" aria-hidden="true" tabindex="-1"></a>newdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">explanatory =</span> <span class="fu">seq</span>(<span class="fu">min</span>(example_data<span class="sc">$</span>explanatory),</span>
<span id="cb46-1110"><a href="#cb46-1110" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">max</span>(example_data<span class="sc">$</span>explanatory),</span>
<span id="cb46-1111"><a href="#cb46-1111" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">length.out =</span> <span class="dv">100</span>))</span>
<span id="cb46-1112"><a href="#cb46-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1113"><a href="#cb46-1113" aria-hidden="true" tabindex="-1"></a><span class="co"># Then add predictions to the newdata for the link and response scale</span></span>
<span id="cb46-1114"><a href="#cb46-1114" aria-hidden="true" tabindex="-1"></a><span class="co"># REMEMBER the confidence intervals too, you can access these by taking the </span></span>
<span id="cb46-1115"><a href="#cb46-1115" aria-hidden="true" tabindex="-1"></a><span class="co"># $se.fit of the results of `predict`, $fit is the mean prediction</span></span>
<span id="cb46-1116"><a href="#cb46-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1117"><a href="#cb46-1117" aria-hidden="true" tabindex="-1"></a>predictions <span class="ot">&lt;-</span> newdata <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">predictions_link =</span> <span class="fu">predict</span>(glm_model_object, </span>
<span id="cb46-1118"><a href="#cb46-1118" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">newdata =</span> newdata, </span>
<span id="cb46-1119"><a href="#cb46-1119" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1120"><a href="#cb46-1120" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">se.fit=</span><span class="cn">TRUE</span>)<span class="sc">$</span>fit,</span>
<span id="cb46-1121"><a href="#cb46-1121" aria-hidden="true" tabindex="-1"></a>                              <span class="at">SE_link =</span> <span class="fu">predict</span>(glm_model_object, </span>
<span id="cb46-1122"><a href="#cb46-1122" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">newdata =</span> newdata, </span>
<span id="cb46-1123"><a href="#cb46-1123" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1124"><a href="#cb46-1124" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">se.fit=</span><span class="cn">TRUE</span>)<span class="sc">$</span>se.fit,</span>
<span id="cb46-1125"><a href="#cb46-1125" aria-hidden="true" tabindex="-1"></a>                              <span class="at">predictions_response =</span> <span class="fu">predict</span>(glm_model_object, </span>
<span id="cb46-1126"><a href="#cb46-1126" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">newdata =</span> newdata, </span>
<span id="cb46-1127"><a href="#cb46-1127" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">type=</span><span class="st">"response"</span>, <span class="at">se.fit=</span><span class="cn">TRUE</span>)<span class="sc">$</span>fit,</span>
<span id="cb46-1128"><a href="#cb46-1128" aria-hidden="true" tabindex="-1"></a>                              <span class="at">SE_response =</span> <span class="fu">predict</span>(glm_model_object, </span>
<span id="cb46-1129"><a href="#cb46-1129" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">newdata =</span> newdata, </span>
<span id="cb46-1130"><a href="#cb46-1130" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">type=</span><span class="st">"response"</span>,</span>
<span id="cb46-1131"><a href="#cb46-1131" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">se.fit=</span><span class="cn">TRUE</span>)<span class="sc">$</span>se.fit) <span class="sc">%&gt;%</span></span>
<span id="cb46-1132"><a href="#cb46-1132" aria-hidden="true" tabindex="-1"></a>                <span class="fu">as_tibble</span>()</span>
<span id="cb46-1133"><a href="#cb46-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1134"><a href="#cb46-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1135"><a href="#cb46-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1136"><a href="#cb46-1136" aria-hidden="true" tabindex="-1"></a>predictions</span>
<span id="cb46-1137"><a href="#cb46-1137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1138"><a href="#cb46-1138" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1139"><a href="#cb46-1139" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1140"><a href="#cb46-1140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1141"><a href="#cb46-1141" aria-hidden="true" tabindex="-1"></a>You should notice that all predictions on the response scale around bounded </span>
<span id="cb46-1142"><a href="#cb46-1142" aria-hidden="true" tabindex="-1"></a>between 0 and 1, but this is not the case on the link scale. </span>
<span id="cb46-1143"><a href="#cb46-1143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1144"><a href="#cb46-1144" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1145"><a href="#cb46-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1146"><a href="#cb46-1146" aria-hidden="true" tabindex="-1"></a>Then you can plot these. Remember these are confidence intervals not prediction</span>
<span id="cb46-1147"><a href="#cb46-1147" aria-hidden="true" tabindex="-1"></a>intervals. The <span class="in">`predict`</span> function will not give prediction intervals for a GLM. </span>
<span id="cb46-1148"><a href="#cb46-1148" aria-hidden="true" tabindex="-1"></a>In fact, it is not always an easy or sensible thing to create in this context,</span>
<span id="cb46-1149"><a href="#cb46-1149" aria-hidden="true" tabindex="-1"></a>especially for a Binomial GLM. When the mean prediction is close to 0 or 1, the</span>
<span id="cb46-1150"><a href="#cb46-1150" aria-hidden="true" tabindex="-1"></a>prediction interval can easily fill the whole possible value space (i.e. span</span>
<span id="cb46-1151"><a href="#cb46-1151" aria-hidden="true" tabindex="-1"></a>from 0 right to 1), which is not very logical and is hard to interpret. </span>
<span id="cb46-1152"><a href="#cb46-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1153"><a href="#cb46-1153" aria-hidden="true" tabindex="-1"></a>To get the upper and lower bounds of the confidence interval from</span>
<span id="cb46-1154"><a href="#cb46-1154" aria-hidden="true" tabindex="-1"></a>a GLM using <span class="in">`predict()`</span> you need to calculate the bounds yourself. </span>
<span id="cb46-1155"><a href="#cb46-1155" aria-hidden="true" tabindex="-1"></a>You need to add or</span>
<span id="cb46-1156"><a href="#cb46-1156" aria-hidden="true" tabindex="-1"></a>subtract 1.96 times the standard error from the mean prediction. </span>
<span id="cb46-1157"><a href="#cb46-1157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1158"><a href="#cb46-1158" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-plot_ci_example, fig.cap = "Examples of plots of fitted GLM line and 95% confidence intervals (grey shaded area) on the link scale (A) and on the response scale (B)"}</span></span>
<span id="cb46-1159"><a href="#cb46-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1160"><a href="#cb46-1160" aria-hidden="true" tabindex="-1"></a>predictions <span class="ot">&lt;-</span> predictions <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">upper_ci_link =</span> predictions_link <span class="sc">+</span> (<span class="dv">2</span><span class="sc">*</span>SE_link),</span>
<span id="cb46-1161"><a href="#cb46-1161" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">lower_ci_link =</span> predictions_link <span class="sc">-</span> (<span class="dv">2</span><span class="sc">*</span>SE_link),</span>
<span id="cb46-1162"><a href="#cb46-1162" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">upper_ci_response =</span> predictions_response <span class="sc">+</span> (<span class="dv">2</span><span class="sc">*</span>SE_response),</span>
<span id="cb46-1163"><a href="#cb46-1163" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">lower_ci_response =</span> predictions_response <span class="sc">-</span> (<span class="dv">2</span><span class="sc">*</span>SE_response))</span>
<span id="cb46-1164"><a href="#cb46-1164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1165"><a href="#cb46-1165" aria-hidden="true" tabindex="-1"></a><span class="co"># plot predictions on the link scale </span></span>
<span id="cb46-1166"><a href="#cb46-1166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1167"><a href="#cb46-1167" aria-hidden="true" tabindex="-1"></a>example_predictions_link <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> predictions, </span>
<span id="cb46-1168"><a href="#cb46-1168" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">aes</span>(<span class="at">x =</span> explanatory, </span>
<span id="cb46-1169"><a href="#cb46-1169" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">y =</span> predictions_link))<span class="sc">+</span></span>
<span id="cb46-1170"><a href="#cb46-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color=</span><span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-1171"><a href="#cb46-1171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> lower_ci_link, </span>
<span id="cb46-1172"><a href="#cb46-1172" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax =</span> upper_ci_link), </span>
<span id="cb46-1173"><a href="#cb46-1173" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb46-1174"><a href="#cb46-1174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"A. Predictions on link scale"</span>,</span>
<span id="cb46-1175"><a href="#cb46-1175" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Log odds of response"</span>,</span>
<span id="cb46-1176"><a href="#cb46-1176" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Explanatory"</span>)</span>
<span id="cb46-1177"><a href="#cb46-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1178"><a href="#cb46-1178" aria-hidden="true" tabindex="-1"></a><span class="co"># For this plot, the base can be the response data</span></span>
<span id="cb46-1179"><a href="#cb46-1179" aria-hidden="true" tabindex="-1"></a>example_predictions_response <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb46-1180"><a href="#cb46-1180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">data =</span> example_data, </span>
<span id="cb46-1181"><a href="#cb46-1181" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> response), </span>
<span id="cb46-1182"><a href="#cb46-1182" aria-hidden="true" tabindex="-1"></a>              <span class="at">colour =</span> <span class="st">"grey"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">height =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb46-1183"><a href="#cb46-1183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">data =</span> predictions,</span>
<span id="cb46-1184"><a href="#cb46-1184" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> explanatory,</span>
<span id="cb46-1185"><a href="#cb46-1185" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin =</span> lower_ci_response, </span>
<span id="cb46-1186"><a href="#cb46-1186" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax =</span> upper_ci_response), </span>
<span id="cb46-1187"><a href="#cb46-1187" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb46-1188"><a href="#cb46-1188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> predictions, </span>
<span id="cb46-1189"><a href="#cb46-1189" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> predictions_response), </span>
<span id="cb46-1190"><a href="#cb46-1190" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb46-1191"><a href="#cb46-1191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"B. Predictions on response scale"</span>,</span>
<span id="cb46-1192"><a href="#cb46-1192" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Probability"</span>,</span>
<span id="cb46-1193"><a href="#cb46-1193" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Explanatory"</span>)</span>
<span id="cb46-1194"><a href="#cb46-1194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1195"><a href="#cb46-1195" aria-hidden="true" tabindex="-1"></a>example_predictions_link <span class="sc">+</span> example_predictions_response</span>
<span id="cb46-1196"><a href="#cb46-1196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1197"><a href="#cb46-1197" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1198"><a href="#cb46-1198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1199"><a href="#cb46-1199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1200"><a href="#cb46-1200" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-1201"><a href="#cb46-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1202"><a href="#cb46-1202" aria-hidden="true" tabindex="-1"></a>At the end of the last section, we created of our sheep model and looked at the </span>
<span id="cb46-1203"><a href="#cb46-1203" aria-hidden="true" tabindex="-1"></a>estimated model line. </span>
<span id="cb46-1204"><a href="#cb46-1204" aria-hidden="true" tabindex="-1"></a>Now we will add uncertainty to that plot.</span>
<span id="cb46-1205"><a href="#cb46-1205" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-1206"><a href="#cb46-1206" aria-hidden="true" tabindex="-1"></a>First, we should look at the confidence intervals of our parameter estimates.</span>
<span id="cb46-1207"><a href="#cb46-1207" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-1210"><a href="#cb46-1210" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb46-1211"><a href="#cb46-1211" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">confint</span>(sheep_model),<span class="dv">2</span>)</span>
<span id="cb46-1212"><a href="#cb46-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1213"><a href="#cb46-1213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1214"><a href="#cb46-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1215"><a href="#cb46-1215" aria-hidden="true" tabindex="-1"></a>The confidence interval has been rounded to two decimal places to make it </span>
<span id="cb46-1216"><a href="#cb46-1216" aria-hidden="true" tabindex="-1"></a>easier to read. </span>
<span id="cb46-1217"><a href="#cb46-1217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1218"><a href="#cb46-1218" aria-hidden="true" tabindex="-1"></a>To add these intervals to the plot, </span>
<span id="cb46-1219"><a href="#cb46-1219" aria-hidden="true" tabindex="-1"></a>we need to make new predictions including the upper and lower bounds of the </span>
<span id="cb46-1220"><a href="#cb46-1220" aria-hidden="true" tabindex="-1"></a>prediction interval. We will do this on the link scale and the response </span>
<span id="cb46-1221"><a href="#cb46-1221" aria-hidden="true" tabindex="-1"></a>response scale. </span>
<span id="cb46-1222"><a href="#cb46-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1223"><a href="#cb46-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1224"><a href="#cb46-1224" aria-hidden="true" tabindex="-1"></a><span class="in">```{r sheep_predictions}</span></span>
<span id="cb46-1225"><a href="#cb46-1225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1226"><a href="#cb46-1226" aria-hidden="true" tabindex="-1"></a><span class="co"># first make some new data to predict for</span></span>
<span id="cb46-1227"><a href="#cb46-1227" aria-hidden="true" tabindex="-1"></a>newdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Weight =</span> <span class="fu">seq</span>(<span class="fu">min</span>(sheep_data<span class="sc">$</span>Weight),</span>
<span id="cb46-1228"><a href="#cb46-1228" aria-hidden="true" tabindex="-1"></a>               <span class="fu">max</span>(sheep_data<span class="sc">$</span>Weight),</span>
<span id="cb46-1229"><a href="#cb46-1229" aria-hidden="true" tabindex="-1"></a>               <span class="at">length.out =</span> <span class="dv">100</span>))</span>
<span id="cb46-1230"><a href="#cb46-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1231"><a href="#cb46-1231" aria-hidden="true" tabindex="-1"></a>sheep_predictions <span class="ot">&lt;-</span> newdata <span class="sc">%&gt;%</span> </span>
<span id="cb46-1232"><a href="#cb46-1232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">predictions_link =</span> <span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1233"><a href="#cb46-1233" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1234"><a href="#cb46-1234" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1235"><a href="#cb46-1235" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit,</span>
<span id="cb46-1236"><a href="#cb46-1236" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the next lines add the lower and upper ci bounds</span></span>
<span id="cb46-1237"><a href="#cb46-1237" aria-hidden="true" tabindex="-1"></a>            <span class="at">lower_ci_link =</span> <span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1238"><a href="#cb46-1238" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1239"><a href="#cb46-1239" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1240"><a href="#cb46-1240" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit <span class="sc">-</span> </span>
<span id="cb46-1241"><a href="#cb46-1241" aria-hidden="true" tabindex="-1"></a>                                    (<span class="fl">1.96</span><span class="sc">*</span><span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1242"><a href="#cb46-1242" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1243"><a href="#cb46-1243" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1244"><a href="#cb46-1244" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>se.fit),</span>
<span id="cb46-1245"><a href="#cb46-1245" aria-hidden="true" tabindex="-1"></a>            <span class="at">upper_ci_link =</span> <span class="fu">predict</span>(sheep_model, </span>
<span id="cb46-1246"><a href="#cb46-1246" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1247"><a href="#cb46-1247" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1248"><a href="#cb46-1248" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit <span class="sc">+</span> </span>
<span id="cb46-1249"><a href="#cb46-1249" aria-hidden="true" tabindex="-1"></a>                                    (<span class="fl">1.96</span><span class="sc">*</span><span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1250"><a href="#cb46-1250" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1251"><a href="#cb46-1251" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"link"</span>, </span>
<span id="cb46-1252"><a href="#cb46-1252" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>se.fit),</span>
<span id="cb46-1253"><a href="#cb46-1253" aria-hidden="true" tabindex="-1"></a>            <span class="co"># then predict on the response scale</span></span>
<span id="cb46-1254"><a href="#cb46-1254" aria-hidden="true" tabindex="-1"></a>            <span class="at">predictions_response =</span> <span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1255"><a href="#cb46-1255" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1256"><a href="#cb46-1256" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"response"</span>, </span>
<span id="cb46-1257"><a href="#cb46-1257" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit,</span>
<span id="cb46-1258"><a href="#cb46-1258" aria-hidden="true" tabindex="-1"></a>            <span class="co"># and add the confidence interval bounds</span></span>
<span id="cb46-1259"><a href="#cb46-1259" aria-hidden="true" tabindex="-1"></a>            <span class="at">lower_ci_response =</span> <span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1260"><a href="#cb46-1260" aria-hidden="true" tabindex="-1"></a>                                    newdata, </span>
<span id="cb46-1261"><a href="#cb46-1261" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"response"</span>, </span>
<span id="cb46-1262"><a href="#cb46-1262" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit <span class="sc">-</span> </span>
<span id="cb46-1263"><a href="#cb46-1263" aria-hidden="true" tabindex="-1"></a>                                    (<span class="fl">1.96</span><span class="sc">*</span><span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1264"><a href="#cb46-1264" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1265"><a href="#cb46-1265" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"response"</span>, </span>
<span id="cb46-1266"><a href="#cb46-1266" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>se.fit),</span>
<span id="cb46-1267"><a href="#cb46-1267" aria-hidden="true" tabindex="-1"></a>            <span class="at">upper_ci_response =</span> <span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1268"><a href="#cb46-1268" aria-hidden="true" tabindex="-1"></a>                                    newdata, </span>
<span id="cb46-1269"><a href="#cb46-1269" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"response"</span>, </span>
<span id="cb46-1270"><a href="#cb46-1270" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>fit <span class="sc">+</span> </span>
<span id="cb46-1271"><a href="#cb46-1271" aria-hidden="true" tabindex="-1"></a>                                    (<span class="fl">1.96</span><span class="sc">*</span><span class="fu">predict</span>(sheep_model,</span>
<span id="cb46-1272"><a href="#cb46-1272" aria-hidden="true" tabindex="-1"></a>                                    newdata,</span>
<span id="cb46-1273"><a href="#cb46-1273" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">type=</span><span class="st">"response"</span>, </span>
<span id="cb46-1274"><a href="#cb46-1274" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">se.fit =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>se.fit))</span>
<span id="cb46-1275"><a href="#cb46-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1276"><a href="#cb46-1276" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1277"><a href="#cb46-1277" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1278"><a href="#cb46-1278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1279"><a href="#cb46-1279" aria-hidden="true" tabindex="-1"></a>Once we have created predictions of $Y$ from the model object, </span>
<span id="cb46-1280"><a href="#cb46-1280" aria-hidden="true" tabindex="-1"></a>we can plot them using <span class="in">`geom_line()`</span> and <span class="in">`geom_ribbon()`</span> as in the code below.</span>
<span id="cb46-1281"><a href="#cb46-1281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1282"><a href="#cb46-1282" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep_predictions_plot, fig.cap = "A. plot of the fitted GLM line on the link scale including 95% confidence interval (grey), B. plot of the fitted GLM line on the response scale including 95% confidence interval (grey)"}</span></span>
<span id="cb46-1283"><a href="#cb46-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1284"><a href="#cb46-1284" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot of Confidence Intervals on the link scale</span></span>
<span id="cb46-1285"><a href="#cb46-1285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1286"><a href="#cb46-1286" aria-hidden="true" tabindex="-1"></a>sheep_predictions_link <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> sheep_predictions, </span>
<span id="cb46-1287"><a href="#cb46-1287" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">aes</span>(<span class="at">x =</span> Weight, </span>
<span id="cb46-1288"><a href="#cb46-1288" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">y =</span> predictions_link))<span class="sc">+</span></span>
<span id="cb46-1289"><a href="#cb46-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color=</span><span class="st">"blue"</span>)<span class="sc">+</span></span>
<span id="cb46-1290"><a href="#cb46-1290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> lower_ci_link, </span>
<span id="cb46-1291"><a href="#cb46-1291" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax =</span> upper_ci_link), </span>
<span id="cb46-1292"><a href="#cb46-1292" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb46-1293"><a href="#cb46-1293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"A. Predictions on link scale"</span>,</span>
<span id="cb46-1294"><a href="#cb46-1294" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Log odds of survival"</span>,</span>
<span id="cb46-1295"><a href="#cb46-1295" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Body weight (kg)"</span>)</span>
<span id="cb46-1296"><a href="#cb46-1296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1297"><a href="#cb46-1297" aria-hidden="true" tabindex="-1"></a><span class="co"># For this plot, the base can be the response data</span></span>
<span id="cb46-1298"><a href="#cb46-1298" aria-hidden="true" tabindex="-1"></a>sheep_predictions_response <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb46-1299"><a href="#cb46-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">data =</span> sheep_data, </span>
<span id="cb46-1300"><a href="#cb46-1300" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> Survival), </span>
<span id="cb46-1301"><a href="#cb46-1301" aria-hidden="true" tabindex="-1"></a>              <span class="at">colour =</span> <span class="st">"grey"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">height =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb46-1302"><a href="#cb46-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">data =</span> sheep_predictions,</span>
<span id="cb46-1303"><a href="#cb46-1303" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Weight,</span>
<span id="cb46-1304"><a href="#cb46-1304" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin =</span> lower_ci_response, </span>
<span id="cb46-1305"><a href="#cb46-1305" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax =</span> upper_ci_response), </span>
<span id="cb46-1306"><a href="#cb46-1306" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb46-1307"><a href="#cb46-1307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> sheep_predictions, </span>
<span id="cb46-1308"><a href="#cb46-1308" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> predictions_response), </span>
<span id="cb46-1309"><a href="#cb46-1309" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb46-1310"><a href="#cb46-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"B. Predictions on response scale"</span>,</span>
<span id="cb46-1311"><a href="#cb46-1311" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Probability of survival"</span>,</span>
<span id="cb46-1312"><a href="#cb46-1312" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Body weight (kg)"</span>)</span>
<span id="cb46-1313"><a href="#cb46-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1314"><a href="#cb46-1314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1315"><a href="#cb46-1315" aria-hidden="true" tabindex="-1"></a>sheep_predictions_link <span class="sc">+</span> sheep_predictions_response</span>
<span id="cb46-1316"><a href="#cb46-1316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1317"><a href="#cb46-1317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1318"><a href="#cb46-1318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1319"><a href="#cb46-1319" aria-hidden="true" tabindex="-1"></a>It can be observed from the figure above that the confidence intervals for the </span>
<span id="cb46-1320"><a href="#cb46-1320" aria-hidden="true" tabindex="-1"></a>predictions around $1$ are very narrower. </span>
<span id="cb46-1321"><a href="#cb46-1321" aria-hidden="true" tabindex="-1"></a>This means that as we get closer to the true classification (0 or 1), </span>
<span id="cb46-1322"><a href="#cb46-1322" aria-hidden="true" tabindex="-1"></a>we are able to estimate them with much certainty.</span>
<span id="cb46-1323"><a href="#cb46-1323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1324"><a href="#cb46-1324" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1325"><a href="#cb46-1325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1326"><a href="#cb46-1326" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="fas fa-tasks"&gt;&lt;/i&gt; Model checking {.tabset .tabset-fade}</span></span>
<span id="cb46-1327"><a href="#cb46-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1328"><a href="#cb46-1328" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1329"><a href="#cb46-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1330"><a href="#cb46-1330" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-1331"><a href="#cb46-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1332"><a href="#cb46-1332" aria-hidden="true" tabindex="-1"></a>Until this point, we have run a Binomial GLM, got parameter estimates and </span>
<span id="cb46-1333"><a href="#cb46-1333" aria-hidden="true" tabindex="-1"></a>quantified uncertainty. Next, **we will look at how to know if the model is </span>
<span id="cb46-1334"><a href="#cb46-1334" aria-hidden="true" tabindex="-1"></a>any good.** To do this, we need to check if the assumptions for a Binomial GLM</span>
<span id="cb46-1335"><a href="#cb46-1335" aria-hidden="true" tabindex="-1"></a>are met for our data. </span>
<span id="cb46-1336"><a href="#cb46-1336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1337"><a href="#cb46-1337" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1338"><a href="#cb46-1338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1339"><a href="#cb46-1339" aria-hidden="true" tabindex="-1"></a>To check the assumptions of a Binomial GLM, we will follow the same procedure as</span>
<span id="cb46-1340"><a href="#cb46-1340" aria-hidden="true" tabindex="-1"></a>for GLMs more broadly. For a reminder of this, go back to the </span>
<span id="cb46-1341"><a href="#cb46-1341" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Introduction to GLMs page</span><span class="co">]()</span>.</span>
<span id="cb46-1342"><a href="#cb46-1342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1343"><a href="#cb46-1343" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1344"><a href="#cb46-1344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1345"><a href="#cb46-1345" aria-hidden="true" tabindex="-1"></a>The first step is to:</span>
<span id="cb46-1346"><a href="#cb46-1346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1347"><a href="#cb46-1347" aria-hidden="true" tabindex="-1"></a>**Plot the model predictions and the data together**</span>
<span id="cb46-1348"><a href="#cb46-1348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1349"><a href="#cb46-1349" aria-hidden="true" tabindex="-1"></a>From this plot, it is possible to test: </span>
<span id="cb46-1350"><a href="#cb46-1350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1351"><a href="#cb46-1351" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>that the fitted model line is capturing the relationship between x and y on</span>
<span id="cb46-1352"><a href="#cb46-1352" aria-hidden="true" tabindex="-1"></a>the response scale</span>
<span id="cb46-1353"><a href="#cb46-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1354"><a href="#cb46-1354" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1355"><a href="#cb46-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1356"><a href="#cb46-1356" aria-hidden="true" tabindex="-1"></a>To do this, we can re-plot the example plot from the section above. </span>
<span id="cb46-1357"><a href="#cb46-1357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1360"><a href="#cb46-1360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb46-1361"><a href="#cb46-1361" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-example_predictions_plot</span></span>
<span id="cb46-1362"><a href="#cb46-1362" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "plot of the fitted GLM line on the response scale including 95% confidence interval (grey)"</span></span>
<span id="cb46-1363"><a href="#cb46-1363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1364"><a href="#cb46-1364" aria-hidden="true" tabindex="-1"></a>example_predictions_response <span class="sc">+</span></span>
<span id="cb46-1365"><a href="#cb46-1365" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model predictions and data on response scale"</span>)</span>
<span id="cb46-1366"><a href="#cb46-1366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1367"><a href="#cb46-1367" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1368"><a href="#cb46-1368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1369"><a href="#cb46-1369" aria-hidden="true" tabindex="-1"></a>If you look carefully at @fig-example_predictions_plot you can see that </span>
<span id="cb46-1370"><a href="#cb46-1370" aria-hidden="true" tabindex="-1"></a>there are more data points that are 0 for the response for </span>
<span id="cb46-1371"><a href="#cb46-1371" aria-hidden="true" tabindex="-1"></a>negative values of the explanatory variable.</span>
<span id="cb46-1372"><a href="#cb46-1372" aria-hidden="true" tabindex="-1"></a>As the values of the explanatory variable increase, we find progressively more</span>
<span id="cb46-1373"><a href="#cb46-1373" aria-hidden="true" tabindex="-1"></a>1 values for the response. </span>
<span id="cb46-1374"><a href="#cb46-1374" aria-hidden="true" tabindex="-1"></a>Therefore, we would expect a model line that that an intercept of close to 0 on</span>
<span id="cb46-1375"><a href="#cb46-1375" aria-hidden="true" tabindex="-1"></a>the response scale and then shows a positive relationship between x and y, </span>
<span id="cb46-1376"><a href="#cb46-1376" aria-hidden="true" tabindex="-1"></a>reaching an asymptote (levelling off) at a response value of 1 at the extreme</span>
<span id="cb46-1377"><a href="#cb46-1377" aria-hidden="true" tabindex="-1"></a>end of the observed values of the explanatory variable (around 20). </span>
<span id="cb46-1378"><a href="#cb46-1378" aria-hidden="true" tabindex="-1"></a>This is exactly what the fitted model line does! </span>
<span id="cb46-1379"><a href="#cb46-1379" aria-hidden="true" tabindex="-1"></a>Therefore, in this case, it is possible to see that the model fits the data</span>
<span id="cb46-1380"><a href="#cb46-1380" aria-hidden="true" tabindex="-1"></a>quite well.</span>
<span id="cb46-1381"><a href="#cb46-1381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1382"><a href="#cb46-1382" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1383"><a href="#cb46-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1384"><a href="#cb46-1384" aria-hidden="true" tabindex="-1"></a>Note that if you had proportional data or those from multiple trials that the </span>
<span id="cb46-1385"><a href="#cb46-1385" aria-hidden="true" tabindex="-1"></a>data points might be distributed around the fitted line. </span>
<span id="cb46-1386"><a href="#cb46-1386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1387"><a href="#cb46-1387" aria-hidden="true" tabindex="-1"></a>Then it would be possible to also test:</span>
<span id="cb46-1388"><a href="#cb46-1388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1389"><a href="#cb46-1389" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>whether the data points are evenly distributed around the fitted line</span>
<span id="cb46-1390"><a href="#cb46-1390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1391"><a href="#cb46-1391" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-example_predictions_plot_proportion, fig.cap = "plot of the fitted GLM line on the response scale with proportional response, including 95% confidence interval (grey)"}</span></span>
<span id="cb46-1392"><a href="#cb46-1392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1393"><a href="#cb46-1393" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> explanatory, <span class="at">y =</span> successes<span class="sc">/</span>(successes <span class="sc">+</span> failures)), </span>
<span id="cb46-1394"><a href="#cb46-1394" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> example_data2)<span class="sc">+</span></span>
<span id="cb46-1395"><a href="#cb46-1395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>)<span class="sc">+</span></span>
<span id="cb46-1396"><a href="#cb46-1396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span>stats<span class="sc">::</span>glm, <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span><span class="st">"binomial"</span>), </span>
<span id="cb46-1397"><a href="#cb46-1397" aria-hidden="true" tabindex="-1"></a>              <span class="at">se=</span><span class="cn">TRUE</span>)</span>
<span id="cb46-1398"><a href="#cb46-1398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1399"><a href="#cb46-1399" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1400"><a href="#cb46-1400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1401"><a href="#cb46-1401" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span> </span>
<span id="cb46-1402"><a href="#cb46-1402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1403"><a href="#cb46-1403" aria-hidden="true" tabindex="-1"></a>In this example, the fit is much worse. </span>
<span id="cb46-1404"><a href="#cb46-1404" aria-hidden="true" tabindex="-1"></a>Although there are an even number of data points above and below the fitted model</span>
<span id="cb46-1405"><a href="#cb46-1405" aria-hidden="true" tabindex="-1"></a>line, they are very widely spread away from it. </span>
<span id="cb46-1406"><a href="#cb46-1406" aria-hidden="true" tabindex="-1"></a>The model line does not seem to be explaining the data well, that is partly </span>
<span id="cb46-1407"><a href="#cb46-1407" aria-hidden="true" tabindex="-1"></a>because there is so little data (only 10 data points).</span>
<span id="cb46-1408"><a href="#cb46-1408" aria-hidden="true" tabindex="-1"></a>But hopefully shows what a less well fitting model looks like. </span>
<span id="cb46-1409"><a href="#cb46-1409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1410"><a href="#cb46-1410" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1411"><a href="#cb46-1411" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1412"><a href="#cb46-1412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1413"><a href="#cb46-1413" aria-hidden="true" tabindex="-1"></a>The second step for checking model assumptions is: </span>
<span id="cb46-1414"><a href="#cb46-1414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1415"><a href="#cb46-1415" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1416"><a href="#cb46-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1417"><a href="#cb46-1417" aria-hidden="true" tabindex="-1"></a>**Overdispersion**</span>
<span id="cb46-1418"><a href="#cb46-1418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1419"><a href="#cb46-1419" aria-hidden="true" tabindex="-1"></a>Checking for overdispersion tests if these assumptions are met:</span>
<span id="cb46-1420"><a href="#cb46-1420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1421"><a href="#cb46-1421" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Correct variance function is used</span>
<span id="cb46-1422"><a href="#cb46-1422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1423"><a href="#cb46-1423" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Dispersion parameter is constant</span>
<span id="cb46-1424"><a href="#cb46-1424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1425"><a href="#cb46-1425" aria-hidden="true" tabindex="-1"></a>In a Binomial GLM it is assumed that the variance is controlled by the </span>
<span id="cb46-1426"><a href="#cb46-1426" aria-hidden="true" tabindex="-1"></a>mean. But this is not always true. </span>
<span id="cb46-1427"><a href="#cb46-1427" aria-hidden="true" tabindex="-1"></a>It is something we need to check as part of the model checking process.</span>
<span id="cb46-1428"><a href="#cb46-1428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1429"><a href="#cb46-1429" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1430"><a href="#cb46-1430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1431"><a href="#cb46-1431" aria-hidden="true" tabindex="-1"></a>To check for overdispersion in GLMs we need to calculate the deviance ratio. </span>
<span id="cb46-1432"><a href="#cb46-1432" aria-hidden="true" tabindex="-1"></a>This is done by taking the ratio of residual deviance and residual degrees of </span>
<span id="cb46-1433"><a href="#cb46-1433" aria-hidden="true" tabindex="-1"></a>freedom.</span>
<span id="cb46-1434"><a href="#cb46-1434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1435"><a href="#cb46-1435" aria-hidden="true" tabindex="-1"></a>The assumption is that the deviance = 1 for Binomial GLMs. </span>
<span id="cb46-1436"><a href="#cb46-1436" aria-hidden="true" tabindex="-1"></a>Values higher than 2 indicates that the residuals likely have more dispersion</span>
<span id="cb46-1437"><a href="#cb46-1437" aria-hidden="true" tabindex="-1"></a>than expected. </span>
<span id="cb46-1438"><a href="#cb46-1438" aria-hidden="true" tabindex="-1"></a>But there is no fixed threshold for when overdispersion is occurring,</span>
<span id="cb46-1439"><a href="#cb46-1439" aria-hidden="true" tabindex="-1"></a>each researcher has to decide on their own how high a deviance ratio is a problem.</span>
<span id="cb46-1440"><a href="#cb46-1440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1441"><a href="#cb46-1441" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1442"><a href="#cb46-1442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1443"><a href="#cb46-1443" aria-hidden="true" tabindex="-1"></a>The deviance values are found in <span class="in">`summary()`</span> at the bottom of the table</span>
<span id="cb46-1444"><a href="#cb46-1444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1445"><a href="#cb46-1445" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, echo=TRUE, include=TRUE}</span></span>
<span id="cb46-1446"><a href="#cb46-1446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1447"><a href="#cb46-1447" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_model_object)</span>
<span id="cb46-1448"><a href="#cb46-1448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1449"><a href="#cb46-1449" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1450"><a href="#cb46-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1451"><a href="#cb46-1451" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span> </span>
<span id="cb46-1452"><a href="#cb46-1452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1453"><a href="#cb46-1453" aria-hidden="true" tabindex="-1"></a>For the example above </span>
<span id="cb46-1454"><a href="#cb46-1454" aria-hidden="true" tabindex="-1"></a>the **Deviance ratio = 124.79/98 = `r 124.79/98`.**</span>
<span id="cb46-1455"><a href="#cb46-1455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1456"><a href="#cb46-1456" aria-hidden="true" tabindex="-1"></a>This value is a bit above 1, but much lower than 2, so the dispersion assumption</span>
<span id="cb46-1457"><a href="#cb46-1457" aria-hidden="true" tabindex="-1"></a>seems to be met in the example. </span>
<span id="cb46-1458"><a href="#cb46-1458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1459"><a href="#cb46-1459" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1460"><a href="#cb46-1460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1461"><a href="#cb46-1461" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-1462"><a href="#cb46-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1463"><a href="#cb46-1463" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1464"><a href="#cb46-1464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1465"><a href="#cb46-1465" aria-hidden="true" tabindex="-1"></a>Using the theory covered in the previous section, </span>
<span id="cb46-1466"><a href="#cb46-1466" aria-hidden="true" tabindex="-1"></a>we can now check our sheepModel </span>
<span id="cb46-1467"><a href="#cb46-1467" aria-hidden="true" tabindex="-1"></a>to ensure that it meets the assumptions of a Binomial GLM.</span>
<span id="cb46-1468"><a href="#cb46-1468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1469"><a href="#cb46-1469" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1470"><a href="#cb46-1470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1471"><a href="#cb46-1471" aria-hidden="true" tabindex="-1"></a>First, we will look again at **the plot of the model predictions and the data</span>
<span id="cb46-1472"><a href="#cb46-1472" aria-hidden="true" tabindex="-1"></a>together.**</span>
<span id="cb46-1473"><a href="#cb46-1473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1474"><a href="#cb46-1474" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1475"><a href="#cb46-1475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1476"><a href="#cb46-1476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep_predictions_plot2, fig.cap = "Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)"}</span></span>
<span id="cb46-1477"><a href="#cb46-1477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1478"><a href="#cb46-1478" aria-hidden="true" tabindex="-1"></a>sheep_predictions_response <span class="sc">+</span></span>
<span id="cb46-1479"><a href="#cb46-1479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model predictions on response scale"</span>)</span>
<span id="cb46-1480"><a href="#cb46-1480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1481"><a href="#cb46-1481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1482"><a href="#cb46-1482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1483"><a href="#cb46-1483" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span> </span>
<span id="cb46-1484"><a href="#cb46-1484" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb46-1485"><a href="#cb46-1485" aria-hidden="true" tabindex="-1"></a>The fit of this model is perhaps a little harder than the example in the theory</span>
<span id="cb46-1486"><a href="#cb46-1486" aria-hidden="true" tabindex="-1"></a>section. </span>
<span id="cb46-1487"><a href="#cb46-1487" aria-hidden="true" tabindex="-1"></a>This is partly caused because there is more data, so it overlaps a lot. </span>
<span id="cb46-1488"><a href="#cb46-1488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1489"><a href="#cb46-1489" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep_predictions_plot3, fig.cap = "Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)"}</span></span>
<span id="cb46-1490"><a href="#cb46-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1491"><a href="#cb46-1491" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb46-1492"><a href="#cb46-1492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">data =</span> sheep_data, </span>
<span id="cb46-1493"><a href="#cb46-1493" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> Survival), </span>
<span id="cb46-1494"><a href="#cb46-1494" aria-hidden="true" tabindex="-1"></a>              <span class="at">colour =</span> <span class="st">"grey70"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">height =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb46-1495"><a href="#cb46-1495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">data =</span> sheep_predictions,</span>
<span id="cb46-1496"><a href="#cb46-1496" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Weight,</span>
<span id="cb46-1497"><a href="#cb46-1497" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymin =</span> lower_ci_response, </span>
<span id="cb46-1498"><a href="#cb46-1498" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ymax =</span> upper_ci_response), </span>
<span id="cb46-1499"><a href="#cb46-1499" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb46-1500"><a href="#cb46-1500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> sheep_predictions, </span>
<span id="cb46-1501"><a href="#cb46-1501" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> Weight, <span class="at">y =</span> predictions_response), </span>
<span id="cb46-1502"><a href="#cb46-1502" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb46-1503"><a href="#cb46-1503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model predictions on response scale"</span>,</span>
<span id="cb46-1504"><a href="#cb46-1504" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Probability of survival"</span>,</span>
<span id="cb46-1505"><a href="#cb46-1505" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Body weight (kg)"</span>)</span>
<span id="cb46-1506"><a href="#cb46-1506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1507"><a href="#cb46-1507" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1508"><a href="#cb46-1508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1509"><a href="#cb46-1509" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1510"><a href="#cb46-1510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1511"><a href="#cb46-1511" aria-hidden="true" tabindex="-1"></a>If we increase the jitter of the points from 0.01 to 0.1, it becomes a bit easier</span>
<span id="cb46-1512"><a href="#cb46-1512" aria-hidden="true" tabindex="-1"></a>to see the pattern. </span>
<span id="cb46-1513"><a href="#cb46-1513" aria-hidden="true" tabindex="-1"></a>Now it is possible to see there are many more data points with a response value</span>
<span id="cb46-1514"><a href="#cb46-1514" aria-hidden="true" tabindex="-1"></a>of 1, than there are data points with a response value of 0. </span>
<span id="cb46-1515"><a href="#cb46-1515" aria-hidden="true" tabindex="-1"></a>There are especially few response values of 0 (death) for sheep with a body</span>
<span id="cb46-1516"><a href="#cb46-1516" aria-hidden="true" tabindex="-1"></a>weight above 25kg.</span>
<span id="cb46-1517"><a href="#cb46-1517" aria-hidden="true" tabindex="-1"></a>Very few of these big sheep die. </span>
<span id="cb46-1518"><a href="#cb46-1518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1519"><a href="#cb46-1519" aria-hidden="true" tabindex="-1"></a>So, we would expect a model line to fit well, if it captures small sheep having</span>
<span id="cb46-1520"><a href="#cb46-1520" aria-hidden="true" tabindex="-1"></a>a lower survival change than big sheep. </span>
<span id="cb46-1521"><a href="#cb46-1521" aria-hidden="true" tabindex="-1"></a>It seems that the model does capture the data quite well in this case. </span>
<span id="cb46-1522"><a href="#cb46-1522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1523"><a href="#cb46-1523" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1524"><a href="#cb46-1524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1525"><a href="#cb46-1525" aria-hidden="true" tabindex="-1"></a>Next, we should **check overdispersion.**</span>
<span id="cb46-1526"><a href="#cb46-1526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1527"><a href="#cb46-1527" aria-hidden="true" tabindex="-1"></a><span class="in">```{r overdispersion_sheep}</span></span>
<span id="cb46-1528"><a href="#cb46-1528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1529"><a href="#cb46-1529" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(sheep_model)</span>
<span id="cb46-1530"><a href="#cb46-1530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1531"><a href="#cb46-1531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1532"><a href="#cb46-1532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1533"><a href="#cb46-1533" aria-hidden="true" tabindex="-1"></a>The deviance ratio for the sheep model is:</span>
<span id="cb46-1534"><a href="#cb46-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1535"><a href="#cb46-1535" aria-hidden="true" tabindex="-1"></a>1251.5/1326 = <span class="in">`r 1251.5/1326`</span>. </span>
<span id="cb46-1536"><a href="#cb46-1536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1537"><a href="#cb46-1537" aria-hidden="true" tabindex="-1"></a>This is actually slightly below the expected dispersion of 1, but it is pretty </span>
<span id="cb46-1538"><a href="#cb46-1538" aria-hidden="true" tabindex="-1"></a>close.</span>
<span id="cb46-1539"><a href="#cb46-1539" aria-hidden="true" tabindex="-1"></a>So, it seems the dispersion assumption is met for the sheep model. </span>
<span id="cb46-1540"><a href="#cb46-1540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1541"><a href="#cb46-1541" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1542"><a href="#cb46-1542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1543"><a href="#cb46-1543" aria-hidden="true" tabindex="-1"></a>In the next section we will interpret our results.</span>
<span id="cb46-1544"><a href="#cb46-1544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1545"><a href="#cb46-1545" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span> </span>
<span id="cb46-1546"><a href="#cb46-1546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1547"><a href="#cb46-1547" aria-hidden="true" tabindex="-1"></a><span class="fu">## &lt;i class="far fa-lightbulb"&gt;&lt;/i&gt; Draw conclusions {.tabset .tabset-fade}</span></span>
<span id="cb46-1548"><a href="#cb46-1548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1549"><a href="#cb46-1549" aria-hidden="true" tabindex="-1"></a><span class="fu">### Theory</span></span>
<span id="cb46-1550"><a href="#cb46-1550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1551"><a href="#cb46-1551" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1552"><a href="#cb46-1552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1553"><a href="#cb46-1553" aria-hidden="true" tabindex="-1"></a>In the previous sections you learned how to run a Binomial GLM, </span>
<span id="cb46-1554"><a href="#cb46-1554" aria-hidden="true" tabindex="-1"></a>what the parameters of the model mean, </span>
<span id="cb46-1555"><a href="#cb46-1555" aria-hidden="true" tabindex="-1"></a>how to quantify uncertainty in the parameters, and </span>
<span id="cb46-1556"><a href="#cb46-1556" aria-hidden="true" tabindex="-1"></a>how to check the assumptions of the model. </span>
<span id="cb46-1557"><a href="#cb46-1557" aria-hidden="true" tabindex="-1"></a>Now, we can bring everything together to draw some conclusions. </span>
<span id="cb46-1558"><a href="#cb46-1558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1559"><a href="#cb46-1559" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1560"><a href="#cb46-1560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1561"><a href="#cb46-1561" aria-hidden="true" tabindex="-1"></a>There are several components required in drawing a conclusion:</span>
<span id="cb46-1562"><a href="#cb46-1562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1563"><a href="#cb46-1563" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>statement of the maximum likelihood estimate of the parameters of </span>
<span id="cb46-1564"><a href="#cb46-1564" aria-hidden="true" tabindex="-1"></a>interest (including strength and direction).</span>
<span id="cb46-1565"><a href="#cb46-1565" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>statement of the uncertainty in the estimate</span>
<span id="cb46-1566"><a href="#cb46-1566" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>link the results to biology and the question asked</span>
<span id="cb46-1567"><a href="#cb46-1567" aria-hidden="true" tabindex="-1"></a><span class="ss"> * </span>Discussion of next directions</span>
<span id="cb46-1568"><a href="#cb46-1568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1569"><a href="#cb46-1569" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1570"><a href="#cb46-1570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1571"><a href="#cb46-1571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1572"><a href="#cb46-1572" aria-hidden="true" tabindex="-1"></a><span class="fu">### Worked example </span></span>
<span id="cb46-1573"><a href="#cb46-1573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1574"><a href="#cb46-1574" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1575"><a href="#cb46-1575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1576"><a href="#cb46-1576" aria-hidden="true" tabindex="-1"></a>This is the final section of our analysis of the data on survival of sheep. </span>
<span id="cb46-1577"><a href="#cb46-1577" aria-hidden="true" tabindex="-1"></a>We will now bring together all of the results we have obtained and draw a </span>
<span id="cb46-1578"><a href="#cb46-1578" aria-hidden="true" tabindex="-1"></a>conclusion following the same format as the in the theory section.</span>
<span id="cb46-1579"><a href="#cb46-1579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1580"><a href="#cb46-1580" aria-hidden="true" tabindex="-1"></a>A reminder, we were asking: **Does weight influence survival of Soay sheep?**</span>
<span id="cb46-1581"><a href="#cb46-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1582"><a href="#cb46-1582" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1583"><a href="#cb46-1583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1584"><a href="#cb46-1584" aria-hidden="true" tabindex="-1"></a>The maximum likelihood estimate of the relationship between weight and survival </span>
<span id="cb46-1585"><a href="#cb46-1585" aria-hidden="true" tabindex="-1"></a>probability is <span class="in">`r coef(sheep_model)[2]`</span>. </span>
<span id="cb46-1586"><a href="#cb46-1586" aria-hidden="true" tabindex="-1"></a>In other words, for every 1 kg increase in body weight,</span>
<span id="cb46-1587"><a href="#cb46-1587" aria-hidden="true" tabindex="-1"></a>the log-odds of the survival of the sheep increases by</span>
<span id="cb46-1588"><a href="#cb46-1588" aria-hidden="true" tabindex="-1"></a><span class="in">`r round(coef(sheep_model)[2],2)`</span>. </span>
<span id="cb46-1589"><a href="#cb46-1589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1590"><a href="#cb46-1590" aria-hidden="true" tabindex="-1"></a>On the response scale this increase is non-linear and is better shown with a</span>
<span id="cb46-1591"><a href="#cb46-1591" aria-hidden="true" tabindex="-1"></a>graph. </span>
<span id="cb46-1592"><a href="#cb46-1592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1593"><a href="#cb46-1593" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig-sheep_predictions_plot4, fig.cap = "Plot of the fitted GLM line on the response scale including 95% confidence interval (grey)"}</span></span>
<span id="cb46-1594"><a href="#cb46-1594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1595"><a href="#cb46-1595" aria-hidden="true" tabindex="-1"></a>sheep_predictions_response <span class="sc">+</span></span>
<span id="cb46-1596"><a href="#cb46-1596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model predictions on response scale"</span>)</span>
<span id="cb46-1597"><a href="#cb46-1597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1598"><a href="#cb46-1598" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb46-1599"><a href="#cb46-1599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1600"><a href="#cb46-1600" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1601"><a href="#cb46-1601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1602"><a href="#cb46-1602" aria-hidden="true" tabindex="-1"></a>Here it is clearer to see the increase in survival probability with body weight. </span>
<span id="cb46-1603"><a href="#cb46-1603" aria-hidden="true" tabindex="-1"></a>The increase in survival is steepest for smaller sheep, levelling off after</span>
<span id="cb46-1604"><a href="#cb46-1604" aria-hidden="true" tabindex="-1"></a>around 25kg at a 0.95 probability of survival in a given year for these big sheep. </span>
<span id="cb46-1605"><a href="#cb46-1605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1606"><a href="#cb46-1606" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1607"><a href="#cb46-1607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1608"><a href="#cb46-1608" aria-hidden="true" tabindex="-1"></a>When we look at the uncertainty in this estimate, we see the 95% confidence </span>
<span id="cb46-1609"><a href="#cb46-1609" aria-hidden="true" tabindex="-1"></a>interval is <span class="in">`r round(confint(sheep_model)[2,],2)`</span>. </span>
<span id="cb46-1610"><a href="#cb46-1610" aria-hidden="true" tabindex="-1"></a>The confidence interval limits are same sign, </span>
<span id="cb46-1611"><a href="#cb46-1611" aria-hidden="true" tabindex="-1"></a>meaning that 0 is not included as a plausible value for the strength of </span>
<span id="cb46-1612"><a href="#cb46-1612" aria-hidden="true" tabindex="-1"></a>the relationship.</span>
<span id="cb46-1613"><a href="#cb46-1613" aria-hidden="true" tabindex="-1"></a>Therefore, we can conclude that weight has seems to have a positive impact </span>
<span id="cb46-1614"><a href="#cb46-1614" aria-hidden="true" tabindex="-1"></a>on the survival of sheep.  </span>
<span id="cb46-1615"><a href="#cb46-1615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1616"><a href="#cb46-1616" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/br&gt;</span></span>
<span id="cb46-1617"><a href="#cb46-1617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1618"><a href="#cb46-1618" aria-hidden="true" tabindex="-1"></a>In model checking, all the assumptions of the Binomial GLMs were met. </span>
<span id="cb46-1619"><a href="#cb46-1619" aria-hidden="true" tabindex="-1"></a>The model was not overdispersed, the data are Binomial (coming from single trials)</span>
<span id="cb46-1620"><a href="#cb46-1620" aria-hidden="true" tabindex="-1"></a>and the fitted model line captured the pattern of the data well. </span>
<span id="cb46-1621"><a href="#cb46-1621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1622"><a href="#cb46-1622" aria-hidden="true" tabindex="-1"></a>It should be noted that there is also data on Age in the Soay sheep dataset. </span>
<span id="cb46-1623"><a href="#cb46-1623" aria-hidden="true" tabindex="-1"></a>It is reasonable to assume that Age and Body weight have some relationship. </span>
<span id="cb46-1624"><a href="#cb46-1624" aria-hidden="true" tabindex="-1"></a>Older sheep are likely to be bigger than new lambs. </span>
<span id="cb46-1625"><a href="#cb46-1625" aria-hidden="true" tabindex="-1"></a>Therefore, it might be good to also include Age in the model possibly conducting</span>
<span id="cb46-1626"><a href="#cb46-1626" aria-hidden="true" tabindex="-1"></a>model selection to decide if this variable is needed. </span>
<span id="cb46-1627"><a href="#cb46-1627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1628"><a href="#cb46-1628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1629"><a href="#cb46-1629" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb46-1630"><a href="#cb46-1630" aria-hidden="true" tabindex="-1"></a><span class="fu">## What's next? </span></span>
<span id="cb46-1631"><a href="#cb46-1631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1632"><a href="#cb46-1632" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**Poisson Generalized linear models** for analyses when your response </span>
<span id="cb46-1633"><a href="#cb46-1633" aria-hidden="true" tabindex="-1"></a>variable is not normally distributed.</span>
<span id="cb46-1634"><a href="#cb46-1634" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb46-1635"><a href="#cb46-1635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1636"><a href="#cb46-1636" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb46-1637"><a href="#cb46-1637" aria-hidden="true" tabindex="-1"></a><span class="fu">## Further reading</span></span>
<span id="cb46-1638"><a href="#cb46-1638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1639"><a href="#cb46-1639" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb46-1640"><a href="#cb46-1640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1641"><a href="#cb46-1641" aria-hidden="true" tabindex="-1"></a>::: {.column-margin}</span>
<span id="cb46-1642"><a href="#cb46-1642" aria-hidden="true" tabindex="-1"></a><span class="fu">### Contributors {.unlisted .unnumbered}</span></span>
<span id="cb46-1643"><a href="#cb46-1643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1644"><a href="#cb46-1644" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Kwaku Peprah Adjei (lead)</span>
<span id="cb46-1645"><a href="#cb46-1645" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Emily G. Simmonds</span>
<span id="cb46-1646"><a href="#cb46-1646" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Bob O'Hara</span>
<span id="cb46-1647"><a href="#cb46-1647" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb46-1648"><a href="#cb46-1648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-1649"><a href="#cb46-1649" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>